// @flyfish-group/file-viewer3@1.0.5/dist/components/ImageViewer.js
import { defineComponent as Se, ref as qe, watch as Tt, nextTick as Ot, onMounted as Ue, onUnmounted as Be, openBlock as ft, createElementBlock as Dt, renderSlot as Ze, computed as $e, createBlock as Ke, unref as Ge, withCtx as Qe, Fragment as Je, renderList as ti } from "vue";
import { _ as ei } from "./index.js";
/*!
 * Viewer.js v1.11.7
 * https://fengyuanchen.github.io/viewerjs
 *
 * Copyright 2015-present Chen Fengyuan
 * Released under the MIT license
 *
 * Date: 2024-11-24T04:32:19.116Z
 */
function ii(e, t) {
  if (!(e instanceof t))
    throw new TypeError("Cannot call a class as a function");
}
function Yt(e, t) {
  for (var i = 0; i < t.length; i++) {
    var n = t[i];
    n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), Object.defineProperty(e, xe(n.key), n);
  }
}
function ni(e, t, i) {
  return t && Yt(e.prototype, t), i && Yt(e, i), Object.defineProperty(e, "prototype", {
    writable: !1
  }), e;
}
function ri(e, t, i) {
  return (t = xe(t)) in e ? Object.defineProperty(e, t, {
    value: i,
    enumerable: !0,
    configurable: !0,
    writable: !0
  }) : e[t] = i, e;
}
function Wt(e, t) {
  var i = Object.keys(e);
  if (Object.getOwnPropertySymbols) {
    var n = Object.getOwnPropertySymbols(e);
    t && (n = n.filter(function(r) {
      return Object.getOwnPropertyDescriptor(e, r).enumerable;
    })), i.push.apply(i, n);
  }
  return i;
}
function kt(e) {
  for (var t = 1; t < arguments.length; t++) {
    var i = arguments[t] != null ? arguments[t] : {};
    t % 2 ? Wt(Object(i), !0).forEach(function(n) {
      ri(e, n, i[n]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(i)) : Wt(Object(i)).forEach(function(n) {
      Object.defineProperty(e, n, Object.getOwnPropertyDescriptor(i, n));
    });
  }
  return e;
}
function ai(e, t) {
  if (typeof e != "object" || !e)
    return e;
  var i = e[Symbol.toPrimitive];
  if (i !== void 0) {
    var n = i.call(e, t || "default");
    if (typeof n != "object")
      return n;
    throw new TypeError("@@toPrimitive must return a primitive value.");
  }
  return (t === "string" ? String : Number)(e);
}
function xe(e) {
  var t = ai(e, "string");
  return typeof t == "symbol" ? t : t + "";
}
function _t(e) {
  "@babel/helpers - typeof";
  return _t = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(t) {
    return typeof t;
  } : function(t) {
    return t && typeof Symbol == "function" && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t;
  }, _t(e);
}
var Xt = {
  /**
   * Enable a modal backdrop, specify `static` for a backdrop
   * which doesn't close the modal on click.
   * @type {boolean}
   */
  backdrop: !0,
  /**
   * Show the button on the top-right of the viewer.
   * @type {boolean}
   */
  button: !0,
  /**
   * Show the navbar.
   * @type {boolean | number}
   */
  navbar: !0,
  /**
   * Specify the visibility and the content of the title.
   * @type {boolean | number | Function | Array}
   */
  title: !0,
  /**
   * Show the toolbar.
   * @type {boolean | number | Object}
   */
  toolbar: !0,
  /**
   * Custom class name(s) to add to the viewer's root element.
   * @type {string}
   */
  className: "",
  /**
   * Define where to put the viewer in modal mode.
   * @type {string | Element}
   */
  container: "body",
  /**
   * Filter the images for viewing. Return true if the image is viewable.
   * @type {Function}
   */
  filter: null,
  /**
   * Enable to request fullscreen when play.
   * {@link https://developer.mozilla.org/en-US/docs/Web/API/FullscreenOptions}
   * @type {boolean|FullscreenOptions}
   */
  fullscreen: !0,
  /**
   * Define the extra attributes to inherit from the original image.
   * @type {Array}
   */
  inheritedAttributes: ["crossOrigin", "decoding", "isMap", "loading", "referrerPolicy", "sizes", "srcset", "useMap"],
  /**
   * Define the initial coverage of the viewing image.
   * @type {number}
   */
  initialCoverage: 0.9,
  /**
   * Define the initial index of the image for viewing.
   * @type {number}
   */
  initialViewIndex: 0,
  /**
   * Enable inline mode.
   * @type {boolean}
   */
  inline: !1,
  /**
   * The amount of time to delay between automatically cycling an image when playing.
   * @type {number}
   */
  interval: 5e3,
  /**
   * Enable keyboard support.
   * @type {boolean}
   */
  keyboard: !0,
  /**
   * Focus the viewer when initialized.
   * @type {boolean}
   */
  focus: !0,
  /**
   * Indicate if show a loading spinner when load image or not.
   * @type {boolean}
   */
  loading: !0,
  /**
   * Indicate if enable loop viewing or not.
   * @type {boolean}
   */
  loop: !0,
  /**
   * Min width of the viewer in inline mode.
   * @type {number}
   */
  minWidth: 200,
  /**
   * Min height of the viewer in inline mode.
   * @type {number}
   */
  minHeight: 100,
  /**
   * Enable to move the image.
   * @type {boolean}
   */
  movable: !0,
  /**
   * Enable to rotate the image.
   * @type {boolean}
   */
  rotatable: !0,
  /**
   * Enable to scale the image.
   * @type {boolean}
   */
  scalable: !0,
  /**
   * Enable to zoom the image.
   * @type {boolean}
   */
  zoomable: !0,
  /**
   * Enable to zoom the current image by dragging on the touch screen.
   * @type {boolean}
   */
  zoomOnTouch: !0,
  /**
   * Enable to zoom the image by wheeling mouse.
   * @type {boolean}
   */
  zoomOnWheel: !0,
  /**
   * Enable to slide to the next or previous image by swiping on the touch screen.
   * @type {boolean}
   */
  slideOnTouch: !0,
  /**
   * Indicate if toggle the image size between its natural size
   * and initial size when double click on the image or not.
   * @type {boolean}
   */
  toggleOnDblclick: !0,
  /**
   * Show the tooltip with image ratio (percentage) when zoom in or zoom out.
   * @type {boolean}
   */
  tooltip: !0,
  /**
   * Enable CSS3 Transition for some special elements.
   * @type {boolean}
   */
  transition: !0,
  /**
   * Define the CSS `z-index` value of viewer in modal mode.
   * @type {number}
   */
  zIndex: 2015,
  /**
   * Define the CSS `z-index` value of viewer in inline mode.
   * @type {number}
   */
  zIndexInline: 0,
  /**
   * Define the ratio when zoom the image by wheeling mouse.
   * @type {number}
   */
  zoomRatio: 0.1,
  /**
   * Define the min ratio of the image when zoom out.
   * @type {number}
   */
  minZoomRatio: 0.01,
  /**
   * Define the max ratio of the image when zoom in.
   * @type {number}
   */
  maxZoomRatio: 100,
  /**
   * Define where to get the original image URL for viewing.
   * @type {string | Function}
   */
  url: "src",
  /**
   * Event shortcuts.
   * @type {Function}
   */
  ready: null,
  show: null,
  shown: null,
  hide: null,
  hidden: null,
  view: null,
  viewed: null,
  move: null,
  moved: null,
  rotate: null,
  rotated: null,
  scale: null,
  scaled: null,
  zoom: null,
  zoomed: null,
  play: null,
  stop: null
}, oi = '<div class="viewer-container" tabindex="-1" touch-action="none"><div class="viewer-canvas"></div><div class="viewer-footer"><div class="viewer-title"></div><div class="viewer-toolbar"></div><div class="viewer-navbar"><ul class="viewer-list" role="navigation"></ul></div></div><div class="viewer-tooltip" role="alert" aria-hidden="true"></div><div class="viewer-button" data-viewer-action="mix" role="button"></div><div class="viewer-player"></div></div>', yt = typeof window != "undefined" && typeof window.document != "undefined", F = yt ? window : {}, K = yt && F.document.documentElement ? "ontouchstart" in F.document.documentElement : !1, jt = yt ? "PointerEvent" in F : !1, v = "viewer", dt = "move", Ae = "switch", it = "zoom", lt = "".concat(v, "-active"), si = "".concat(v, "-close"), vt = "".concat(v, "-fade"), It = "".concat(v, "-fixed"), li = "".concat(v, "-fullscreen"), Ht = "".concat(v, "-fullscreen-exit"), W = "".concat(v, "-hide"), ui = "".concat(v, "-hide-md-down"), ci = "".concat(v, "-hide-sm-down"), hi = "".concat(v, "-hide-xs-down"), k = "".concat(v, "-in"), nt = "".concat(v, "-invisible"), G = "".concat(v, "-loading"), fi = "".concat(v, "-move"), qt = "".concat(v, "-open"), q = "".concat(v, "-show"), x = "".concat(v, "-transition"), Q = "click", Nt = "dblclick", Ut = "dragstart", Bt = "focusin", Zt = "keydown", j = "load", X = "error", di = K ? "touchend touchcancel" : "mouseup", vi = K ? "touchmove" : "mousemove", mi = K ? "touchstart" : "mousedown", $t = jt ? "pointerdown" : mi, Kt = jt ? "pointermove" : vi, Gt = jt ? "pointerup pointercancel" : di, Qt = "resize", z = "transitionend", Jt = "wheel", te = "ready", ee = "show", ie = "shown", ne = "hide", re = "hidden", ae = "view", at = "viewed", oe = "move", se = "moved", le = "rotate", ue = "rotated", ce = "scale", he = "scaled", fe = "zoom", de = "zoomed", ve = "play", me = "stop", gt = "".concat(v, "Action"), zt = /\s\s*/, ut = ["zoom-in", "zoom-out", "one-to-one", "reset", "prev", "play", "next", "rotate-left", "rotate-right", "flip-horizontal", "flip-vertical"];
function ot(e) {
  return typeof e == "string";
}
var gi = Number.isNaN || F.isNaN;
function O(e) {
  return typeof e == "number" && !gi(e);
}
function B(e) {
  return typeof e == "undefined";
}
function J(e) {
  return _t(e) === "object" && e !== null;
}
var pi = Object.prototype.hasOwnProperty;
function Z(e) {
  if (!J(e))
    return !1;
  try {
    var t = e.constructor, i = t.prototype;
    return t && i && pi.call(i, "isPrototypeOf");
  } catch (n) {
    return !1;
  }
}
function p(e) {
  return typeof e == "function";
}
function y(e, t) {
  if (e && p(t))
    if (Array.isArray(e) || O(e.length)) {
      var i = e.length, n;
      for (n = 0; n < i && t.call(e, e[n], n, e) !== !1; n += 1)
        ;
    } else
      J(e) && Object.keys(e).forEach(function(r) {
        t.call(e, e[r], r, e);
      });
  return e;
}
var C = Object.assign || function(t) {
  for (var i = arguments.length, n = new Array(i > 1 ? i - 1 : 0), r = 1; r < i; r++)
    n[r - 1] = arguments[r];
  return J(t) && n.length > 0 && n.forEach(function(a) {
    J(a) && Object.keys(a).forEach(function(o) {
      t[o] = a[o];
    });
  }), t;
}, bi = /^(?:width|height|left|top|marginLeft|marginTop)$/;
function L(e, t) {
  var i = e.style;
  y(t, function(n, r) {
    bi.test(r) && O(n) && (n += "px"), i[r] = n;
  });
}
function yi(e) {
  return ot(e) ? e.replace(/&(?!amp;|quot;|#39;|lt;|gt;)/g, "&amp;").replace(/"/g, "&quot;").replace(/'/g, "&#39;").replace(/</g, "&lt;").replace(/>/g, "&gt;") : e;
}
function U(e, t) {
  return !e || !t ? !1 : e.classList ? e.classList.contains(t) : e.className.indexOf(t) > -1;
}
function f(e, t) {
  if (!(!e || !t)) {
    if (O(e.length)) {
      y(e, function(n) {
        f(n, t);
      });
      return;
    }
    if (e.classList) {
      e.classList.add(t);
      return;
    }
    var i = e.className.trim();
    i ? i.indexOf(t) < 0 && (e.className = "".concat(i, " ").concat(t)) : e.className = t;
  }
}
function g(e, t) {
  if (!(!e || !t)) {
    if (O(e.length)) {
      y(e, function(i) {
        g(i, t);
      });
      return;
    }
    if (e.classList) {
      e.classList.remove(t);
      return;
    }
    e.className.indexOf(t) >= 0 && (e.className = e.className.replace(t, ""));
  }
}
function st(e, t, i) {
  if (t) {
    if (O(e.length)) {
      y(e, function(n) {
        st(n, t, i);
      });
      return;
    }
    i ? f(e, t) : g(e, t);
  }
}
var wi = /([a-z\d])([A-Z])/g;
function Lt(e) {
  return e.replace(wi, "$1-$2").toLowerCase();
}
function $(e, t) {
  return J(e[t]) ? e[t] : e.dataset ? e.dataset[t] : e.getAttribute("data-".concat(Lt(t)));
}
function Ct(e, t, i) {
  J(i) ? e[t] = i : e.dataset ? e.dataset[t] = i : e.setAttribute("data-".concat(Lt(t)), i);
}
var De = function() {
  var e = !1;
  if (yt) {
    var t = !1, i = function() {
    }, n = Object.defineProperty({}, "once", {
      get: function() {
        return e = !0, t;
      },
      /**
       * This setter can fix a `TypeError` in strict mode
       * {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Getter_only}
       * @param {boolean} value - The value to set
       */
      set: function(a) {
        t = a;
      }
    });
    F.addEventListener("test", i, n), F.removeEventListener("test", i, n);
  }
  return e;
}();
function b(e, t, i) {
  var n = arguments.length > 3 && arguments[3] !== void 0 ? arguments[3] : {}, r = i;
  t.trim().split(zt).forEach(function(a) {
    if (!De) {
      var o = e.listeners;
      o && o[a] && o[a][i] && (r = o[a][i], delete o[a][i], Object.keys(o[a]).length === 0 && delete o[a], Object.keys(o).length === 0 && delete e.listeners);
    }
    e.removeEventListener(a, r, n);
  });
}
function d(e, t, i) {
  var n = arguments.length > 3 && arguments[3] !== void 0 ? arguments[3] : {}, r = i;
  t.trim().split(zt).forEach(function(a) {
    if (n.once && !De) {
      var o = e.listeners, s = o === void 0 ? {} : o;
      r = function() {
        delete s[a][i], e.removeEventListener(a, r, n);
        for (var l = arguments.length, u = new Array(l), h = 0; h < l; h++)
          u[h] = arguments[h];
        i.apply(e, u);
      }, s[a] || (s[a] = {}), s[a][i] && e.removeEventListener(a, s[a][i], n), s[a][i] = r, e.listeners = s;
    }
    e.addEventListener(a, r, n);
  });
}
function D(e, t, i, n) {
  var r;
  return p(Event) && p(CustomEvent) ? r = new CustomEvent(t, kt({
    bubbles: !0,
    cancelable: !0,
    detail: i
  }, n)) : (r = document.createEvent("CustomEvent"), r.initCustomEvent(t, !0, !0, i)), e.dispatchEvent(r);
}
function Ei(e) {
  var t = e.getBoundingClientRect();
  return {
    left: t.left + (window.pageXOffset - document.documentElement.clientLeft),
    top: t.top + (window.pageYOffset - document.documentElement.clientTop)
  };
}
function mt(e) {
  var t = e.rotate, i = e.scaleX, n = e.scaleY, r = e.translateX, a = e.translateY, o = [];
  O(r) && r !== 0 && o.push("translateX(".concat(r, "px)")), O(a) && a !== 0 && o.push("translateY(".concat(a, "px)")), O(t) && t !== 0 && o.push("rotate(".concat(t, "deg)")), O(i) && i !== 1 && o.push("scaleX(".concat(i, ")")), O(n) && n !== 1 && o.push("scaleY(".concat(n, ")"));
  var s = o.length ? o.join(" ") : "none";
  return {
    WebkitTransform: s,
    msTransform: s,
    transform: s
  };
}
function Ti(e) {
  return ot(e) ? decodeURIComponent(e.replace(/^.*\//, "").replace(/[?&#].*$/, "")) : "";
}
var St = F.navigator && /Version\/\d+(\.\d+)+?\s+Safari/i.test(F.navigator.userAgent);
function _e(e, t, i) {
  var n = document.createElement("img");
  if (e.naturalWidth && !St)
    return i(e.naturalWidth, e.naturalHeight), n;
  var r = document.body || document.documentElement;
  return n.onload = function() {
    i(n.width, n.height), St || r.removeChild(n);
  }, y(t.inheritedAttributes, function(a) {
    var o = e.getAttribute(a);
    o !== null && n.setAttribute(a, o);
  }), n.src = e.src, St || (n.style.cssText = "left:0;max-height:none!important;max-width:none!important;min-height:0!important;min-width:0!important;opacity:0;position:absolute;top:0;z-index:-1;", r.appendChild(n)), n;
}
function ct(e) {
  switch (e) {
    case 2:
      return hi;
    case 3:
      return ci;
    case 4:
      return ui;
    default:
      return "";
  }
}
function Oi(e) {
  var t = kt({}, e), i = [];
  return y(e, function(n, r) {
    delete t[r], y(t, function(a) {
      var o = Math.abs(n.startX - a.startX), s = Math.abs(n.startY - a.startY), c = Math.abs(n.endX - a.endX), l = Math.abs(n.endY - a.endY), u = Math.sqrt(o * o + s * s), h = Math.sqrt(c * c + l * l), m = (h - u) / u;
      i.push(m);
    });
  }), i.sort(function(n, r) {
    return Math.abs(n) < Math.abs(r);
  }), i[0];
}
function ht(e, t) {
  var i = e.pageX, n = e.pageY, r = {
    endX: i,
    endY: n
  };
  return t ? r : kt({
    timeStamp: Date.now(),
    startX: i,
    startY: n
  }, r);
}
function Si(e) {
  var t = 0, i = 0, n = 0;
  return y(e, function(r) {
    var a = r.startX, o = r.startY;
    t += a, i += o, n += 1;
  }), t /= n, i /= n, {
    pageX: t,
    pageY: i
  };
}
var xi = {
  render: function() {
    this.initContainer(), this.initViewer(), this.initList(), this.renderViewer();
  },
  initBody: function() {
    var t = this.element.ownerDocument, i = t.body || t.documentElement;
    this.body = i, this.scrollbarWidth = window.innerWidth - t.documentElement.clientWidth, this.initialBodyPaddingRight = i.style.paddingRight, this.initialBodyComputedPaddingRight = window.getComputedStyle(i).paddingRight;
  },
  initContainer: function() {
    this.containerData = {
      width: window.innerWidth,
      height: window.innerHeight
    };
  },
  initViewer: function() {
    var t = this.options, i = this.parent, n;
    t.inline && (n = {
      width: Math.max(i.offsetWidth, t.minWidth),
      height: Math.max(i.offsetHeight, t.minHeight)
    }, this.parentData = n), (this.fulled || !n) && (n = this.containerData), this.viewerData = C({}, n);
  },
  renderViewer: function() {
    this.options.inline && !this.fulled && L(this.viewer, this.viewerData);
  },
  initList: function() {
    var t = this, i = this.element, n = this.options, r = this.list, a = [];
    r.innerHTML = "", y(this.images, function(o, s) {
      var c = o.src, l = o.alt || Ti(c), u = t.getImageURL(o);
      if (c || u) {
        var h = document.createElement("li"), m = document.createElement("img");
        y(n.inheritedAttributes, function(A) {
          var S = o.getAttribute(A);
          S !== null && m.setAttribute(A, S);
        }), n.navbar && (m.src = c || u), m.alt = l, m.setAttribute("data-original-url", u || c), h.setAttribute("data-index", s), h.setAttribute("data-viewer-action", "view"), h.setAttribute("role", "button"), n.keyboard && h.setAttribute("tabindex", 0), h.appendChild(m), r.appendChild(h), a.push(h);
      }
    }), this.items = a, y(a, function(o) {
      var s = o.firstElementChild, c, l;
      Ct(s, "filled", !0), n.loading && f(o, G), d(s, j, c = function(h) {
        b(s, X, l), n.loading && g(o, G), t.loadImage(h);
      }, {
        once: !0
      }), d(s, X, l = function() {
        b(s, j, c), n.loading && g(o, G);
      }, {
        once: !0
      });
    }), n.transition && d(i, at, function() {
      f(r, x);
    }, {
      once: !0
    });
  },
  renderList: function() {
    var t = this.index, i = this.items[t];
    if (i) {
      var n = i.nextElementSibling, r = parseInt(window.getComputedStyle(n || i).marginLeft, 10), a = i.offsetWidth, o = a + r;
      L(this.list, C({
        width: o * this.length - r
      }, mt({
        translateX: (this.viewerData.width - a) / 2 - o * t
      })));
    }
  },
  resetList: function() {
    var t = this.list;
    t.innerHTML = "", g(t, x), L(t, mt({
      translateX: 0
    }));
  },
  initImage: function(t) {
    var i = this, n = this.options, r = this.image, a = this.viewerData, o = this.footer.offsetHeight, s = a.width, c = Math.max(a.height - o, o), l = this.imageData || {}, u;
    this.imageInitializing = {
      abort: function() {
        u.onload = null;
      }
    }, u = _e(r, n, function(h, m) {
      var A = h / m, S = Math.max(0, Math.min(1, n.initialCoverage)), T = s, _ = c;
      i.imageInitializing = !1, c * A > s ? _ = s / A : T = c * A, S = O(S) ? S : 0.9, T = Math.min(T * S, h), _ = Math.min(_ * S, m);
      var I = (s - T) / 2, N = (c - _) / 2, E = {
        left: I,
        top: N,
        x: I,
        y: N,
        width: T,
        height: _,
        oldRatio: 1,
        ratio: T / h,
        aspectRatio: A,
        naturalWidth: h,
        naturalHeight: m
      }, R = C({}, E);
      n.rotatable && (E.rotate = l.rotate || 0, R.rotate = 0), n.scalable && (E.scaleX = l.scaleX || 1, E.scaleY = l.scaleY || 1, R.scaleX = 1, R.scaleY = 1), i.imageData = E, i.initialImageData = R, t && t();
    });
  },
  renderImage: function(t) {
    var i = this, n = this.image, r = this.imageData;
    if (L(n, C({
      width: r.width,
      height: r.height,
      // XXX: Not to use translateX/Y to avoid image shaking when zooming
      marginLeft: r.x,
      marginTop: r.y
    }, mt(r))), t)
      if ((this.viewing || this.moving || this.rotating || this.scaling || this.zooming) && this.options.transition && U(n, x)) {
        var a = function() {
          i.imageRendering = !1, t();
        };
        this.imageRendering = {
          abort: function() {
            b(n, z, a);
          }
        }, d(n, z, a, {
          once: !0
        });
      } else
        t();
  },
  resetImage: function() {
    var t = this.image;
    t && (this.viewing && this.viewing.abort(), t.parentNode.removeChild(t), this.image = null, this.title.innerHTML = "");
  }
}, Ai = {
  bind: function() {
    var t = this.options, i = this.viewer, n = this.canvas, r = this.element.ownerDocument;
    d(i, Q, this.onClick = this.click.bind(this)), d(i, Ut, this.onDragStart = this.dragstart.bind(this)), d(n, $t, this.onPointerDown = this.pointerdown.bind(this)), d(r, Kt, this.onPointerMove = this.pointermove.bind(this)), d(r, Gt, this.onPointerUp = this.pointerup.bind(this)), d(r, Zt, this.onKeyDown = this.keydown.bind(this)), d(window, Qt, this.onResize = this.resize.bind(this)), t.zoomable && t.zoomOnWheel && d(i, Jt, this.onWheel = this.wheel.bind(this), {
      passive: !1,
      capture: !0
    }), t.toggleOnDblclick && d(n, Nt, this.onDblclick = this.dblclick.bind(this));
  },
  unbind: function() {
    var t = this.options, i = this.viewer, n = this.canvas, r = this.element.ownerDocument;
    b(i, Q, this.onClick), b(i, Ut, this.onDragStart), b(n, $t, this.onPointerDown), b(r, Kt, this.onPointerMove), b(r, Gt, this.onPointerUp), b(r, Zt, this.onKeyDown), b(window, Qt, this.onResize), t.zoomable && t.zoomOnWheel && b(i, Jt, this.onWheel, {
      passive: !1,
      capture: !0
    }), t.toggleOnDblclick && b(n, Nt, this.onDblclick);
  }
}, Di = {
  click: function(t) {
    var i = this.options, n = this.imageData, r = t.target, a = $(r, gt);
    switch (!a && r.localName === "img" && r.parentElement.localName === "li" && (r = r.parentElement, a = $(r, gt)), K && t.isTrusted && r === this.canvas && clearTimeout(this.clickCanvasTimeout), a) {
      case "mix":
        this.played ? this.stop() : i.inline ? this.fulled ? this.exit() : this.full() : this.hide();
        break;
      case "hide":
        this.pointerMoved || this.hide();
        break;
      case "view":
        this.view($(r, "index"));
        break;
      case "zoom-in":
        this.zoom(0.1, !0);
        break;
      case "zoom-out":
        this.zoom(-0.1, !0);
        break;
      case "one-to-one":
        this.toggle();
        break;
      case "reset":
        this.reset();
        break;
      case "prev":
        this.prev(i.loop);
        break;
      case "play":
        this.play(i.fullscreen);
        break;
      case "next":
        this.next(i.loop);
        break;
      case "rotate-left":
        this.rotate(-90);
        break;
      case "rotate-right":
        this.rotate(90);
        break;
      case "flip-horizontal":
        this.scaleX(-n.scaleX || -1);
        break;
      case "flip-vertical":
        this.scaleY(-n.scaleY || -1);
        break;
      default:
        this.played && this.stop();
    }
  },
  dblclick: function(t) {
    t.preventDefault(), this.viewed && t.target === this.image && (K && t.isTrusted && clearTimeout(this.doubleClickImageTimeout), this.toggle(t.isTrusted ? t : t.detail && t.detail.originalEvent));
  },
  load: function() {
    var t = this;
    this.timeout && (clearTimeout(this.timeout), this.timeout = !1);
    var i = this.element, n = this.options, r = this.image, a = this.index, o = this.viewerData;
    g(r, nt), n.loading && g(this.canvas, G), r.style.cssText = "height:0;" + "margin-left:".concat(o.width / 2, "px;") + "margin-top:".concat(o.height / 2, "px;") + "max-width:none!important;position:relative;width:0;", this.initImage(function() {
      st(r, fi, n.movable), st(r, x, n.transition), t.renderImage(function() {
        t.viewed = !0, t.viewing = !1, p(n.viewed) && d(i, at, n.viewed, {
          once: !0
        }), D(i, at, {
          originalImage: t.images[a],
          index: a,
          image: r
        }, {
          cancelable: !1
        });
      });
    });
  },
  loadImage: function(t) {
    var i = t.target, n = i.parentNode, r = n.offsetWidth || 30, a = n.offsetHeight || 50, o = !!$(i, "filled");
    _e(i, this.options, function(s, c) {
      var l = s / c, u = r, h = a;
      a * l > r ? o ? u = a * l : h = r / l : o ? h = r / l : u = a * l, L(i, C({
        width: u,
        height: h
      }, mt({
        translateX: (r - u) / 2,
        translateY: (a - h) / 2
      })));
    });
  },
  keydown: function(t) {
    var i = this.options;
    if (i.keyboard) {
      var n = t.keyCode || t.which || t.charCode;
      switch (n) {
        case 13:
          this.viewer.contains(t.target) && this.click(t);
          break;
      }
      if (this.fulled)
        switch (n) {
          case 27:
            this.played ? this.stop() : i.inline ? this.fulled && this.exit() : this.hide();
            break;
          case 32:
            this.played && this.stop();
            break;
          case 37:
            this.played && this.playing ? this.playing.prev() : this.prev(i.loop);
            break;
          case 38:
            t.preventDefault(), this.zoom(i.zoomRatio, !0);
            break;
          case 39:
            this.played && this.playing ? this.playing.next() : this.next(i.loop);
            break;
          case 40:
            t.preventDefault(), this.zoom(-i.zoomRatio, !0);
            break;
          case 48:
          case 49:
            t.ctrlKey && (t.preventDefault(), this.toggle());
            break;
        }
    }
  },
  dragstart: function(t) {
    t.target.localName === "img" && t.preventDefault();
  },
  pointerdown: function(t) {
    var i = this.options, n = this.pointers, r = t.buttons, a = t.button;
    if (this.pointerMoved = !1, !(!this.viewed || this.showing || this.viewing || this.hiding || (t.type === "mousedown" || t.type === "pointerdown" && t.pointerType === "mouse") && // No primary button (Usually the left button)
    (O(r) && r !== 1 || O(a) && a !== 0 || t.ctrlKey))) {
      t.preventDefault(), t.changedTouches ? y(t.changedTouches, function(s) {
        n[s.identifier] = ht(s);
      }) : n[t.pointerId || 0] = ht(t);
      var o = i.movable ? dt : !1;
      i.zoomOnTouch && i.zoomable && Object.keys(n).length > 1 ? o = it : i.slideOnTouch && (t.pointerType === "touch" || t.type === "touchstart") && this.isSwitchable() && (o = Ae), i.transition && (o === dt || o === it) && g(this.image, x), this.action = o;
    }
  },
  pointermove: function(t) {
    var i = this.pointers, n = this.action;
    !this.viewed || !n || (t.preventDefault(), t.changedTouches ? y(t.changedTouches, function(r) {
      C(i[r.identifier] || {}, ht(r, !0));
    }) : C(i[t.pointerId || 0] || {}, ht(t, !0)), this.change(t));
  },
  pointerup: function(t) {
    var i = this, n = this.options, r = this.action, a = this.pointers, o;
    t.changedTouches ? y(t.changedTouches, function(s) {
      o = a[s.identifier], delete a[s.identifier];
    }) : (o = a[t.pointerId || 0], delete a[t.pointerId || 0]), r && (t.preventDefault(), n.transition && (r === dt || r === it) && f(this.image, x), this.action = !1, K && r !== it && o && Date.now() - o.timeStamp < 500 && (clearTimeout(this.clickCanvasTimeout), clearTimeout(this.doubleClickImageTimeout), n.toggleOnDblclick && this.viewed && t.target === this.image ? this.imageClicked ? (this.imageClicked = !1, this.doubleClickImageTimeout = setTimeout(function() {
      D(i.image, Nt, {
        originalEvent: t
      });
    }, 50)) : (this.imageClicked = !0, this.doubleClickImageTimeout = setTimeout(function() {
      i.imageClicked = !1;
    }, 500)) : (this.imageClicked = !1, n.backdrop && n.backdrop !== "static" && t.target === this.canvas && (this.clickCanvasTimeout = setTimeout(function() {
      D(i.canvas, Q, {
        originalEvent: t
      });
    }, 50)))));
  },
  resize: function() {
    var t = this;
    if (!(!this.isShown || this.hiding) && (this.fulled && (this.close(), this.initBody(), this.open()), this.initContainer(), this.initViewer(), this.renderViewer(), this.renderList(), this.viewed && this.initImage(function() {
      t.renderImage();
    }), this.played)) {
      if (this.options.fullscreen && this.fulled && !(document.fullscreenElement || document.webkitFullscreenElement || document.mozFullScreenElement || document.msFullscreenElement)) {
        this.stop();
        return;
      }
      y(this.player.getElementsByTagName("img"), function(i) {
        d(i, j, t.loadImage.bind(t), {
          once: !0
        }), D(i, j);
      });
    }
  },
  wheel: function(t) {
    var i = this;
    if (this.viewed && (t.preventDefault(), !this.wheeling)) {
      this.wheeling = !0, setTimeout(function() {
        i.wheeling = !1;
      }, 50);
      var n = Number(this.options.zoomRatio) || 0.1, r = 1;
      t.deltaY ? r = t.deltaY > 0 ? 1 : -1 : t.wheelDelta ? r = -t.wheelDelta / 120 : t.detail && (r = t.detail > 0 ? 1 : -1), this.zoom(-r * n, !0, null, t);
    }
  }
}, _i = {
  /** Show the viewer (only available in modal mode)
   * @param {boolean} [immediate=false] - Indicates if show the viewer immediately or not.
   * @returns {Viewer} this
   */
  show: function() {
    var t = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : !1, i = this.element, n = this.options;
    if (n.inline || this.showing || this.isShown || this.showing)
      return this;
    if (!this.ready)
      return this.build(), this.ready && this.show(t), this;
    if (p(n.show) && d(i, ee, n.show, {
      once: !0
    }), D(i, ee) === !1 || !this.ready)
      return this;
    this.hiding && this.transitioning.abort(), this.showing = !0, this.open();
    var r = this.viewer;
    if (g(r, W), r.setAttribute("role", "dialog"), r.setAttribute("aria-labelledby", this.title.id), r.setAttribute("aria-modal", !0), r.removeAttribute("aria-hidden"), n.transition && !t) {
      var a = this.shown.bind(this);
      this.transitioning = {
        abort: function() {
          b(r, z, a), g(r, k);
        }
      }, f(r, x), r.initialOffsetWidth = r.offsetWidth, d(r, z, a, {
        once: !0
      }), f(r, k);
    } else
      f(r, k), this.shown();
    return this;
  },
  /**
   * Hide the viewer (only available in modal mode)
   * @param {boolean} [immediate=false] - Indicates if hide the viewer immediately or not.
   * @returns {Viewer} this
   */
  hide: function() {
    var t = this, i = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : !1, n = this.element, r = this.options;
    if (r.inline || this.hiding || !(this.isShown || this.showing))
      return this;
    if (p(r.hide) && d(n, ne, r.hide, {
      once: !0
    }), D(n, ne) === !1)
      return this;
    this.showing && this.transitioning.abort(), this.hiding = !0, this.played ? this.stop() : this.viewing && this.viewing.abort();
    var a = this.viewer, o = this.image, s = function() {
      g(a, k), t.hidden();
    };
    if (r.transition && !i) {
      var c = function(h) {
        h && h.target === a && (b(a, z, c), t.hidden());
      }, l = function() {
        U(a, x) ? (d(a, z, c), g(a, k)) : s();
      };
      this.transitioning = {
        abort: function() {
          t.viewed && U(o, x) ? b(o, z, l) : U(a, x) && b(a, z, c);
        }
      }, this.viewed && U(o, x) ? (d(o, z, l, {
        once: !0
      }), this.zoomTo(0, !1, null, null, !0)) : l();
    } else
      s();
    return this;
  },
  /**
   * View one of the images with image's index
   * @param {number} index - The index of the image to view.
   * @returns {Viewer} this
   */
  view: function() {
    var t = this, i = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : this.options.initialViewIndex;
    if (i = Number(i) || 0, this.hiding || this.played || i < 0 || i >= this.length || this.viewed && i === this.index)
      return this;
    if (!this.isShown)
      return this.index = i, this.show();
    this.viewing && this.viewing.abort();
    var n = this.element, r = this.options, a = this.title, o = this.canvas, s = this.items[i], c = s.querySelector("img"), l = $(c, "originalUrl"), u = c.getAttribute("alt"), h = document.createElement("img");
    if (y(r.inheritedAttributes, function(_) {
      var I = c.getAttribute(_);
      I !== null && h.setAttribute(_, I);
    }), h.src = l, h.alt = u, p(r.view) && d(n, ae, r.view, {
      once: !0
    }), D(n, ae, {
      originalImage: this.images[i],
      index: i,
      image: h
    }) === !1 || !this.isShown || this.hiding || this.played)
      return this;
    var m = this.items[this.index];
    m && (g(m, lt), m.removeAttribute("aria-selected")), f(s, lt), s.setAttribute("aria-selected", !0), r.focus && s.focus(), this.image = h, this.viewed = !1, this.index = i, this.imageData = {}, f(h, nt), r.loading && f(o, G), o.innerHTML = "", o.appendChild(h), this.renderList(), a.innerHTML = "";
    var A = function() {
      var I = t.imageData, N = Array.isArray(r.title) ? r.title[1] : r.title;
      a.innerHTML = yi(p(N) ? N.call(t, h, I) : "".concat(u, " (").concat(I.naturalWidth, " × ").concat(I.naturalHeight, ")"));
    }, S, T;
    return d(n, at, A, {
      once: !0
    }), this.viewing = {
      abort: function() {
        b(n, at, A), h.complete ? t.imageRendering ? t.imageRendering.abort() : t.imageInitializing && t.imageInitializing.abort() : (h.src = "", b(h, j, S), t.timeout && clearTimeout(t.timeout));
      }
    }, h.complete ? this.load() : (d(h, j, S = function() {
      b(h, X, T), t.load();
    }, {
      once: !0
    }), d(h, X, T = function() {
      b(h, j, S), t.timeout && (clearTimeout(t.timeout), t.timeout = !1), g(h, nt), r.loading && g(t.canvas, G);
    }, {
      once: !0
    }), this.timeout && clearTimeout(this.timeout), this.timeout = setTimeout(function() {
      g(h, nt), t.timeout = !1;
    }, 1e3)), this;
  },
  /**
   * View the previous image
   * @param {boolean} [loop=false] - Indicate if view the last one
   * when it is the first one at present.
   * @returns {Viewer} this
   */
  prev: function() {
    var t = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : !1, i = this.index - 1;
    return i < 0 && (i = t ? this.length - 1 : 0), this.view(i), this;
  },
  /**
   * View the next image
   * @param {boolean} [loop=false] - Indicate if view the first one
   * when it is the last one at present.
   * @returns {Viewer} this
   */
  next: function() {
    var t = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : !1, i = this.length - 1, n = this.index + 1;
    return n > i && (n = t ? 0 : i), this.view(n), this;
  },
  /**
   * Move the image with relative offsets.
   * @param {number} x - The moving distance in the horizontal direction.
   * @param {number} [y=x] The moving distance in the vertical direction.
   * @returns {Viewer} this
   */
  move: function(t) {
    var i = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : t, n = this.imageData;
    return this.moveTo(B(t) ? t : n.x + Number(t), B(i) ? i : n.y + Number(i)), this;
  },
  /**
   * Move the image to an absolute point.
   * @param {number} x - The new position in the horizontal direction.
   * @param {number} [y=x] - The new position in the vertical direction.
   * @param {Event} [_originalEvent=null] - The original event if any.
   * @returns {Viewer} this
   */
  moveTo: function(t) {
    var i = this, n = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : t, r = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : null, a = this.element, o = this.options, s = this.imageData;
    if (t = Number(t), n = Number(n), this.viewed && !this.played && o.movable) {
      var c = s.x, l = s.y, u = !1;
      if (O(t) ? u = !0 : t = c, O(n) ? u = !0 : n = l, u) {
        if (p(o.move) && d(a, oe, o.move, {
          once: !0
        }), D(a, oe, {
          x: t,
          y: n,
          oldX: c,
          oldY: l,
          originalEvent: r
        }) === !1)
          return this;
        s.x = t, s.y = n, s.left = t, s.top = n, this.moving = !0, this.renderImage(function() {
          i.moving = !1, p(o.moved) && d(a, se, o.moved, {
            once: !0
          }), D(a, se, {
            x: t,
            y: n,
            oldX: c,
            oldY: l,
            originalEvent: r
          }, {
            cancelable: !1
          });
        });
      }
    }
    return this;
  },
  /**
   * Rotate the image with a relative degree.
   * @param {number} degree - The rotate degree.
   * @returns {Viewer} this
   */
  rotate: function(t) {
    return this.rotateTo((this.imageData.rotate || 0) + Number(t)), this;
  },
  /**
   * Rotate the image to an absolute degree.
   * @param {number} degree - The rotate degree.
   * @returns {Viewer} this
   */
  rotateTo: function(t) {
    var i = this, n = this.element, r = this.options, a = this.imageData;
    if (t = Number(t), O(t) && this.viewed && !this.played && r.rotatable) {
      var o = a.rotate;
      if (p(r.rotate) && d(n, le, r.rotate, {
        once: !0
      }), D(n, le, {
        degree: t,
        oldDegree: o
      }) === !1)
        return this;
      a.rotate = t, this.rotating = !0, this.renderImage(function() {
        i.rotating = !1, p(r.rotated) && d(n, ue, r.rotated, {
          once: !0
        }), D(n, ue, {
          degree: t,
          oldDegree: o
        }, {
          cancelable: !1
        });
      });
    }
    return this;
  },
  /**
   * Scale the image on the x-axis.
   * @param {number} scaleX - The scale ratio on the x-axis.
   * @returns {Viewer} this
   */
  scaleX: function(t) {
    return this.scale(t, this.imageData.scaleY), this;
  },
  /**
   * Scale the image on the y-axis.
   * @param {number} scaleY - The scale ratio on the y-axis.
   * @returns {Viewer} this
   */
  scaleY: function(t) {
    return this.scale(this.imageData.scaleX, t), this;
  },
  /**
   * Scale the image.
   * @param {number} scaleX - The scale ratio on the x-axis.
   * @param {number} [scaleY=scaleX] - The scale ratio on the y-axis.
   * @returns {Viewer} this
   */
  scale: function(t) {
    var i = this, n = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : t, r = this.element, a = this.options, o = this.imageData;
    if (t = Number(t), n = Number(n), this.viewed && !this.played && a.scalable) {
      var s = o.scaleX, c = o.scaleY, l = !1;
      if (O(t) ? l = !0 : t = s, O(n) ? l = !0 : n = c, l) {
        if (p(a.scale) && d(r, ce, a.scale, {
          once: !0
        }), D(r, ce, {
          scaleX: t,
          scaleY: n,
          oldScaleX: s,
          oldScaleY: c
        }) === !1)
          return this;
        o.scaleX = t, o.scaleY = n, this.scaling = !0, this.renderImage(function() {
          i.scaling = !1, p(a.scaled) && d(r, he, a.scaled, {
            once: !0
          }), D(r, he, {
            scaleX: t,
            scaleY: n,
            oldScaleX: s,
            oldScaleY: c
          }, {
            cancelable: !1
          });
        });
      }
    }
    return this;
  },
  /**
   * Zoom the image with a relative ratio.
   * @param {number} ratio - The target ratio.
   * @param {boolean} [showTooltip=false] - Indicates whether to show the tooltip.
   * @param {Object} [pivot] - The pivot point coordinate for zooming.
   * @param {Event} [_originalEvent=null] - The original event if any.
   * @returns {Viewer} this
   */
  zoom: function(t) {
    var i = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : !1, n = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : null, r = arguments.length > 3 && arguments[3] !== void 0 ? arguments[3] : null, a = this.imageData;
    return t = Number(t), t < 0 ? t = 1 / (1 - t) : t = 1 + t, this.zoomTo(a.width * t / a.naturalWidth, i, n, r), this;
  },
  /**
   * Zoom the image to an absolute ratio.
   * @param {number} ratio - The target ratio.
   * @param {boolean} [showTooltip] - Indicates whether to show the tooltip.
   * @param {Object} [pivot] - The pivot point coordinate for zooming.
   * @param {Event} [_originalEvent=null] - The original event if any.
   * @param {Event} [_zoomable=false] - Indicates if the current zoom is available or not.
   * @returns {Viewer} this
   */
  zoomTo: function(t) {
    var i = this, n = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : !1, r = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : null, a = arguments.length > 3 && arguments[3] !== void 0 ? arguments[3] : null, o = arguments.length > 4 && arguments[4] !== void 0 ? arguments[4] : !1, s = this.element, c = this.options, l = this.pointers, u = this.imageData, h = u.x, m = u.y, A = u.width, S = u.height, T = u.naturalWidth, _ = u.naturalHeight;
    if (t = Math.max(0, t), O(t) && this.viewed && !this.played && (o || c.zoomable)) {
      if (!o) {
        var I = Math.max(0.01, c.minZoomRatio), N = Math.min(100, c.maxZoomRatio);
        t = Math.min(Math.max(t, I), N);
      }
      if (a)
        switch (a.type) {
          case "wheel":
            c.zoomRatio >= 0.055 && t > 0.95 && t < 1.05 && (t = 1);
            break;
          case "pointermove":
          case "touchmove":
          case "mousemove":
            t > 0.99 && t < 1.01 && (t = 1);
            break;
        }
      var E = T * t, R = _ * t, M = E - A, V = R - S, Y = u.ratio;
      if (p(c.zoom) && d(s, fe, c.zoom, {
        once: !0
      }), D(s, fe, {
        ratio: t,
        oldRatio: Y,
        originalEvent: a
      }) === !1)
        return this;
      if (this.zooming = !0, a) {
        var tt = Ei(this.viewer), H = l && Object.keys(l).length > 0 ? Si(l) : {
          pageX: a.pageX,
          pageY: a.pageY
        };
        u.x -= M * ((H.pageX - tt.left - h) / A), u.y -= V * ((H.pageY - tt.top - m) / S);
      } else
        Z(r) && O(r.x) && O(r.y) ? (u.x -= M * ((r.x - h) / A), u.y -= V * ((r.y - m) / S)) : (u.x -= M / 2, u.y -= V / 2);
      u.left = u.x, u.top = u.y, u.width = E, u.height = R, u.oldRatio = Y, u.ratio = t, this.renderImage(function() {
        i.zooming = !1, p(c.zoomed) && d(s, de, c.zoomed, {
          once: !0
        }), D(s, de, {
          ratio: t,
          oldRatio: Y,
          originalEvent: a
        }, {
          cancelable: !1
        });
      }), n && this.tooltip();
    }
    return this;
  },
  /**
   * Play the images
   * @param {boolean|FullscreenOptions} [fullscreen=false] - Indicate if request fullscreen or not.
   * @returns {Viewer} this
   */
  play: function() {
    var t = this, i = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : !1;
    if (!this.isShown || this.played)
      return this;
    var n = this.element, r = this.options;
    if (p(r.play) && d(n, ve, r.play, {
      once: !0
    }), D(n, ve) === !1)
      return this;
    var a = this.player, o = this.loadImage.bind(this), s = [], c = 0, l = 0;
    if (this.played = !0, this.onLoadWhenPlay = o, i && this.requestFullscreen(i), f(a, q), y(this.items, function(m, A) {
      var S = m.querySelector("img"), T = document.createElement("img");
      T.src = $(S, "originalUrl"), T.alt = S.getAttribute("alt"), T.referrerPolicy = S.referrerPolicy, c += 1, f(T, vt), st(T, x, r.transition), U(m, lt) && (f(T, k), l = A), s.push(T), d(T, j, o, {
        once: !0
      }), a.appendChild(T);
    }), O(r.interval) && r.interval > 0) {
      var u = function() {
        clearTimeout(t.playing.timeout), g(s[l], k), l -= 1, l = l >= 0 ? l : c - 1, f(s[l], k), t.playing.timeout = setTimeout(u, r.interval);
      }, h = function() {
        clearTimeout(t.playing.timeout), g(s[l], k), l += 1, l = l < c ? l : 0, f(s[l], k), t.playing.timeout = setTimeout(h, r.interval);
      };
      c > 1 && (this.playing = {
        prev: u,
        next: h,
        timeout: setTimeout(h, r.interval)
      });
    }
    return this;
  },
  // Stop play
  stop: function() {
    var t = this;
    if (!this.played)
      return this;
    var i = this.element, n = this.options;
    if (p(n.stop) && d(i, me, n.stop, {
      once: !0
    }), D(i, me) === !1)
      return this;
    var r = this.player;
    return clearTimeout(this.playing.timeout), this.playing = !1, this.played = !1, y(r.getElementsByTagName("img"), function(a) {
      b(a, j, t.onLoadWhenPlay);
    }), g(r, q), r.innerHTML = "", this.exitFullscreen(), this;
  },
  // Enter modal mode (only available in inline mode)
  full: function() {
    var t = this, i = this.options, n = this.viewer, r = this.image, a = this.list;
    return !this.isShown || this.played || this.fulled || !i.inline ? this : (this.fulled = !0, this.open(), f(this.button, Ht), i.transition && (g(a, x), this.viewed && g(r, x)), f(n, It), n.setAttribute("role", "dialog"), n.setAttribute("aria-labelledby", this.title.id), n.setAttribute("aria-modal", !0), n.removeAttribute("style"), L(n, {
      zIndex: i.zIndex
    }), i.focus && this.enforceFocus(), this.initContainer(), this.viewerData = C({}, this.containerData), this.renderList(), this.viewed && this.initImage(function() {
      t.renderImage(function() {
        i.transition && setTimeout(function() {
          f(r, x), f(a, x);
        }, 0);
      });
    }), this);
  },
  // Exit modal mode (only available in inline mode)
  exit: function() {
    var t = this, i = this.options, n = this.viewer, r = this.image, a = this.list;
    return !this.isShown || this.played || !this.fulled || !i.inline ? this : (this.fulled = !1, this.close(), g(this.button, Ht), i.transition && (g(a, x), this.viewed && g(r, x)), i.focus && this.clearEnforceFocus(), n.removeAttribute("role"), n.removeAttribute("aria-labelledby"), n.removeAttribute("aria-modal"), g(n, It), L(n, {
      zIndex: i.zIndexInline
    }), this.viewerData = C({}, this.parentData), this.renderViewer(), this.renderList(), this.viewed && this.initImage(function() {
      t.renderImage(function() {
        i.transition && setTimeout(function() {
          f(r, x), f(a, x);
        }, 0);
      });
    }), this);
  },
  // Show the current ratio of the image with percentage
  tooltip: function() {
    var t = this, i = this.options, n = this.tooltipBox, r = this.imageData;
    return !this.viewed || this.played || !i.tooltip ? this : (n.textContent = "".concat(Math.round(r.ratio * 100), "%"), this.tooltipping ? clearTimeout(this.tooltipping) : i.transition ? (this.fading && D(n, z), f(n, q), f(n, vt), f(n, x), n.removeAttribute("aria-hidden"), n.initialOffsetWidth = n.offsetWidth, f(n, k)) : (f(n, q), n.removeAttribute("aria-hidden")), this.tooltipping = setTimeout(function() {
      i.transition ? (d(n, z, function() {
        g(n, q), g(n, vt), g(n, x), n.setAttribute("aria-hidden", !0), t.fading = !1;
      }, {
        once: !0
      }), g(n, k), t.fading = !0) : (g(n, q), n.setAttribute("aria-hidden", !0)), t.tooltipping = !1;
    }, 1e3), this);
  },
  /**
   * Toggle the image size between its current size and natural size
   * @param {Event} [_originalEvent=null] - The original event if any.
   * @returns {Viewer} this
   */
  toggle: function() {
    var t = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : null;
    return this.imageData.ratio === 1 ? this.zoomTo(this.imageData.oldRatio, !0, null, t) : this.zoomTo(1, !0, null, t), this;
  },
  // Reset the image to its initial state
  reset: function() {
    return this.viewed && !this.played && (this.imageData = C({}, this.initialImageData), this.renderImage()), this;
  },
  // Update viewer when images changed
  update: function() {
    var t = this, i = this.element, n = this.options, r = this.isImg;
    if (r && !i.parentNode)
      return this.destroy();
    var a = [];
    if (y(r ? [i] : i.querySelectorAll("img"), function(l) {
      p(n.filter) ? n.filter.call(t, l) && a.push(l) : t.getImageURL(l) && a.push(l);
    }), !a.length)
      return this;
    if (this.images = a, this.length = a.length, this.ready) {
      var o = [];
      if (y(this.items, function(l, u) {
        var h = l.querySelector("img"), m = a[u];
        m && h ? (m.src !== h.src || m.alt !== h.alt) && o.push(u) : o.push(u);
      }), L(this.list, {
        width: "auto"
      }), this.initList(), this.isShown)
        if (this.length) {
          if (this.viewed) {
            var s = o.indexOf(this.index);
            if (s >= 0)
              this.viewed = !1, this.view(Math.max(Math.min(this.index - s, this.length - 1), 0));
            else {
              var c = this.items[this.index];
              f(c, lt), c.setAttribute("aria-selected", !0);
            }
          }
        } else
          this.image = null, this.viewed = !1, this.index = 0, this.imageData = {}, this.canvas.innerHTML = "", this.title.innerHTML = "";
    } else
      this.build();
    return this;
  },
  // Destroy the viewer
  destroy: function() {
    var t = this.element, i = this.options;
    return t[v] ? (this.destroyed = !0, this.ready ? (this.played && this.stop(), i.inline ? (this.fulled && this.exit(), this.unbind()) : this.isShown ? (this.viewing && (this.imageRendering ? this.imageRendering.abort() : this.imageInitializing && this.imageInitializing.abort()), this.hiding && this.transitioning.abort(), this.hidden()) : this.showing && (this.transitioning.abort(), this.hidden()), this.ready = !1, this.viewer.parentNode.removeChild(this.viewer)) : i.inline && (this.delaying ? this.delaying.abort() : this.initializing && this.initializing.abort()), i.inline || b(t, Q, this.onStart), t[v] = void 0, this) : this;
  }
}, Ii = {
  getImageURL: function(t) {
    var i = this.options.url;
    return ot(i) ? i = t.getAttribute(i) : p(i) ? i = i.call(this, t) : i = "", i;
  },
  enforceFocus: function() {
    var t = this;
    this.clearEnforceFocus(), d(document, Bt, this.onFocusin = function(i) {
      var n = t.viewer, r = i.target;
      if (!(r === document || r === n || n.contains(r))) {
        for (; r; ) {
          if (r.getAttribute("tabindex") !== null || r.getAttribute("aria-modal") === "true")
            return;
          r = r.parentElement;
        }
        n.focus();
      }
    });
  },
  clearEnforceFocus: function() {
    this.onFocusin && (b(document, Bt, this.onFocusin), this.onFocusin = null);
  },
  open: function() {
    var t = this.body;
    f(t, qt), this.scrollbarWidth > 0 && (t.style.paddingRight = "".concat(this.scrollbarWidth + (parseFloat(this.initialBodyComputedPaddingRight) || 0), "px"));
  },
  close: function() {
    var t = this.body;
    g(t, qt), this.scrollbarWidth > 0 && (t.style.paddingRight = this.initialBodyPaddingRight);
  },
  shown: function() {
    var t = this.element, i = this.options, n = this.viewer;
    this.fulled = !0, this.isShown = !0, this.render(), this.bind(), this.showing = !1, i.focus && (n.focus(), this.enforceFocus()), p(i.shown) && d(t, ie, i.shown, {
      once: !0
    }), D(t, ie) !== !1 && this.ready && this.isShown && !this.hiding && this.view(this.index);
  },
  hidden: function() {
    var t = this.element, i = this.options, n = this.viewer;
    i.fucus && this.clearEnforceFocus(), this.close(), this.unbind(), f(n, W), n.removeAttribute("role"), n.removeAttribute("aria-labelledby"), n.removeAttribute("aria-modal"), n.setAttribute("aria-hidden", !0), this.resetList(), this.resetImage(), this.fulled = !1, this.viewed = !1, this.isShown = !1, this.hiding = !1, this.destroyed || (p(i.hidden) && d(t, re, i.hidden, {
      once: !0
    }), D(t, re, null, {
      cancelable: !1
    }));
  },
  requestFullscreen: function(t) {
    var i = this.element.ownerDocument;
    if (this.fulled && !(i.fullscreenElement || i.webkitFullscreenElement || i.mozFullScreenElement || i.msFullscreenElement)) {
      var n = i.documentElement;
      n.requestFullscreen ? Z(t) ? n.requestFullscreen(t) : n.requestFullscreen() : n.webkitRequestFullscreen ? n.webkitRequestFullscreen(Element.ALLOW_KEYBOARD_INPUT) : n.mozRequestFullScreen ? n.mozRequestFullScreen() : n.msRequestFullscreen && n.msRequestFullscreen();
    }
  },
  exitFullscreen: function() {
    var t = this.element.ownerDocument;
    this.fulled && (t.fullscreenElement || t.webkitFullscreenElement || t.mozFullScreenElement || t.msFullscreenElement) && (t.exitFullscreen ? t.exitFullscreen() : t.webkitExitFullscreen ? t.webkitExitFullscreen() : t.mozCancelFullScreen ? t.mozCancelFullScreen() : t.msExitFullscreen && t.msExitFullscreen());
  },
  change: function(t) {
    var i = this.options, n = this.pointers, r = n[Object.keys(n)[0]];
    if (r) {
      var a = r.endX - r.startX, o = r.endY - r.startY;
      switch (this.action) {
        case dt:
          (a !== 0 || o !== 0) && (this.pointerMoved = !0, this.move(a, o, t));
          break;
        case it:
          this.zoom(Oi(n), !1, null, t);
          break;
        case Ae: {
          this.action = "switched";
          var s = Math.abs(a);
          s > 1 && s > Math.abs(o) && (this.pointers = {}, a > 1 ? this.prev(i.loop) : a < -1 && this.next(i.loop));
          break;
        }
      }
      y(n, function(c) {
        c.startX = c.endX, c.startY = c.endY;
      });
    }
  },
  isSwitchable: function() {
    var t = this.imageData, i = this.viewerData;
    return this.length > 1 && t.x >= 0 && t.y >= 0 && t.width <= i.width && t.height <= i.height;
  }
}, Ni = F.Viewer, Ci = function(e) {
  return function() {
    return e += 1, e;
  };
}(-1), Ie = /* @__PURE__ */ function() {
  function e(t) {
    var i = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {};
    if (ii(this, e), !t || t.nodeType !== 1)
      throw new Error("The first argument is required and must be an element.");
    this.element = t, this.options = C({}, Xt, Z(i) && i), this.action = !1, this.fading = !1, this.fulled = !1, this.hiding = !1, this.imageClicked = !1, this.imageData = {}, this.index = this.options.initialViewIndex, this.isImg = !1, this.isShown = !1, this.length = 0, this.moving = !1, this.played = !1, this.playing = !1, this.pointers = {}, this.ready = !1, this.rotating = !1, this.scaling = !1, this.showing = !1, this.timeout = !1, this.tooltipping = !1, this.viewed = !1, this.viewing = !1, this.wheeling = !1, this.zooming = !1, this.pointerMoved = !1, this.id = Ci(), this.init();
  }
  return ni(e, [{
    key: "init",
    value: function() {
      var i = this, n = this.element, r = this.options;
      if (!n[v]) {
        n[v] = this, r.focus && !r.keyboard && (r.focus = !1);
        var a = n.localName === "img", o = [];
        if (y(a ? [n] : n.querySelectorAll("img"), function(l) {
          p(r.filter) ? r.filter.call(i, l) && o.push(l) : i.getImageURL(l) && o.push(l);
        }), this.isImg = a, this.length = o.length, this.images = o, this.initBody(), B(document.createElement(v).style.transition) && (r.transition = !1), r.inline) {
          var s = 0, c = function() {
            if (s += 1, s === i.length) {
              var u;
              i.initializing = !1, i.delaying = {
                abort: function() {
                  clearTimeout(u);
                }
              }, u = setTimeout(function() {
                i.delaying = !1, i.build();
              }, 0);
            }
          };
          this.initializing = {
            abort: function() {
              y(o, function(u) {
                u.complete || (b(u, j, c), b(u, X, c));
              });
            }
          }, y(o, function(l) {
            if (l.complete)
              c();
            else {
              var u, h;
              d(l, j, u = function() {
                b(l, X, h), c();
              }, {
                once: !0
              }), d(l, X, h = function() {
                b(l, j, u), c();
              }, {
                once: !0
              });
            }
          });
        } else
          d(n, Q, this.onStart = function(l) {
            var u = l.target;
            u.localName === "img" && (!p(r.filter) || r.filter.call(i, u)) && i.view(i.images.indexOf(u));
          });
      }
    }
  }, {
    key: "build",
    value: function() {
      if (!this.ready) {
        var i = this.element, n = this.options, r = i.parentNode, a = document.createElement("div");
        a.innerHTML = oi;
        var o = a.querySelector(".".concat(v, "-container")), s = o.querySelector(".".concat(v, "-title")), c = o.querySelector(".".concat(v, "-toolbar")), l = o.querySelector(".".concat(v, "-navbar")), u = o.querySelector(".".concat(v, "-button")), h = o.querySelector(".".concat(v, "-canvas"));
        if (this.parent = r, this.viewer = o, this.title = s, this.toolbar = c, this.navbar = l, this.button = u, this.canvas = h, this.footer = o.querySelector(".".concat(v, "-footer")), this.tooltipBox = o.querySelector(".".concat(v, "-tooltip")), this.player = o.querySelector(".".concat(v, "-player")), this.list = o.querySelector(".".concat(v, "-list")), o.id = "".concat(v).concat(this.id), s.id = "".concat(v, "Title").concat(this.id), f(s, n.title ? ct(Array.isArray(n.title) ? n.title[0] : n.title) : W), f(l, n.navbar ? ct(n.navbar) : W), st(u, W, !n.button), n.keyboard && u.setAttribute("tabindex", 0), n.backdrop && (f(o, "".concat(v, "-backdrop")), !n.inline && n.backdrop !== "static" && Ct(h, gt, "hide")), ot(n.className) && n.className && n.className.split(zt).forEach(function(E) {
          f(o, E);
        }), n.toolbar) {
          var m = document.createElement("ul"), A = Z(n.toolbar), S = ut.slice(0, 3), T = ut.slice(7, 9), _ = ut.slice(9);
          A || f(c, ct(n.toolbar)), y(A ? n.toolbar : ut, function(E, R) {
            var M = A && Z(E), V = A ? Lt(R) : E, Y = M && !B(E.show) ? E.show : E;
            if (!(!Y || !n.zoomable && S.indexOf(V) !== -1 || !n.rotatable && T.indexOf(V) !== -1 || !n.scalable && _.indexOf(V) !== -1)) {
              var tt = M && !B(E.size) ? E.size : E, H = M && !B(E.click) ? E.click : E, P = document.createElement("li");
              n.keyboard && P.setAttribute("tabindex", 0), P.setAttribute("role", "button"), f(P, "".concat(v, "-").concat(V)), p(H) || Ct(P, gt, V), O(Y) && f(P, ct(Y)), ["small", "large"].indexOf(tt) !== -1 ? f(P, "".concat(v, "-").concat(tt)) : V === "play" && f(P, "".concat(v, "-large")), p(H) && d(P, Q, H), m.appendChild(P);
            }
          }), c.appendChild(m);
        } else
          f(c, W);
        if (!n.rotatable) {
          var I = c.querySelectorAll('li[class*="rotate"]');
          f(I, nt), y(I, function(E) {
            c.appendChild(E);
          });
        }
        if (n.inline)
          f(u, li), L(o, {
            zIndex: n.zIndexInline
          }), window.getComputedStyle(r).position === "static" && L(r, {
            position: "relative"
          }), r.insertBefore(o, i.nextSibling);
        else {
          f(u, si), f(o, It), f(o, vt), f(o, W), L(o, {
            zIndex: n.zIndex
          });
          var N = n.container;
          ot(N) && (N = i.ownerDocument.querySelector(N)), N || (N = this.body), N.appendChild(o);
        }
        if (n.inline && (this.render(), this.bind(), this.isShown = !0), this.ready = !0, p(n.ready) && d(i, te, n.ready, {
          once: !0
        }), D(i, te) === !1) {
          this.ready = !1;
          return;
        }
        this.ready && n.inline && this.view(this.index);
      }
    }
    /**
     * Get the no conflict viewer class.
     * @returns {Viewer} The viewer class.
     */
  }], [{
    key: "noConflict",
    value: function() {
      return window.Viewer = Ni, e;
    }
    /**
     * Change the default options.
     * @param {Object} options - The new default options.
     */
  }, {
    key: "setDefaults",
    value: function(i) {
      C(Xt, Z(i) && i);
    }
  }]);
}();
C(Ie.prototype, xi, Ai, Di, _i, Ii);
var ki = typeof global == "object" && global && global.Object === Object && global;
const Ne = ki;
var ji = typeof self == "object" && self && self.Object === Object && self, zi = Ne || ji || Function("return this")();
const Vt = zi;
var Li = Vt.Symbol;
const pt = Li;
var Ce = Object.prototype, Vi = Ce.hasOwnProperty, Ri = Ce.toString, et = pt ? pt.toStringTag : void 0;
function Pi(e) {
  var t = Vi.call(e, et), i = e[et];
  try {
    e[et] = void 0;
    var n = !0;
  } catch (a) {
  }
  var r = Ri.call(e);
  return n && (t ? e[et] = i : delete e[et]), r;
}
var Fi = Object.prototype, Mi = Fi.toString;
function Yi(e) {
  return Mi.call(e);
}
var Wi = "[object Null]", Xi = "[object Undefined]", ge = pt ? pt.toStringTag : void 0;
function Rt(e) {
  return e == null ? e === void 0 ? Xi : Wi : ge && ge in Object(e) ? Pi(e) : Yi(e);
}
function Pt(e) {
  return e != null && typeof e == "object";
}
var Hi = Array.isArray;
const qi = Hi;
function wt(e) {
  var t = typeof e;
  return e != null && (t == "object" || t == "function");
}
function ke(e) {
  return e;
}
var Ui = "[object AsyncFunction]", Bi = "[object Function]", Zi = "[object GeneratorFunction]", $i = "[object Proxy]";
function je(e) {
  if (!wt(e))
    return !1;
  var t = Rt(e);
  return t == Bi || t == Zi || t == Ui || t == $i;
}
var Ki = Vt["__core-js_shared__"];
const xt = Ki;
var pe = function() {
  var e = /[^.]+$/.exec(xt && xt.keys && xt.keys.IE_PROTO || "");
  return e ? "Symbol(src)_1." + e : "";
}();
function Gi(e) {
  return !!pe && pe in e;
}
var Qi = Function.prototype, Ji = Qi.toString;
function tn(e) {
  if (e != null) {
    try {
      return Ji.call(e);
    } catch (t) {
    }
    try {
      return e + "";
    } catch (t) {
    }
  }
  return "";
}
var en = /[\\^$.*+?()[\]{}|]/g, nn = /^\[object .+?Constructor\]$/, rn = Function.prototype, an = Object.prototype, on = rn.toString, sn = an.hasOwnProperty, ln = RegExp(
  "^" + on.call(sn).replace(en, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function un(e) {
  if (!wt(e) || Gi(e))
    return !1;
  var t = je(e) ? ln : nn;
  return t.test(tn(e));
}
function cn(e, t) {
  return e == null ? void 0 : e[t];
}
function hn(e, t) {
  var i = cn(e, t);
  return un(i) ? i : void 0;
}
function fn(e, t, i) {
  switch (i.length) {
    case 0:
      return e.call(t);
    case 1:
      return e.call(t, i[0]);
    case 2:
      return e.call(t, i[0], i[1]);
    case 3:
      return e.call(t, i[0], i[1], i[2]);
  }
  return e.apply(t, i);
}
var dn = 800, vn = 16, mn = Date.now;
function gn(e) {
  var t = 0, i = 0;
  return function() {
    var n = mn(), r = vn - (n - i);
    if (i = n, r > 0) {
      if (++t >= dn)
        return arguments[0];
    } else
      t = 0;
    return e.apply(void 0, arguments);
  };
}
function pn(e) {
  return function() {
    return e;
  };
}
var bn = function() {
  try {
    var e = hn(Object, "defineProperty");
    return e({}, "", {}), e;
  } catch (t) {
  }
}();
const bt = bn;
var yn = bt ? function(e, t) {
  return bt(e, "toString", {
    configurable: !0,
    enumerable: !1,
    value: pn(t),
    writable: !0
  });
} : ke;
const wn = yn;
var En = gn(wn);
const Tn = En;
var On = 9007199254740991, Sn = /^(?:0|[1-9]\d*)$/;
function ze(e, t) {
  var i = typeof e;
  return t = t == null ? On : t, !!t && (i == "number" || i != "symbol" && Sn.test(e)) && e > -1 && e % 1 == 0 && e < t;
}
function Le(e, t, i) {
  t == "__proto__" && bt ? bt(e, t, {
    configurable: !0,
    enumerable: !0,
    value: i,
    writable: !0
  }) : e[t] = i;
}
function Ft(e, t) {
  return e === t || e !== e && t !== t;
}
var xn = Object.prototype, An = xn.hasOwnProperty;
function Ve(e, t, i) {
  var n = e[t];
  (!(An.call(e, t) && Ft(n, i)) || i === void 0 && !(t in e)) && Le(e, t, i);
}
function Dn(e, t, i, n) {
  var r = !i;
  i || (i = {});
  for (var a = -1, o = t.length; ++a < o; ) {
    var s = t[a], c = n ? n(i[s], e[s], s, i, e) : void 0;
    c === void 0 && (c = e[s]), r ? Le(i, s, c) : Ve(i, s, c);
  }
  return i;
}
var be = Math.max;
function _n(e, t, i) {
  return t = be(t === void 0 ? e.length - 1 : t, 0), function() {
    for (var n = arguments, r = -1, a = be(n.length - t, 0), o = Array(a); ++r < a; )
      o[r] = n[t + r];
    r = -1;
    for (var s = Array(t + 1); ++r < t; )
      s[r] = n[r];
    return s[t] = i(o), fn(e, this, s);
  };
}
function Re(e, t) {
  return Tn(_n(e, t, ke), e + "");
}
var In = 9007199254740991;
function Pe(e) {
  return typeof e == "number" && e > -1 && e % 1 == 0 && e <= In;
}
function Et(e) {
  return e != null && Pe(e.length) && !je(e);
}
function Fe(e, t, i) {
  if (!wt(i))
    return !1;
  var n = typeof t;
  return (n == "number" ? Et(i) && ze(t, i.length) : n == "string" && t in i) ? Ft(i[t], e) : !1;
}
function Nn(e) {
  return Re(function(t, i) {
    var n = -1, r = i.length, a = r > 1 ? i[r - 1] : void 0, o = r > 2 ? i[2] : void 0;
    for (a = e.length > 3 && typeof a == "function" ? (r--, a) : void 0, o && Fe(i[0], i[1], o) && (a = r < 3 ? void 0 : a, r = 1), t = Object(t); ++n < r; ) {
      var s = i[n];
      s && e(t, s, n, a);
    }
    return t;
  });
}
var Cn = Object.prototype;
function Mt(e) {
  var t = e && e.constructor, i = typeof t == "function" && t.prototype || Cn;
  return e === i;
}
function kn(e, t) {
  for (var i = -1, n = Array(e); ++i < e; )
    n[i] = t(i);
  return n;
}
var jn = "[object Arguments]";
function ye(e) {
  return Pt(e) && Rt(e) == jn;
}
var Me = Object.prototype, zn = Me.hasOwnProperty, Ln = Me.propertyIsEnumerable, Vn = ye(function() {
  return arguments;
}()) ? ye : function(e) {
  return Pt(e) && zn.call(e, "callee") && !Ln.call(e, "callee");
};
const Rn = Vn;
function Pn() {
  return !1;
}
var Ye = typeof exports == "object" && exports && !exports.nodeType && exports, we = Ye && typeof module == "object" && module && !module.nodeType && module, Fn = we && we.exports === Ye, Ee = Fn ? Vt.Buffer : void 0, Mn = Ee ? Ee.isBuffer : void 0, Yn = Mn || Pn;
const Wn = Yn;
var Xn = "[object Arguments]", Hn = "[object Array]", qn = "[object Boolean]", Un = "[object Date]", Bn = "[object Error]", Zn = "[object Function]", $n = "[object Map]", Kn = "[object Number]", Gn = "[object Object]", Qn = "[object RegExp]", Jn = "[object Set]", tr = "[object String]", er = "[object WeakMap]", ir = "[object ArrayBuffer]", nr = "[object DataView]", rr = "[object Float32Array]", ar = "[object Float64Array]", or = "[object Int8Array]", sr = "[object Int16Array]", lr = "[object Int32Array]", ur = "[object Uint8Array]", cr = "[object Uint8ClampedArray]", hr = "[object Uint16Array]", fr = "[object Uint32Array]", w = {};
w[rr] = w[ar] = w[or] = w[sr] = w[lr] = w[ur] = w[cr] = w[hr] = w[fr] = !0;
w[Xn] = w[Hn] = w[ir] = w[qn] = w[nr] = w[Un] = w[Bn] = w[Zn] = w[$n] = w[Kn] = w[Gn] = w[Qn] = w[Jn] = w[tr] = w[er] = !1;
function dr(e) {
  return Pt(e) && Pe(e.length) && !!w[Rt(e)];
}
function vr(e) {
  return function(t) {
    return e(t);
  };
}
var We = typeof exports == "object" && exports && !exports.nodeType && exports, rt = We && typeof module == "object" && module && !module.nodeType && module, mr = rt && rt.exports === We, At = mr && Ne.process, gr = function() {
  try {
    var e = rt && rt.require && rt.require("util").types;
    return e || At && At.binding && At.binding("util");
  } catch (t) {
  }
}();
const Te = gr;
var Oe = Te && Te.isTypedArray, pr = Oe ? vr(Oe) : dr;
const br = pr;
var yr = Object.prototype, wr = yr.hasOwnProperty;
function Xe(e, t) {
  var i = qi(e), n = !i && Rn(e), r = !i && !n && Wn(e), a = !i && !n && !r && br(e), o = i || n || r || a, s = o ? kn(e.length, String) : [], c = s.length;
  for (var l in e)
    (t || wr.call(e, l)) && !(o && (l == "length" || r && (l == "offset" || l == "parent") || a && (l == "buffer" || l == "byteLength" || l == "byteOffset") || ze(l, c))) && s.push(l);
  return s;
}
function Er(e, t) {
  return function(i) {
    return e(t(i));
  };
}
var Tr = Er(Object.keys, Object);
const Or = Tr;
var Sr = Object.prototype, xr = Sr.hasOwnProperty;
function Ar(e) {
  if (!Mt(e))
    return Or(e);
  var t = [];
  for (var i in Object(e))
    xr.call(e, i) && i != "constructor" && t.push(i);
  return t;
}
function Dr(e) {
  return Et(e) ? Xe(e) : Ar(e);
}
var _r = Object.prototype, Ir = _r.hasOwnProperty;
Nn(function(e, t) {
  if (Mt(t) || Et(t)) {
    Dn(t, Dr(t), e);
    return;
  }
  for (var i in t)
    Ir.call(t, i) && Ve(e, i, t[i]);
});
function Nr(e) {
  var t = [];
  if (e != null)
    for (var i in Object(e))
      t.push(i);
  return t;
}
var Cr = Object.prototype, kr = Cr.hasOwnProperty;
function jr(e) {
  if (!wt(e))
    return Nr(e);
  var t = Mt(e), i = [];
  for (var n in e)
    n == "constructor" && (t || !kr.call(e, n)) || i.push(n);
  return i;
}
function zr(e) {
  return Et(e) ? Xe(e, !0) : jr(e);
}
var He = Object.prototype, Lr = He.hasOwnProperty;
Re(function(e, t) {
  e = Object(e);
  var i = -1, n = t.length, r = n > 2 ? t[2] : void 0;
  for (r && Fe(t[0], t[1], r) && (n = 1); ++i < n; )
    for (var a = t[i], o = zr(a), s = -1, c = o.length; ++s < c; ) {
      var l = o[s], u = e[l];
      (u === void 0 || Ft(u, He[l]) && !Lr.call(e, l)) && (e[l] = a[l]);
    }
  return e;
});
const Vr = Se({
  name: "Viewer",
  props: {
    images: {
      type: Array,
      default: () => []
    },
    rebuild: {
      type: Boolean,
      default: !1
    },
    trigger: {
      type: Object,
      default: null
    },
    options: {
      type: Object,
      default: () => null
    }
  },
  emits: ["inited"],
  setup(e, { emit: t }) {
    let i;
    const n = qe();
    function r() {
      i = new Ie(n.value, e.options), t("inited", i);
    }
    function a() {
      i && i.destroy();
    }
    function o() {
      a(), r();
    }
    function s() {
      i ? (i.update(), t("inited", i)) : r();
    }
    function c() {
      e.rebuild ? o() : s();
    }
    const l = { deep: !0 };
    return Tt(() => e.images, () => {
      Ot(() => {
        c();
      });
    }, l), Tt(() => e.trigger, () => {
      Ot(() => {
        c();
      });
    }, l), Tt(() => e.options, () => {
      Ot(() => {
        o();
      });
    }, l), Ue(() => r()), Be(() => a()), {
      root: n,
      createViewer: r,
      rebuildViewer: o,
      updateViewer: s,
      destroyViewer: a
    };
  }
}), Rr = (e, t) => {
  const i = e.__vccOpts || e;
  for (const [n, r] of t)
    i[n] = r;
  return i;
}, Pr = { ref: "root" };
function Fr(e, t, i, n, r, a) {
  return ft(), Dt("div", Pr, [
    Ze(e.$slots, "default", {
      images: e.images,
      options: e.options
    })
  ], 512);
}
const Mr = /* @__PURE__ */ Rr(Vr, [["render", Fr]]);
const Yr = ["src"], Wr = /* @__PURE__ */ Se({
  __name: "ImageViewer",
  props: {
    image: {}
  },
  setup(e) {
    const t = e, i = $e(() => t.image ? [{ src: t.image, index: 0 }] : []);
    return (n, r) => (ft(), Ke(Ge(Mr), {
      images: i.value,
      style: { height: "100%" }
    }, {
      default: Qe(() => [
        (ft(!0), Dt(Je, null, ti(i.value, (a) => (ft(), Dt("img", {
          alt: "图片",
          src: a.src,
          key: a.index,
          class: "image"
        }, null, 8, Yr))), 128))
      ]),
      _: 1
    }, 8, ["images"]));
  }
});
const qr = /* @__PURE__ */ ei(Wr, [["__scopeId", "data-v-b0f61c51"]]);
export {
  qr as default
};
