// @flyfish-group/file-viewer3@1.0.5/dist/components/index.js
var Xn = Object.defineProperty, Qn = Object.defineProperties;
var Yn = Object.getOwnPropertyDescriptors;
var Dt = Object.getOwnPropertySymbols;
var Zn = Object.prototype.hasOwnProperty, er = Object.prototype.propertyIsEnumerable;
var xe = (e, t) => {
  if (t = Symbol[e])
    return t;
  throw Error("Symbol." + e + " is not defined");
};
var nt = (e, t, r) => t in e ? Xn(e, t, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[t] = r, J = (e, t) => {
  for (var r in t || (t = {}))
    Zn.call(t, r) && nt(e, r, t[r]);
  if (Dt)
    for (var r of Dt(t))
      er.call(t, r) && nt(e, r, t[r]);
  return e;
}, oe = (e, t) => Qn(e, Yn(t));
var ye = (e, t, r) => (nt(e, typeof t != "symbol" ? t + "" : t, r), r);
var B = (e, t, r) => new Promise((n, s) => {
  var o = (l) => {
    try {
      i(r.next(l));
    } catch (c) {
      s(c);
    }
  }, a = (l) => {
    try {
      i(r.throw(l));
    } catch (c) {
      s(c);
    }
  }, i = (l) => l.done ? n(l.value) : Promise.resolve(l.value).then(o, a);
  i((r = r.apply(e, t)).next());
}), ae = function(e, t) {
  this[0] = e, this[1] = t;
}, rt = (e, t, r) => {
  var n = (a, i, l, c) => {
    try {
      var u = r[a](i), f = (i = u.value) instanceof ae, h = u.done;
      Promise.resolve(f ? i[0] : i).then((b) => f ? n(a === "return" ? a : "next", i[1] ? { done: b.done, value: b.value } : b, l, c) : l({ value: b, done: h })).catch((b) => n("throw", b, l, c));
    } catch (b) {
      c(b);
    }
  }, s = (a) => o[a] = (i) => new Promise((l, c) => n(a, i, l, c)), o = {};
  return r = r.apply(e, t), o[Symbol.asyncIterator] = () => o, s("next"), s("throw"), s("return"), o;
}, st = (e) => {
  var t = e[xe("asyncIterator")], r = !1, n, s = {};
  return t == null ? (t = e[xe("iterator")](), n = (o) => s[o] = (a) => t[o](a)) : (t = t.call(e), n = (o) => s[o] = (a) => {
    if (r) {
      if (r = !1, o === "throw")
        throw a;
      return a;
    }
    return r = !0, {
      done: !1,
      value: new ae(new Promise((i) => {
        var l = t[o](a);
        if (!(l instanceof Object))
          throw TypeError("Object expected");
        i(l);
      }), 1)
    };
  }), s[xe("iterator")] = () => s, n("next"), "throw" in t ? n("throw") : s.throw = (o) => {
    throw o;
  }, "return" in t && n("return"), s;
}, Lt = (e, t, r) => (t = e[xe("asyncIterator")]) ? t.call(e) : (e = e[xe("iterator")](), t = {}, r = (n, s) => (s = e[n]) && (t[n] = (o) => new Promise((a, i, l) => (o = s.call(e, o), l = o.done, Promise.resolve(o.value).then((c) => a({ value: c, done: l }), i)))), r("next"), r("return"), t);
import { defineAsyncComponent as Be, createApp as he, createVNode as me, defineComponent as ln, openBlock as Oe, createElementBlock as Ie, toDisplayString as De, ref as we, onMounted as tr, watch as _t, createElementVNode as ot, Fragment as nr, withDirectives as Mt, vShow as Nt } from "vue";
const rr = (() => {
  const e = {
    module: null,
    load() {
      return B(this, null, function* () {
        return this.module || (this.module = import("./docx-preview.js")), this.module;
      });
    }
  };
  return () => B(void 0, null, function* () {
    return yield e.load();
  });
})();
function sr(e, t) {
  return B(this, null, function* () {
    const { defaultOptions: r, renderAsync: n } = yield rr(), s = Object.assign(r, {
      debug: !0,
      experimental: !0
    });
    yield n(e, t, void 0, s);
  });
}
const Ut = globalThis.Buffer;
let $t = 0;
function te(e) {
  return String(e).replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;").replace(/"/g, "&quot;").replace(/'/g, "&#39;");
}
function G(e = "id") {
  return $t += 1, `${e}-${$t}`;
}
function Pe(e) {
  return e % 2 === 0 ? e : e + 1;
}
function or(e) {
  const t = e instanceof Uint8Array ? e : e instanceof ArrayBuffer ? new Uint8Array(e) : new Uint8Array(e.buffer, e.byteOffset, e.byteLength);
  if (Ut)
    return Ut.from(t).toString("base64");
  let r = "";
  const n = 32768;
  for (let s = 0; s < t.length; s += n)
    r += String.fromCharCode(...t.subarray(s, Math.min(s + n, t.length)));
  return btoa(r);
}
function He(e, t = "application/octet-stream") {
  return `data:${t};base64,${or(e)}`;
}
function un(e) {
  return String(e != null ? e : "").toLowerCase().replace(/[^a-z0-9\u4e00-\u9fa5]+/g, "-").replace(/^-+|-+$/g, "") || "item";
}
function fn(e, t) {
  if (e === t)
    return !0;
  if (!e || !t)
    return !1;
  const r = Object.keys(e), n = Object.keys(t);
  if (r.length !== n.length)
    return !1;
  for (const s of r)
    if (e[s] !== t[s])
      return !1;
  return !0;
}
function Le(e, t, r) {
  const n = r ? J({ message: t }, r) : { message: t };
  return e.push(n), n;
}
function ne(e, t = 96) {
  if (!(e == null || Number.isNaN(e)))
    return e / 1440 * t;
}
function ar(e) {
  return String(e != null ? e : "").replace(/\u000b/g, `
`).replace(/\u000c/g, `
`);
}
function mt(e = "utf-8") {
  return new TextDecoder(e, { fatal: !1 });
}
const Ht = mt("utf-16le"), ir = mt(typeof TextDecoder != "undefined" ? "windows-1252" : "latin1");
class k {
  constructor(t) {
    ye(this, "bytes");
    ye(this, "view");
    ye(this, "length");
    if (t instanceof Uint8Array)
      this.bytes = t;
    else if (t instanceof ArrayBuffer)
      this.bytes = new Uint8Array(t);
    else if (ArrayBuffer.isView(t))
      this.bytes = new Uint8Array(t.buffer, t.byteOffset, t.byteLength);
    else
      throw new TypeError("BinaryReader expects ArrayBuffer or Uint8Array");
    this.view = new DataView(this.bytes.buffer, this.bytes.byteOffset, this.bytes.byteLength), this.length = this.bytes.byteLength;
  }
  ensure(t, r = 1) {
    return t >= 0 && t + r <= this.length;
  }
  u8(t) {
    return this.ensure(t, 1) ? this.view.getUint8(t) : 0;
  }
  i8(t) {
    return this.ensure(t, 1) ? this.view.getInt8(t) : 0;
  }
  u16(t) {
    return this.ensure(t, 2) ? this.view.getUint16(t, !0) : 0;
  }
  i16(t) {
    return this.ensure(t, 2) ? this.view.getInt16(t, !0) : 0;
  }
  u32(t) {
    return this.ensure(t, 4) ? this.view.getUint32(t, !0) : 0;
  }
  i32(t) {
    return this.ensure(t, 4) ? this.view.getInt32(t, !0) : 0;
  }
  u64(t) {
    const r = this.u32(t);
    return this.u32(t + 4) * 4294967296 + r;
  }
  slice(t, r) {
    return this.ensure(t, r) ? this.bytes.subarray(t, t + r) : new Uint8Array(0);
  }
  array(t, r, n) {
    var a;
    const s = [], o = (a = n.byteWidth) != null ? a : 1;
    for (let i = 0; i < r; i += 1)
      s.push(n.call(this, t + i * o));
    return s;
  }
  utf16le(t, r) {
    return this.ensure(t, r) ? Ht.decode(this.slice(t, r)) : "";
  }
  latin1(t, r) {
    return this.ensure(t, r) ? ir.decode(this.slice(t, r)) : "";
  }
  ascii(t, r) {
    var o;
    if (!this.ensure(t, r))
      return "";
    let n = "";
    const s = this.slice(t, r);
    for (let a = 0; a < s.length; a += 1)
      n += String.fromCharCode((o = s[a]) != null ? o : 0);
    return n;
  }
  utf16z(t, r) {
    const n = this.slice(t, r);
    let s = n.length;
    for (let o = 0; o + 1 < n.length; o += 2)
      if (n[o] === 0 && n[o + 1] === 0) {
        s = o;
        break;
      }
    return Ht.decode(n.subarray(0, s));
  }
}
k.prototype.u8.byteWidth = 1;
k.prototype.i8.byteWidth = 1;
k.prototype.u16.byteWidth = 2;
k.prototype.i16.byteWidth = 2;
k.prototype.u32.byteWidth = 4;
k.prototype.i32.byteWidth = 4;
function cr(e) {
  if (e instanceof Uint8Array)
    return e;
  if (e instanceof ArrayBuffer)
    return new Uint8Array(e);
  if (ArrayBuffer.isView(e))
    return new Uint8Array(e.buffer, e.byteOffset, e.byteLength);
  throw new TypeError("Unsupported binary input");
}
const jt = [208, 207, 17, 224, 161, 177, 26, 225], Je = 4294967295, pe = 4294967294, lr = 4294967293, ur = 4294967292, fr = 1 << 20, dr = 4096;
function Wt(e) {
  return e === Je || e === pe || e === lr || e === ur;
}
function gt(e, t, r) {
  const n = 512 + r * t, s = n + t;
  if (n < 0 || s > e.length)
    throw new Error(`Sector ${r} is out of bounds`);
  return e.subarray(n, s);
}
function dn(e, t, r = fr) {
  const n = [], s = /* @__PURE__ */ new Set();
  let o = e, a = 0;
  for (; o !== pe && o !== Je && o >= 0; ) {
    if (s.has(o))
      throw new Error(`Detected sector loop at ${o}`);
    if (s.add(o), n.push(o), o = t(o), a += 1, a > r)
      throw new Error("Sector chain exceeds safe limit");
  }
  return n;
}
function Ae(e, t, r, n, s = null) {
  if (r === pe || r === Je || r < 0)
    return new Uint8Array(0);
  const o = dn(r, (l) => {
    var c;
    return (c = n[l]) != null ? c : pe;
  }), a = new Uint8Array(o.length * t);
  let i = 0;
  for (const l of o)
    a.set(gt(e, t, l), i), i += t;
  return s == null ? a : a.subarray(0, Math.min(s, a.length));
}
function pr(e) {
  function t(s, o) {
    if (s < 0 || s >= e.length)
      return;
    const a = e[s];
    a && (t(a.leftSiblingId, o), o.push(s), t(a.rightSiblingId, o));
  }
  function r(s) {
    const o = e[s];
    if (!o || o.childId < 0)
      return;
    const a = [];
    t(o.childId, a), o.children = a;
    for (const i of a) {
      const l = e[i];
      l && (l.parentId = s, r(i));
    }
  }
  const n = e.find((s) => s.objectType === 5);
  if (!n)
    throw new Error("CFB root storage not found");
  return r(n.id), n;
}
function hr(e) {
  var r;
  const t = /* @__PURE__ */ new Map();
  for (const n of e) {
    if (!n || !n.name)
      continue;
    const s = [];
    let o = n;
    for (; o; )
      o.objectType !== 5 && s.push(o.name), o = o.parentId != null && (r = e[o.parentId]) != null ? r : null;
    const a = `/${s.reverse().join("/")}`;
    n.path = a === "/" ? `/${n.name}` : a, t.set(n.path, n);
  }
  return t;
}
function mr(e) {
  return e.replace(/\u0000+$/, "");
}
function gr(e, t = {}) {
  const r = cr(e), n = new k(r), s = [];
  for (let w = 0; w < jt.length; w += 1)
    if (n.u8(w) !== jt[w])
      throw new Error("Not a Compound File Binary document");
  const o = n.u16(26), a = n.u16(30), i = n.u16(32), l = 1 << a, c = 1 << i, u = n.u32(40), f = n.u32(44), h = n.i32(48), b = n.u32(52), m = n.u32(56), y = n.i32(60), x = n.u32(64), T = n.i32(68), S = n.u32(72);
  m !== dr && Le(s, `Unexpected mini stream cutoff size ${m}`);
  const E = [];
  for (let w = 0; w < 109; w += 1) {
    const R = n.i32(76 + w * 4);
    !Wt(R) && R >= 0 && E.push(R);
  }
  const O = [...E];
  let A = T, D = 0;
  for (; A !== pe && A !== Je && A >= 0; ) {
    const w = gt(r, l, A), R = new k(w), L = l / 4 - 1;
    for (let M = 0; M < L; M += 1) {
      const V = R.i32(M * 4);
      !Wt(V) && V >= 0 && O.push(V);
    }
    if (A = R.i32(l - 4), D += 1, D > S + 4) {
      Le(s, "DIFAT chain exceeded declared sector count; stopping early");
      break;
    }
  }
  f && O.length < f && Le(s, `FAT sector count mismatch: header says ${f}, found ${O.length}`);
  const I = [];
  for (const w of O) {
    const R = gt(r, l, w), L = new k(R);
    for (let M = 0; M < l / 4; M += 1)
      I.push(L.i32(M * 4));
  }
  const U = Ae(r, l, h, I), v = new k(U), P = [];
  for (let w = 0, R = 0; w + 128 <= U.length; w += 128, R += 1) {
    const L = v.u16(w + 64), M = mr(v.utf16le(w, Math.max(0, L - 2)));
    !M && v.u8(w + 66) === 0 || P.push({
      id: R,
      name: M,
      objectType: v.u8(w + 66),
      colorFlag: v.u8(w + 67),
      leftSiblingId: v.i32(w + 68),
      rightSiblingId: v.i32(w + 72),
      childId: v.i32(w + 76),
      clsid: v.slice(w + 80, 16),
      stateBits: v.u32(w + 96),
      creationTime: v.u64(w + 100),
      modifiedTime: v.u64(w + 108),
      startSector: v.i32(w + 116),
      streamSize: o === 3 ? v.u32(w + 120) : v.u64(w + 120),
      children: [],
      parentId: null
    });
  }
  const j = pr(P), z = hr(P), se = [];
  if (x && y >= 0) {
    const w = Ae(r, l, y, I), R = new k(w);
    for (let L = 0; L + 4 <= w.length; L += 4)
      se.push(R.i32(L));
  }
  const tt = Ae(r, l, j.startSector, I, j.streamSize);
  function Jn(w) {
    if (w.startSector < 0)
      return new Uint8Array(0);
    const R = dn(w.startSector, (V) => {
      var Y;
      return (Y = se[V]) != null ? Y : pe;
    }), L = new Uint8Array(R.length * c);
    let M = 0;
    for (const V of R) {
      const Y = V * c, ke = Y + c;
      if (ke > tt.length)
        throw new Error(`Mini sector ${V} is out of bounds`);
      L.set(tt.subarray(Y, ke), M), M += c;
    }
    return L.subarray(0, Math.min(w.streamSize, L.length));
  }
  function Kn(w) {
    var L;
    const R = typeof w == "string" ? (L = z.get(w)) != null ? L : null : w;
    return !R || R.objectType !== 2 && R.objectType !== 5 ? null : R.objectType === 5 ? tt : R.streamSize < m && R.startSector >= 0 && se.length ? Jn(R) : Ae(r, l, R.startSector, I, R.streamSize);
  }
  function Gn(w) {
    var L;
    const R = typeof w == "string" ? (L = z.get(w)) != null ? L : null : w;
    return R ? (R.children || []).map((M) => P[M]).filter((M) => !!M) : [];
  }
  return {
    bytes: r,
    majorVersion: o,
    sectorSize: l,
    miniSectorSize: c,
    numDirSectors: u,
    numFatSectors: f,
    firstDirSector: h,
    transactionSignature: b,
    miniStreamCutoffSize: m,
    warnings: s,
    entries: P,
    root: j,
    pathMap: z,
    getEntry(w) {
      var R;
      return (R = z.get(w)) != null ? R : null;
    },
    getStream: Kn,
    listChildren: Gn,
    findByName(w, R = "/") {
      var V;
      const L = R === "/" ? j : (V = z.get(R)) != null ? V : null;
      if (!L)
        return null;
      const M = [L];
      for (; M.length; ) {
        const Y = M.pop();
        if (Y) {
          if (Y.name === w)
            return Y;
          for (const ke of Y.children || []) {
            const It = P[ke];
            It && M.push(It);
          }
        }
      }
      return null;
    }
  };
}
const br = {
  "": "‚",
  "": "ƒ",
  "": "„",
  "": "…",
  "": "†",
  "": "‡",
  "": "ˆ",
  "": "‰",
  "": "Š",
  "": "‹",
  "": "Œ",
  "": "‘",
  "": "’",
  "": "“",
  "": "”",
  "": "•",
  "": "–",
  "": "—",
  "": "˜",
  "": "™",
  "": "š",
  "": "›",
  "": "œ",
  "": "Ÿ"
};
function xr(e) {
  return e.replace(/[\x82-\x8C\x91-\x9C\x9F]/g, (t) => br[t] || t);
}
function yr(e) {
  var r;
  let t = "";
  for (let n = 0; n < e.length; n += 1)
    t += String.fromCharCode((r = e[n]) != null ? r : 0);
  return xr(t);
}
function wr(e) {
  return new TextDecoder("utf-16le").decode(e);
}
function Tr(e, t) {
  const r = t.fcClx, n = t.lcbClx;
  if (r == null || n == null || n <= 0)
    throw new Error("FIB does not point to a CLX structure");
  const s = e.subarray(r, r + n), o = new k(s);
  let a = 0;
  const i = [];
  for (; a < s.length && o.u8(a) === 1; ) {
    const f = o.u16(a + 1);
    i.push({ type: 1, cbGrpprl: f, bytes: o.slice(a + 3, f) }), a += 3 + f;
  }
  if (o.u8(a) !== 2)
    throw new Error("CLX does not contain a Pcdt marker");
  const l = o.u32(a + 1), c = o.slice(a + 5, l), u = Pr(c);
  return { prcs: i, pcdt: { lcb: l, pieceTable: u } };
}
function Pr(e) {
  var a, i;
  if (!e.length)
    throw new Error("Empty PlcPcd");
  const t = new k(e), r = (e.length - 4) / 12;
  if (!Number.isInteger(r) || r < 0)
    throw new Error("Invalid PlcPcd size");
  const n = [];
  for (let l = 0; l < r + 1; l += 1)
    n.push(t.u32(l * 4));
  const s = [];
  let o = (r + 1) * 4;
  for (let l = 0; l < r; l += 1) {
    const c = t.u16(o), u = t.u32(o + 2) >>> 0, f = t.u16(o + 6), h = !!(u & 1073741824), b = u & 1073741823, m = (a = n[l]) != null ? a : 0, y = (i = n[l + 1]) != null ? i : m, x = h ? Math.floor(b / 2) : b, T = h ? y - m : (y - m) * 2;
    s.push({
      index: l,
      descriptor: c,
      fNoParaLast: !!(c & 1),
      fRaw: !!(c & 2),
      prm: f,
      compressed: h,
      fcRaw: u,
      fc: b,
      cpStart: m,
      cpEnd: y,
      actualByteStart: x,
      byteLength: T,
      actualByteEnd: x + T
    }), o += 8;
  }
  return { cps: n, pieces: s };
}
function Sr(e, t) {
  const r = e.subarray(t.actualByteStart, t.actualByteEnd);
  return t.compressed ? yr(r) : wr(r);
}
function Cr(e, t) {
  return t.pcdt.pieceTable.pieces.map((r) => Sr(e, r));
}
function je(e, t, r = null) {
  const n = e.pcdt.pieceTable.pieces, s = r == null ? !!(t & 1073741824) : r, o = r == null ? t & 1073741823 : t, a = s ? Math.floor(o / 2) : o;
  for (const i of n) {
    if (a < i.actualByteStart || a > i.actualByteEnd)
      continue;
    const l = a - i.actualByteStart, c = i.cpStart + Math.floor(l / (i.compressed ? 1 : 2));
    if (c >= i.cpStart && c <= i.cpEnd)
      return c;
  }
  for (const i of n) {
    if (t < i.actualByteStart || t > i.actualByteEnd)
      continue;
    const l = t - i.actualByteStart, c = i.cpStart + Math.floor(l / (i.compressed ? 1 : 2));
    if (c >= i.cpStart && c <= i.cpEnd)
      return c;
  }
  return null;
}
function Er(e, t, r, n, s) {
  var a;
  if (s <= n)
    return "";
  const o = [];
  for (let i = 0; i < t.pcdt.pieceTable.pieces.length; i += 1) {
    const l = t.pcdt.pieceTable.pieces[i];
    if (l.cpEnd <= n)
      continue;
    if (l.cpStart >= s)
      break;
    const c = Math.max(n, l.cpStart) - l.cpStart, u = Math.min(s, l.cpEnd) - l.cpStart;
    o.push(((a = r[i]) != null ? a : "").slice(c, u));
  }
  return o.join("");
}
function Br(e) {
  const t = [];
  let r = 0;
  for (let n = 0; n < e.length; n += 1) {
    const s = e[n];
    (s === "\r" || s === "\x07") && (t.push({ cpStart: r, cpEnd: n + 1, terminator: s }), r = n + 1);
  }
  return r < e.length && t.push({ cpStart: r, cpEnd: e.length, terminator: "" }), t;
}
const q = {
  paragraph: "\r",
  cellMark: "\x07",
  picture: "",
  fieldStart: "",
  fieldSeparator: "",
  fieldEnd: "",
  annotationRef: "",
  hardLineBreak: "\v",
  pageBreak: "\f",
  nonBreakingHyphen: "",
  nonRequiredHyphen: ""
}, Fr = {
  0: "none",
  1: "single",
  2: "word",
  3: "double",
  4: "dotted",
  5: "hidden",
  6: "thick",
  7: "dash",
  9: "dot-dash",
  10: "dot-dot-dash",
  11: "wave",
  16: "dotted-heavy",
  17: "dashed-heavy",
  18: "dash-long",
  19: "dash-long-heavy",
  20: "dot-dash-heavy",
  21: "dot-dot-dash-heavy",
  22: "wavy-heavy",
  23: "wavy-double"
}, Rr = {
  0: "left",
  1: "center",
  2: "right",
  3: "justify",
  4: "distributed"
}, kr = {
  0: "top",
  1: "center",
  2: "bottom"
}, bt = {
  1: "#000000",
  2: "#0000ff",
  3: "#00ffff",
  4: "#00ff00",
  5: "#ff00ff",
  6: "#ff0000",
  7: "#ffff00",
  8: "#ffffff",
  9: "#000080",
  10: "#008080",
  11: "#008000",
  12: "#800080",
  13: "#800000",
  14: "#808000",
  15: "#808080",
  16: "#c0c0c0"
}, Ar = [
  "StshfOrig",
  "Stshf",
  "PlcffndRef",
  "PlcffndTxt",
  "PlcfandRef",
  "PlcfandTxt",
  "PlcfSed",
  "PlcPad",
  "PlcfPhe",
  "SttbfGlsy",
  "PlcfGlsy",
  "PlcfHdd",
  "PlcfBteChpx",
  "PlcfBtePapx",
  "PlcfSea",
  "SttbfFfn",
  "PlcfFldMom",
  "PlcfFldHdr",
  "PlcfFldFtn",
  "PlcfFldAtn",
  "PlcfFldMcr",
  "SttbfBkmk",
  "PlcfBkf",
  "PlcfBkl",
  "Cmds",
  "Unused1",
  "SttbfMcr",
  "PrDrvr",
  "PrEnvPort",
  "PrEnvLand",
  "Wss",
  "Dop",
  "SttbfAssoc",
  "Clx",
  "PlcfPgdFtn",
  "AutosaveSource",
  "GrpXstAtnOwners",
  "SttbfAtnBkmk",
  "Unused2",
  "Unused3",
  "PlcSpaMom",
  "PlcSpaHdr",
  "PlcfAtnBkf",
  "PlcfAtnBkl",
  "Pms",
  "FormFldSttbs",
  "PlcfendRef",
  "PlcfendTxt",
  "PlcfFldEdn",
  "Unused4",
  "DggInfo",
  "SttbfRMark",
  "SttbfCaption",
  "SttbfAutoCaption",
  "PlcfWkb",
  "PlcfSpl",
  "PlcftxbxTxt",
  "PlcfFldTxbx",
  "PlcfHdrtxbxTxt",
  "PlcffldHdrTxbx",
  "StwUser",
  "SttbTtmbd",
  "CookieData",
  "PgdMotherOldOld",
  "BkdMotherOldOld",
  "PgdFtnOldOld",
  "BkdFtnOldOld",
  "PgdEdnOldOld",
  "BkdEdnOldOld",
  "SttbfIntlFld",
  "RouteSlip",
  "SttbSavedBy",
  "SttbFnm",
  "PlfLst",
  "PlfLfo",
  "PlcfTxbxBkd",
  "PlcfTxbxHdrBkd",
  "DocUndoWord9",
  "RgbUse",
  "Usp",
  "Uskf",
  "PlcupcRgbUse",
  "PlcupcUsp",
  "SttbGlsyStyle",
  "Plgosl",
  "Plcocx",
  "PlcfBteLvc",
  "dwLowDateTime",
  "dwHighDateTime",
  "PlcfLvcPre10",
  "PlcfAsumy",
  "PlcfGram",
  "SttbListNames",
  "SttbfUssr"
];
function vr(e) {
  const t = new k(e), r = Or(t);
  let n = 32;
  const s = t.u16(n);
  n += 2;
  const o = t.slice(n, s * 2);
  n += s * 2;
  const a = t.u16(n);
  n += 2;
  const i = t.slice(n, a * 4);
  n += a * 4;
  const l = t.u16(n);
  n += 2;
  const c = t.slice(n, l * 8);
  n += l * 8;
  const u = t.ensure(n, 2) ? t.u16(n) : 0;
  t.ensure(n, 2) && (n += 2);
  const f = u ? t.slice(n, u * 2) : new Uint8Array(0), h = Ir(i), b = Dr(c, l), m = u ? new k(f).u16(0) : 0;
  return {
    base: r,
    csw: s,
    cslw: a,
    cbRgFcLcb: l,
    cswNew: u,
    nFibNew: m,
    fibRgWBytes: o,
    fibRgLwBytes: i,
    fibRgFcLcbBytes: c,
    fibRgCswNewBytes: f,
    fibRgLw: h,
    fibRgFcLcb: b
  };
}
function Or(e) {
  const t = e.u16(0), r = e.u16(2), n = e.u16(6), s = e.u16(8), o = e.u16(10), a = e.u16(18);
  return {
    wIdent: t,
    nFib: r,
    lid: n,
    pnNext: s,
    flags: o,
    envFlags: a,
    fDot: !!(o & 1),
    fGlsy: !!(o & 2),
    fComplex: !!(o & 4),
    fHasPic: !!(o & 8),
    cQuickSaves: o >> 4 & 15,
    fEncrypted: !!(o & 256),
    fWhichTblStm: o & 512 ? 1 : 0,
    fReadOnlyRecommended: !!(o & 1024),
    fWriteReservation: !!(o & 2048),
    fExtChar: !!(o & 4096),
    fLoadOverride: !!(o & 8192),
    fFarEast: !!(o & 16384),
    fObfuscated: !!(o & 32768),
    nFibBack: e.u16(12),
    lKey: e.u32(14),
    envr: e.u8(18),
    fMac: !!(a & 1),
    fEmptySpecial: !!(a & 2),
    fLoadOverridePage: !!(a & 4)
  };
}
function Ir(e) {
  var n, s, o, a, i, l, c, u, f;
  const t = new k(e), r = [];
  for (let h = 0; h + 4 <= e.length; h += 4)
    r.push(t.i32(h));
  return {
    raw: r,
    cbMac: (n = r[0]) != null ? n : 0,
    ccpText: (s = r[3]) != null ? s : 0,
    ccpFtn: (o = r[4]) != null ? o : 0,
    ccpHdd: (a = r[5]) != null ? a : 0,
    ccpMcr: (i = r[6]) != null ? i : 0,
    ccpAtn: (l = r[7]) != null ? l : 0,
    ccpEdn: (c = r[8]) != null ? c : 0,
    ccpTxbx: (u = r[9]) != null ? u : 0,
    ccpHdrTxbx: (f = r[10]) != null ? f : 0
  };
}
function Dr(e, t) {
  const r = new k(e), n = { _pairs: [] }, s = [];
  for (let o = 0; o < t && o * 8 + 8 <= e.length; o += 1) {
    const a = r.u32(o * 8), i = r.u32(o * 8 + 4), l = Ar[o] || `Field${o}`;
    n[`fc${l}`] = a, n[`lcb${l}`] = i, s.push({ name: l, fc: a, lcb: i, index: o });
  }
  return n._pairs = s, n;
}
const d = {
  sprmCFRMarkDel: 2048,
  sprmCFRMark: 2049,
  sprmCFFldVanish: 2050,
  sprmCPicLocation: 27139,
  sprmCIbstRMark: 18436,
  sprmCDttmRMark: 26629,
  sprmCFData: 2054,
  sprmCFOle2: 2058,
  sprmCHighlight: 10764,
  sprmCIstd: 18992,
  sprmCPlain: 10803,
  sprmCFBold: 2101,
  sprmCFItalic: 2102,
  sprmCFStrike: 2103,
  sprmCFOutline: 2104,
  sprmCFShadow: 2105,
  sprmCFSmallCaps: 2106,
  sprmCFCaps: 2107,
  sprmCFVanish: 2108,
  sprmCKul: 10814,
  sprmCDxaSpace: 34880,
  sprmCIco: 10818,
  sprmCHps: 19011,
  sprmCHpsPos: 18501,
  sprmCSymbol: 27145,
  sprmCHpsKern: 18507,
  sprmCRgFtc0: 19023,
  sprmCRgFtc1: 19024,
  sprmCRgFtc2: 19025,
  sprmCCharScale: 18514,
  sprmCFDStrike: 10835,
  sprmCFImprint: 2132,
  sprmCFSpec: 2133,
  sprmCFObj: 2134,
  sprmCFEmboss: 2136,
  sprmCFBiDi: 2138,
  sprmCFBoldBi: 2140,
  sprmCFItalicBi: 2141,
  sprmCFtcBi: 19038,
  sprmCIcoBi: 19040,
  sprmCHpsBi: 19041,
  sprmPIstd: 17920,
  sprmPIstdPermute: 50689,
  sprmPIncLvl: 9730,
  sprmPJc80: 9219,
  sprmPFKeep: 9221,
  sprmPFKeepFollow: 9222,
  sprmPFPageBreakBefore: 9223,
  sprmPIlvl: 9738,
  sprmPIlfo: 17931,
  sprmPChgTabsPapx: 50701,
  sprmPDxaRight80: 33806,
  sprmPDxaLeft80: 33807,
  sprmPNest80: 17936,
  sprmPDxaLeft180: 33809,
  sprmPDyaLine: 25618,
  sprmPDyaBefore: 42003,
  sprmPDyaAfter: 42004,
  sprmPChgTabs: 50709,
  sprmPFInTable: 9238,
  sprmPFTtp: 9239,
  sprmPDxaAbs: 33816,
  sprmPDyaAbs: 33817,
  sprmPDxaWidth: 33818,
  sprmPPc: 9755,
  sprmPWr: 9251,
  sprmPBrcTop80: 25636,
  sprmPBrcLeft80: 25637,
  sprmPBrcBottom80: 25638,
  sprmPBrcRight80: 25639,
  sprmPBrcBetween80: 25640,
  sprmPBrcBar80: 26153,
  sprmPWHeightAbs: 17451,
  sprmPDcs: 17452,
  sprmPShd80: 17453,
  sprmPDyaFromText: 33838,
  sprmPDxaFromText: 33839,
  sprmPFLocked: 9264,
  sprmPFWidowControl: 9265,
  sprmPFBiDi: 9281,
  sprmPHugePapx: 26182,
  sprmPFAdjustRight: 9288,
  sprmPItap: 26185,
  sprmPDtap: 26186,
  sprmPFInnerTableCell: 9291,
  sprmPFInnerTtp: 9292,
  sprmPShd: 50765,
  sprmPBrcTop: 50766,
  sprmPBrcLeft: 50767,
  sprmPBrcBottom: 50768,
  sprmPBrcRight: 50769,
  sprmPBrcBetween: 50770,
  sprmPBrcBar: 50771,
  sprmPDxcRight: 17493,
  sprmPDxcLeft: 17494,
  sprmPDxcLeft1: 17495,
  sprmPDylBefore: 17496,
  sprmPDylAfter: 17497,
  sprmPFDyaBeforeAuto: 9307,
  sprmPFDyaAfterAuto: 9308,
  sprmPDxaRight: 33885,
  sprmPDxaLeft: 33886,
  sprmPNest: 18015,
  sprmPDxaLeft1: 33888,
  sprmPJc: 9313,
  sprmTJc90: 21504,
  sprmTDxaLeft: 38401,
  sprmTDxaGapHalf: 38402,
  sprmTFCantSplit90: 13315,
  sprmTTableHeader: 13316,
  sprmTTableBorders80: 54789,
  sprmTDyaRowHeight: 37895,
  sprmTDefTable: 54792,
  sprmTDefTableShd80: 54793,
  sprmTTlp: 29706,
  sprmTFBiDi: 22027,
  sprmTDefTableShd3rd: 54796,
  sprmTPc: 13837,
  sprmTDxaAbs: 37902,
  sprmTDyaAbs: 37903,
  sprmTDxaFromText: 37904,
  sprmTDyaFromText: 37905,
  sprmTDefTableShd: 54802,
  sprmTTableBorders: 54803,
  sprmTTableWidth: 62996,
  sprmTFAutofit: 13845,
  sprmTWidthBefore: 62999,
  sprmTWidthAfter: 63e3,
  sprmTSetBrc80: 54816,
  sprmTInsert: 30241,
  sprmTDelete: 22050,
  sprmTDxaCol: 30243,
  sprmTMerge: 22052,
  sprmTSplit: 22053,
  sprmTTextFlow: 30249,
  sprmTVertMerge: 54827,
  sprmTVertAlign: 54828,
  sprmTSetShd: 54829,
  sprmTSetShdOdd: 54830,
  sprmTSetBrc: 54831,
  sprmTCellPadding: 54834,
  sprmTCellPaddingDefault: 54835,
  sprmTCellWidth: 54837,
  sprmTFitText: 63030,
  sprmTFCellNoWrap: 54841,
  sprmTIstd: 22074
}, Lr = /* @__PURE__ */ new Set([
  d.sprmPChgTabs,
  d.sprmPChgTabsPapx,
  d.sprmTDefTable,
  d.sprmTTableBorders80,
  d.sprmTTableBorders,
  d.sprmTDefTableShd,
  d.sprmTDefTableShd80,
  d.sprmTDefTableShd3rd,
  d.sprmTSetBrc80,
  d.sprmTSetBrc,
  d.sprmTSetShd,
  d.sprmTSetShdOdd,
  d.sprmTCellPadding,
  d.sprmTCellPaddingDefault,
  d.sprmTCellWidth,
  d.sprmTVertAlign,
  d.sprmTVertMerge,
  d.sprmTTextFlow,
  d.sprmTDxaCol
]);
function _r(e) {
  return e >> 13 & 7;
}
function $(e, t = 0) {
  var r, n;
  return t + 2 > e.length ? 0 : ((r = e[t]) != null ? r : 0) | ((n = e[t + 1]) != null ? n : 0) << 8;
}
function _(e, t = 0) {
  const r = $(e, t);
  return r > 32767 ? r - 65536 : r;
}
function _e(e, t = 0) {
  var r, n, s, o;
  return t + 4 > e.length ? 0 : (((r = e[t]) != null ? r : 0) | ((n = e[t + 1]) != null ? n : 0) << 8 | ((s = e[t + 2]) != null ? s : 0) << 16 | ((o = e[t + 3]) != null ? o : 0) << 24 >>> 0) >>> 0;
}
function le(e, t = 0) {
  var r, n;
  return {
    first: (r = e[t]) != null ? r : 0,
    lim: (n = e[t + 1]) != null ? n : 0
  };
}
function K(e, t = 0) {
  if (t + 4 > e.length)
    return null;
  const r = _e(e, t) >>> 0, n = r & 255, s = r >> 8 & 255, o = r >> 16 & 255;
  return { raw: r, lineWidth: n, borderType: s, color: o };
}
function Mr(e, t = 0) {
  const r = $(e, t);
  return {
    raw: r,
    horzMerge: r & 3,
    textFlow: r >> 2 & 7,
    vertMerge: r >> 5 & 3,
    vertAlign: r >> 7 & 3,
    ftsWidth: r >> 9 & 7,
    fitText: !!(r & 4096),
    noWrap: !!(r & 8192),
    hideMark: !!(r & 16384)
  };
}
function pn(e, t = 0) {
  return t + 20 > e.length ? null : {
    tcgrf: Mr(e, t),
    wWidth: $(e, t + 2),
    borders: {
      top: K(e, t + 4) || {},
      left: K(e, t + 8) || {},
      bottom: K(e, t + 12) || {},
      right: K(e, t + 16) || {}
    }
  };
}
function Nr(e) {
  var a;
  if (e.length < 3)
    return null;
  const t = $(e, 0), r = (a = e[2]) != null ? a : 0;
  let n = 3;
  const s = [];
  for (let i = 0; i < r + 1 && n + 2 <= e.length; i += 1)
    s.push(_(e, n)), n += 2;
  const o = [];
  for (let i = 0; i < r && n + 20 <= e.length; i += 1) {
    const l = pn(e, n);
    l && o.push(l), n += 20;
  }
  return { cb: t, numberOfColumns: r, rgdxaCenter: s, cells: o };
}
function fe(e) {
  var r, n;
  return e.length ? {
    cb: (r = e[0]) != null ? r : 0,
    range: le(e, 1),
    value: (n = e[3]) != null ? n : 0,
    extra: e.subarray(4)
  } : null;
}
function at(e) {
  var n, s;
  if (!e.length)
    return null;
  const t = (n = e[0]) != null ? n : 0, r = $(e, 4);
  return {
    cb: t,
    range: le(e, 1),
    ftsWidth: (s = e[3]) != null ? s : 0,
    width: r,
    wWidth: r
  };
}
function Ur(e) {
  var t;
  return e.length ? {
    cb: (t = e[0]) != null ? t : 0,
    range: le(e, 1),
    border: K(e, 3) || {},
    extra: e.subarray(7)
  } : null;
}
function it(e) {
  var r;
  if (!e.length)
    return null;
  const t = $(e, 1);
  return {
    ftsWidth: (r = e[0]) != null ? r : 0,
    width: t,
    wWidth: t
  };
}
function $r(e) {
  var i;
  if (!e.length)
    return null;
  const t = (i = e[0]) != null ? i : 0, r = le(e, 1), n = r.lim - r.first;
  let s = 3;
  const o = [];
  for (let l = 0; l < n && s + 2 <= e.length; l += 1)
    o.push(_(e, s)), s += 2;
  const a = [];
  for (let l = 0; l < n && s + 20 <= e.length; l += 1)
    a.push(pn(e, s)), s += 20;
  return { cb: t, range: r, itcFirst: r.first, ctc: n, dxaCol: o, cells: a };
}
function Hr(e, t, r) {
  var s, o;
  const n = r >> 13 & 7;
  if (t >= e.length)
    return 0;
  if (r === d.sprmTDefTable) {
    const a = $(e, t);
    return a ? a + 1 : 0;
  }
  if (Lr.has(r))
    return ((s = e[t]) != null ? s : 0) + 1;
  switch (n) {
    case 0:
      return 1;
    case 1:
      return 1;
    case 2:
      return 2;
    case 3:
      return 4;
    case 4:
      return 2;
    case 5:
      return 2;
    case 6:
      return ((o = e[t]) != null ? o : 0) + 1;
    case 7:
      return 3;
    default:
      return 0;
  }
}
function F(e) {
  var t;
  return !!((t = e[0]) != null && t);
}
function g(e, t, r, n, s) {
  return { kind: e, name: t, value: r, raw: n, operandBytes: s };
}
function jr(e, t) {
  var o, a, i, l, c, u, f, h;
  const r = _r(e), n = t, s = e;
  switch (e) {
    case d.sprmCPicLocation:
      return g("char", "pictureOffset", _e(n, 0) >>> 0, s, n);
    case d.sprmCFData:
      return g("char", "data", F(n), s, n);
    case d.sprmCFOle2:
      return g("char", "ole2", F(n), s, n);
    case d.sprmCIstd:
      return g("char", "charStyleId", $(n, 0), s, n);
    case d.sprmCPlain:
      return g("char", "plain", F(n), s, n);
    case d.sprmCFBold:
      return g("char", "bold", F(n), s, n);
    case d.sprmCFItalic:
      return g("char", "italic", F(n), s, n);
    case d.sprmCFStrike:
      return g("char", "strike", F(n), s, n);
    case d.sprmCFOutline:
      return g("char", "outline", F(n), s, n);
    case d.sprmCFShadow:
      return g("char", "shadow", F(n), s, n);
    case d.sprmCFSmallCaps:
      return g("char", "smallCaps", F(n), s, n);
    case d.sprmCFCaps:
      return g("char", "caps", F(n), s, n);
    case d.sprmCFVanish:
      return g("char", "hidden", F(n), s, n);
    case d.sprmCKul:
      return g("char", "underline", (o = n[0]) != null ? o : 0, s, n);
    case d.sprmCDxaSpace:
      return g("char", "spacing", _(n, 0), s, n);
    case d.sprmCIco:
    case d.sprmCIcoBi:
      return g("char", "colorIndex", (a = n[0]) != null ? a : 0, s, n);
    case d.sprmCHighlight: {
      const b = (i = n[0]) != null ? i : 0, m = { index: b, color: bt[b] };
      return g("char", "highlight", m, s, n);
    }
    case d.sprmCHps:
    case d.sprmCHpsBi:
      return g("char", "fontSizeHalfPoints", $(n, 0), s, n);
    case d.sprmCHpsPos:
      return g("char", "positionHalfPoints", _(n, 0), s, n);
    case d.sprmCSymbol:
      return g("char", "symbol", { font: $(n, 0), charCode: $(n, 2) }, s, n);
    case d.sprmCRgFtc0:
    case d.sprmCRgFtc1:
    case d.sprmCRgFtc2:
    case d.sprmCFtcBi:
      return g("char", "fontFamilyId", $(n, 0), s, n);
    case d.sprmCCharScale:
      return g("char", "scale", $(n, 0), s, n);
    case d.sprmCFDStrike:
      return g("char", "doubleStrike", F(n), s, n);
    case d.sprmCFImprint:
      return g("char", "imprint", F(n), s, n);
    case d.sprmCFSpec:
      return g("char", "special", F(n), s, n);
    case d.sprmCFObj:
      return g("char", "object", F(n), s, n);
    case d.sprmCFEmboss:
      return g("char", "emboss", F(n), s, n);
    case d.sprmCFBiDi:
      return g("char", "rtl", F(n), s, n);
    case d.sprmCFBoldBi:
      return g("char", "boldBi", F(n), s, n);
    case d.sprmCFItalicBi:
      return g("char", "italicBi", F(n), s, n);
    case d.sprmPIstd:
      return g("para", "styleId", $(n, 0), s, n);
    case d.sprmPJc80:
    case d.sprmPJc:
      return g("para", "alignment", (l = n[0]) != null ? l : 0, s, n);
    case d.sprmPFKeep:
      return g("para", "keepLines", F(n), s, n);
    case d.sprmPFKeepFollow:
      return g("para", "keepNext", F(n), s, n);
    case d.sprmPFPageBreakBefore:
      return g("para", "pageBreakBefore", F(n), s, n);
    case d.sprmPIlvl:
      return g("para", "listLevel", (c = n[0]) != null ? c : 0, s, n);
    case d.sprmPIlfo:
      return g("para", "listId", $(n, 0), s, n);
    case d.sprmPDxaRight80:
    case d.sprmPDxaRight:
      return g("para", "rightIndent", _(n, 0), s, n);
    case d.sprmPDxaLeft80:
    case d.sprmPDxaLeft:
      return g("para", "leftIndent", _(n, 0), s, n);
    case d.sprmPDxaLeft180:
    case d.sprmPDxaLeft1:
      return g("para", "firstLineIndent", _(n, 0), s, n);
    case d.sprmPDyaLine:
      return g("para", "lineSpacing", _(n, 0), s, n);
    case d.sprmPDyaBefore:
      return g("para", "spacingBefore", _(n, 0), s, n);
    case d.sprmPDyaAfter:
      return g("para", "spacingAfter", _(n, 0), s, n);
    case d.sprmPFInTable:
      return g("para", "inTable", F(n), s, n);
    case d.sprmPFTtp:
      return g("para", "tableRowEnd", F(n), s, n);
    case d.sprmPDxaAbs:
      return g("para", "frameLeft", _(n, 0), s, n);
    case d.sprmPDyaAbs:
      return g("para", "frameTop", _(n, 0), s, n);
    case d.sprmPDxaWidth:
      return g("para", "frameWidth", _(n, 0), s, n);
    case d.sprmPPc:
      return g("para", "framePosition", (u = n[0]) != null ? u : 0, s, n);
    case d.sprmPWr:
      return g("para", "frameWrap", (f = n[0]) != null ? f : 0, s, n);
    case d.sprmPBrcTop80:
    case d.sprmPBrcTop:
      return g("para", "borderTop", K(n, 0), s, n);
    case d.sprmPBrcLeft80:
    case d.sprmPBrcLeft:
      return g("para", "borderLeft", K(n, 0), s, n);
    case d.sprmPBrcBottom80:
    case d.sprmPBrcBottom:
      return g("para", "borderBottom", K(n, 0), s, n);
    case d.sprmPBrcRight80:
    case d.sprmPBrcRight:
      return g("para", "borderRight", K(n, 0), s, n);
    case d.sprmPBrcBetween80:
    case d.sprmPBrcBetween:
      return g("para", "borderBetween", K(n, 0), s, n);
    case d.sprmPBrcBar80:
    case d.sprmPBrcBar:
      return g("para", "borderBar", K(n, 0), s, n);
    case d.sprmPWHeightAbs:
      return g("para", "frameHeight", _(n, 0), s, n);
    case d.sprmPShd80:
    case d.sprmPShd:
      return g("para", "shading", n.slice(), s, n);
    case d.sprmPDyaFromText:
      return g("para", "distanceFromTextY", _(n, 0), s, n);
    case d.sprmPDxaFromText:
      return g("para", "distanceFromTextX", _(n, 0), s, n);
    case d.sprmPFLocked:
      return g("para", "locked", F(n), s, n);
    case d.sprmPFWidowControl:
      return g("para", "widowControl", F(n), s, n);
    case d.sprmPFBiDi:
      return g("para", "rtlPara", F(n), s, n);
    case d.sprmPFAdjustRight:
      return g("para", "adjustRight", F(n), s, n);
    case d.sprmPItap:
      return g("para", "itap", _e(n, 0), s, n);
    case d.sprmPDtap:
      return g("para", "dtap", _e(n, 0), s, n);
    case d.sprmPFInnerTableCell:
      return g("para", "innerTableCell", F(n), s, n);
    case d.sprmPFInnerTtp:
      return g("para", "innerTableRowEnd", F(n), s, n);
    case d.sprmPFDyaBeforeAuto:
      return g("para", "spacingBeforeAuto", F(n), s, n);
    case d.sprmPFDyaAfterAuto:
      return g("para", "spacingAfterAuto", F(n), s, n);
    case d.sprmTJc90:
      return g("table", "alignment", $(n, 0), s, n);
    case d.sprmTDxaLeft:
      return g("table", "leftIndent", _(n, 0), s, n);
    case d.sprmTDxaGapHalf:
      return g("table", "gapHalf", $(n, 0), s, n);
    case d.sprmTFCantSplit90:
      return g("table", "cantSplit", F(n), s, n);
    case d.sprmTTableHeader:
      return g("table", "header", F(n), s, n);
    case d.sprmTDyaRowHeight:
      return g("table", "rowHeight", _(n, 0), s, n);
    case d.sprmTDefTable:
      return g("table", "defTable", Nr(n), s, n);
    case d.sprmTFBiDi:
      return g("table", "rtl", F(n), s, n);
    case d.sprmTPc:
      return g("table", "positionCode", (h = n[0]) != null ? h : 0, s, n);
    case d.sprmTDxaAbs:
      return g("table", "absLeft", _(n, 0), s, n);
    case d.sprmTDyaAbs:
      return g("table", "absTop", _(n, 0), s, n);
    case d.sprmTDxaFromText:
      return g("table", "distanceLeft", _(n, 0), s, n);
    case d.sprmTDyaFromText:
      return g("table", "distanceTop", _(n, 0), s, n);
    case d.sprmTTableWidth:
      return g("table", "tableWidth", it(n), s, n);
    case d.sprmTFAutofit:
      return g("table", "autoFit", F(n), s, n);
    case d.sprmTWidthBefore:
      return g("table", "widthBefore", it(n), s, n);
    case d.sprmTWidthAfter:
      return g("table", "widthAfter", it(n), s, n);
    case d.sprmTInsert:
      return g("table", "insertCells", $r(n), s, n);
    case d.sprmTDelete:
      return g("table", "deleteCells", le(n, 0), s, n);
    case d.sprmTDxaCol:
      return g("table", "columnWidth", at(n), s, n);
    case d.sprmTMerge:
      return g("table", "merge", le(n, 0), s, n);
    case d.sprmTSplit:
      return g("table", "split", le(n, 0), s, n);
    case d.sprmTTextFlow:
      return g("table", "textFlow", fe(n), s, n);
    case d.sprmTVertMerge:
      return g("table", "vertMerge", fe(n), s, n);
    case d.sprmTVertAlign:
      return g("table", "vertAlign", fe(n), s, n);
    case d.sprmTSetShd:
    case d.sprmTSetShdOdd:
      return g("table", "setShading", fe(n), s, n);
    case d.sprmTSetBrc80:
    case d.sprmTSetBrc:
      return g("table", "setBorder", Ur(n), s, n);
    case d.sprmTCellPadding:
    case d.sprmTCellPaddingDefault:
      return g("table", "cellPadding", at(n), s, n);
    case d.sprmTCellWidth:
      return g("table", "cellWidth", at(n), s, n);
    case d.sprmTFitText:
      return g("table", "fitText", fe(n), s, n);
    case d.sprmTFCellNoWrap:
      return g("table", "cellNoWrap", fe(n), s, n);
    case d.sprmTIstd:
      return g("table", "styleId", $(n, 0), s, n);
    default:
      return { kind: r === 1 ? "para" : r === 2 ? "char" : r === 5 ? "table" : "unknown", name: `sprm_${e.toString(16)}`, value: n, raw: e, operandBytes: n };
  }
}
function Fe(e, t, r) {
  const n = [];
  let s = t;
  for (; s + 2 <= r; ) {
    const o = $(e, s);
    s += 2;
    const a = Hr(e, s, o);
    if (!a || s + a > r)
      break;
    const i = e.subarray(s, s + a);
    s += a, n.push(jr(o, i));
  }
  return n;
}
function hn(e, t, r) {
  if (t == null || r == null || r <= 0)
    return { aFC: [], pages: [] };
  const n = e.subarray(t, t + r), s = new k(n), o = Math.floor((r - 4) / 8);
  if (o <= 0)
    return { aFC: [], pages: [] };
  const a = [];
  for (let c = 0; c <= o; c += 1)
    a.push(s.u32(c * 4) >>> 0);
  const i = [], l = (o + 1) * 4;
  for (let c = 0; c < o; c += 1)
    i.push(s.u32(l + c * 4) & 4194303);
  return { aFC: a, pages: i };
}
function Wr(e, t) {
  const r = t * 512, n = e.subarray(r, r + 512);
  if (n.length < 512)
    return null;
  const s = new k(n), o = s.u8(511);
  if (!o || o > 100)
    return null;
  const a = [];
  for (let u = 0; u <= o; u += 1)
    a.push(s.u32(u * 4) >>> 0);
  const i = [], l = [], c = (o + 1) * 4;
  for (let u = 0; u < o; u += 1) {
    const f = s.u8(c + u);
    i.push(f), l.push(f ? r + f * 2 : 0);
  }
  return { crun: o, rgfc: a, rgb: i, chpxOffsets: l };
}
function zr(e, t) {
  const r = t * 512, n = e.subarray(r, r + 512);
  if (n.length < 512)
    return null;
  const s = new k(n), o = s.u8(511);
  if (o > 29)
    return null;
  const a = [];
  for (let u = 0; u <= o; u += 1)
    a.push(s.u32(u * 4) >>> 0);
  const i = (o + 1) * 4, l = [], c = [];
  for (let u = 0; u < o; u += 1) {
    const f = s.u8(i + u * 13);
    l.push(f), c.push(f ? r + f * 2 : 0);
  }
  return { cpara: o, rgfc: a, bOffsets: l, papxOffsets: c };
}
function qr(e, t, r, n) {
  var l, c;
  const s = r.fibRgFcLcb.fcPlcfBteChpx, o = r.fibRgFcLcb.lcbPlcfBteChpx, a = hn(t, s, o), i = [];
  for (const u of a.pages) {
    const f = Wr(e, u);
    if (f)
      for (let h = 0; h < f.crun; h += 1) {
        const b = (l = f.rgfc[h]) != null ? l : 0, m = (c = f.rgfc[h + 1]) != null ? c : b, y = je(n, b), x = je(n, m);
        if (y == null || x == null || x <= y)
          continue;
        const T = f.chpxOffsets[h] ? Jr(e, f.chpxOffsets[h]) : [];
        i.push({ cpStart: y, cpEnd: x, fcStart: b, fcEnd: m, properties: T });
      }
  }
  return i.sort((u, f) => u.cpStart - f.cpStart || u.cpEnd - f.cpEnd), i;
}
function Vr(e, t, r, n) {
  var l, c, u;
  const s = r.fibRgFcLcb.fcPlcfBtePapx, o = r.fibRgFcLcb.lcbPlcfBtePapx, a = hn(t, s, o), i = [];
  for (const f of a.pages) {
    const h = zr(e, f);
    if (h)
      for (let b = 0; b < h.cpara; b += 1) {
        const m = (l = h.rgfc[b]) != null ? l : 0, y = (c = h.rgfc[b + 1]) != null ? c : m, x = je(n, m), T = je(n, y);
        if (x == null || T == null || T < x)
          continue;
        const S = h.papxOffsets[b] ? Kr(e, h.papxOffsets[b]) : { styleId: 0, properties: [] };
        i.push({ cpStart: x, cpEnd: T, fcStart: m, fcEnd: y, styleId: S.styleId, properties: S.properties, rawOffset: (u = h.papxOffsets[b]) != null ? u : 0 });
      }
  }
  return i.sort((f, h) => f.cpStart - h.cpStart || f.cpEnd - h.cpEnd), i;
}
function Jr(e, t) {
  var o;
  if (t == null || t < 0 || t >= e.length)
    return [];
  const r = (o = e[t]) != null ? o : 0;
  if (!r)
    return [];
  const n = t + 1, s = Math.min(e.length, n + r);
  return Fe(e, n, s);
}
function Kr(e, t) {
  if (t == null || t < 0 || t >= e.length)
    return { styleId: 0, properties: [] };
  const r = new k(e), n = r.u8(t);
  if (n === 0 && !r.ensure(t + 1, 1))
    return { styleId: 0, properties: [] };
  let s, o;
  if (n === 0 ? (o = r.u8(t + 1) * 2, s = t + 2) : (o = n - 1, s = t + 1), o < 2 || !r.ensure(s, Math.max(2, o)))
    return { styleId: 0, properties: [] };
  const a = r.u16(s), i = s + 2, l = Math.min(e.length, s + o), c = Fe(e, i, l);
  return { styleId: a, properties: c };
}
function Gr(e, t) {
  let r = t;
  for (; r + 1 < e.length && !(e[r] === 0 && e[r + 1] === 0); )
    r += 2;
  return { value: new TextDecoder("utf-16le").decode(e.subarray(t, r)), nextOffset: Math.min(e.length, r + 2) };
}
function Xr(e, t) {
  if (!e.length)
    return null;
  const r = new k(e), n = r.u8(0), s = r.u8(1), o = r.u16(2), a = r.u8(4), i = r.u8(5), l = r.slice(6, 10), c = r.slice(16, 24), u = 40;
  let f = "", h = "";
  if (e.length > u) {
    const m = new TextDecoder("utf-16le").decode(e.subarray(u)).split("\0").filter(Boolean);
    if (f = m[0] || "", h = m[1] || "", !h && i) {
      const y = u + i * 2;
      y + 2 <= e.length && (h = Gr(e, y).value);
    }
  }
  return {
    index: t,
    cbFfnM1: n,
    ffid: s,
    weight: o,
    charset: a,
    ixchSzAlt: i,
    panose: l,
    fontSignature: c,
    name: f,
    altName: h
  };
}
function Qr(e, t) {
  const r = t.fcSttbfFfn, n = t.lcbSttbfFfn;
  if (r == null || n == null || n <= 0)
    return {
      header: null,
      fonts: [],
      byIndex(u) {
        return u == null, null;
      }
    };
  const s = e.subarray(r, r + n), o = new k(s), a = o.u16(0), i = o.u16(2);
  let l = 4;
  const c = [];
  for (let u = 0; u < a && l < s.length; u += 1) {
    const h = o.u8(l) + 1, b = s.subarray(l, Math.min(l + h, s.length)), m = Xr(b, u);
    m && c.push(m), l += h + i;
  }
  return {
    header: { cData: a, cbExtra: i },
    fonts: c,
    byIndex(u) {
      return u == null || u < 0 ? null : c[u] || null;
    }
  };
}
function mn(e, t, r = 0) {
  if (r + t.length > e.length)
    return !1;
  for (let n = 0; n < t.length; n += 1)
    if (e[r + n] !== t[n])
      return !1;
  return !0;
}
function Yr(e) {
  const t = [
    { mime: "image/png", magic: [137, 80, 78, 71], end: Zr },
    { mime: "image/jpeg", magic: [255, 216, 255], end: es },
    { mime: "image/gif", magic: [71, 73, 70, 56], end: ts },
    { mime: "image/bmp", magic: [66, 77], end: ns },
    { mime: "image/tiff", magic: [73, 73, 42, 0], end: null },
    { mime: "image/tiff", magic: [77, 77, 0, 42], end: null },
    { mime: "image/emf", magic: [1, 0, 0, 0], end: rs },
    { mime: "image/wmf", magic: [215, 205, 198, 154], end: null }
  ];
  for (let r = 0; r < e.length - 4; r += 1)
    for (const n of t)
      if (mn(e, n.magic, r)) {
        const s = n.end ? n.end(e, r) : e.length;
        return { mime: n.mime, start: r, end: s || e.length };
      }
  return null;
}
function Zr(e, t) {
  for (let r = t + 8; r + 8 < e.length; r += 1)
    if (mn(e, [73, 69, 78, 68], r))
      return r + 8;
  return e.length;
}
function es(e, t) {
  for (let r = t + 2; r + 1 < e.length; r += 1)
    if (e[r] === 255 && e[r + 1] === 217)
      return r + 2;
  return e.length;
}
function ts(e, t) {
  for (let r = t + 6; r < e.length; r += 1)
    if (e[r] === 59)
      return r + 1;
  return e.length;
}
function ns(e, t) {
  if (t + 6 <= e.length) {
    const r = new k(e.subarray(t)).u32(2);
    if (r > 0 && t + r <= e.length)
      return t + r;
  }
  return e.length;
}
function rs(e, t) {
  if (t + 48 <= e.length) {
    const n = new k(e.subarray(t)).u32(40);
    if (n > 0 && t + n <= e.length)
      return t + n;
  }
  return e.length;
}
function ss(e, t, r = {}) {
  if (!e || t == null || t < 0 || t + 68 > e.length)
    return null;
  const n = new k(e), s = n.i32(t), o = n.u16(t + 4), a = s > 0 && t + s <= e.length ? s : Math.min(e.length - t, r.maxPictureBytes || 8 * 1024 * 1024), i = e.subarray(t, t + a), l = Math.min(o || 68, i.length), c = Yr(i.subarray(l));
  if (!c)
    return {
      id: G("asset-img"),
      type: "image",
      mime: "application/octet-stream",
      bytes: i,
      dataUrl: He(i, "application/octet-stream"),
      meta: { pictureOffset: t, lcb: s, cbHeader: o }
    };
  const u = l + c.start, f = Math.min(l + c.end, i.length), h = i.subarray(u, f);
  return {
    id: G("asset-img"),
    type: "image",
    mime: c.mime,
    bytes: h,
    dataUrl: He(h, c.mime),
    meta: { pictureOffset: t, lcb: s, cbHeader: o }
  };
}
function ct(e, t) {
  let r = t;
  for (; r < e.length && e[r] !== 0; )
    r += 1;
  return { value: new TextDecoder("windows-1252").decode(e.subarray(t, r)), nextOffset: r + 1 };
}
function os(e) {
  const t = new k(e), r = [4, 6];
  for (const n of r)
    try {
      let s = n;
      const o = ct(e, s);
      s = o.nextOffset;
      const a = ct(e, s);
      s = a.nextOffset;
      const i = ct(e, s);
      if (s = i.nextOffset, s + 4 > e.length)
        continue;
      const l = t.u32(s);
      if (s += 4, l > 0 && s + l <= e.length) {
        const c = e.subarray(s, s + l);
        return { label: o.value, originalPath: a.value, tempPath: i.value, dataSize: l, bytes: c };
      }
    } catch (s) {
    }
  return null;
}
function as(e, t) {
  const r = e.listChildren(t).filter((l) => l.objectType === 2 || l.objectType === 5), n = new Map(r.map((l) => [l.name.toLowerCase(), l])), s = n.get("objinfo") || n.get("objinfo") || n.get(String.fromCharCode(3) + "objinfo"), o = n.get("ole10native") || n.get("ole10native") || n.get(String.fromCharCode(1) + "ole10native"), a = n.get("package") || n.get("contents") || n.get("content"), i = { entry: t, streams: r, displayName: t.name, attachment: null, objectData: null };
  if (o) {
    const l = e.getStream(o) || new Uint8Array(0), c = os(l);
    if (c) {
      const u = c.label || c.originalPath.split(/[\\/]/).pop() || `${t.name}.bin`, f = {
        id: G("asset-ole"),
        type: "attachment",
        name: u,
        mime: "application/octet-stream",
        bytes: c.bytes,
        dataUrl: He(c.bytes, "application/octet-stream"),
        meta: c
      };
      return i.attachment = f, i;
    }
  }
  if (a) {
    const l = e.getStream(a) || new Uint8Array(0), c = {
      id: G("asset-pkg"),
      type: "attachment",
      name: `${un(t.name)}.bin`,
      mime: "application/octet-stream",
      bytes: l,
      dataUrl: He(l, "application/octet-stream"),
      meta: { stream: a.name }
    };
    i.attachment = c;
  }
  return s && (i.objectData = e.getStream(s) || new Uint8Array(0)), i;
}
function is(e) {
  const t = e.getEntry("/ObjectPool");
  if (!t)
    return /* @__PURE__ */ new Map();
  const r = e.listChildren(t).filter((s) => s.objectType === 1), n = /* @__PURE__ */ new Map();
  for (const s of r)
    n.set(s.name, as(e, s));
  return n;
}
function lt(e) {
  var r;
  const t = {
    bold: !1,
    italic: !1,
    strike: !1,
    underline: 0,
    fontSizeHalfPoints: void 0,
    fontFamilyId: void 0,
    colorIndex: void 0,
    highlight: void 0,
    spacing: 0,
    positionHalfPoints: 0,
    scale: 100,
    hidden: !1,
    smallCaps: !1,
    caps: !1,
    outline: !1,
    shadow: !1,
    emboss: !1,
    imprint: !1,
    rtl: !1,
    pictureOffset: void 0,
    data: !1,
    ole2: !1,
    object: !1,
    special: !1,
    charStyleId: void 0
  };
  for (const n of e || [])
    switch (n.name) {
      case "plain":
        n.value && (t.bold = !1, t.italic = !1, t.strike = !1, t.underline = 0, t.smallCaps = !1, t.caps = !1);
        break;
      case "bold":
      case "italic":
      case "strike":
      case "hidden":
      case "smallCaps":
      case "caps":
      case "outline":
      case "shadow":
      case "emboss":
      case "imprint":
      case "rtl":
      case "data":
      case "ole2":
      case "object":
      case "special":
        t[n.name] = !!n.value;
        break;
      case "underline":
        t.underline = (r = n.value) != null ? r : 0;
        break;
      case "fontSizeHalfPoints":
        t.fontSizeHalfPoints = n.value;
        break;
      case "fontFamilyId":
        t.fontFamilyId = n.value;
        break;
      case "colorIndex":
        t.colorIndex = n.value;
        break;
      case "highlight":
        t.highlight = n.value;
        break;
      case "spacing":
        t.spacing = n.value || 0;
        break;
      case "positionHalfPoints":
        t.positionHalfPoints = n.value || 0;
        break;
      case "scale":
        t.scale = n.value || 100;
        break;
      case "pictureOffset":
        t.pictureOffset = n.value;
        break;
      case "charStyleId":
        t.charStyleId = n.value;
        break;
      default:
        t[n.name] = n.value;
        break;
    }
  return t;
}
function cs(e) {
  var r;
  const t = {
    styleId: 0,
    alignment: 0,
    spacingBefore: 0,
    spacingAfter: 0,
    lineSpacing: 0,
    leftIndent: 0,
    rightIndent: 0,
    firstLineIndent: 0,
    keepLines: !1,
    keepNext: !1,
    pageBreakBefore: !1,
    widowControl: !1,
    inTable: !1,
    tableRowEnd: !1,
    innerTableCell: !1,
    innerTableRowEnd: !1,
    itap: 0,
    dtap: 0,
    listLevel: void 0,
    listId: void 0,
    rtlPara: !1,
    adjustRight: !1,
    frameLeft: void 0,
    frameTop: void 0,
    frameWidth: void 0,
    frameHeight: void 0,
    framePosition: void 0,
    frameWrap: void 0,
    borders: {},
    shading: void 0
  };
  for (const n of e || [])
    switch (n.name) {
      case "styleId":
        t.styleId = n.value || 0;
        break;
      case "alignment":
        t.alignment = (r = n.value) != null ? r : 0;
        break;
      case "spacingBefore":
        t.spacingBefore = n.value || 0;
        break;
      case "spacingAfter":
        t.spacingAfter = n.value || 0;
        break;
      case "lineSpacing":
        t.lineSpacing = n.value || 0;
        break;
      case "leftIndent":
        t.leftIndent = n.value || 0;
        break;
      case "rightIndent":
        t.rightIndent = n.value || 0;
        break;
      case "firstLineIndent":
        t.firstLineIndent = n.value || 0;
        break;
      case "keepLines":
      case "keepNext":
      case "pageBreakBefore":
      case "widowControl":
      case "inTable":
      case "tableRowEnd":
      case "innerTableCell":
      case "innerTableRowEnd":
      case "rtlPara":
      case "adjustRight":
        t[n.name] = !!n.value;
        break;
      case "itap":
        t.itap = n.value || 0;
        break;
      case "dtap":
        t.dtap = n.value || 0;
        break;
      case "listLevel":
        t.listLevel = n.value;
        break;
      case "listId":
        t.listId = n.value;
        break;
      case "frameLeft":
        t.frameLeft = n.value;
        break;
      case "frameTop":
        t.frameTop = n.value;
        break;
      case "frameWidth":
        t.frameWidth = n.value;
        break;
      case "frameHeight":
        t.frameHeight = n.value;
        break;
      case "framePosition":
        t.framePosition = n.value;
        break;
      case "frameWrap":
        t.frameWrap = n.value;
        break;
      case "borderTop":
        t.borders.top = n.value;
        break;
      case "borderLeft":
        t.borders.left = n.value;
        break;
      case "borderBottom":
        t.borders.bottom = n.value;
        break;
      case "borderRight":
        t.borders.right = n.value;
        break;
      case "borderBetween":
        t.borders.between = n.value;
        break;
      case "borderBar":
        t.borders.bar = n.value;
        break;
      case "shading":
        t.shading = n.value;
        break;
      default:
        t[n.name] = n.value;
        break;
    }
  return t;
}
function We(e) {
  var r;
  const t = {
    styleId: void 0,
    alignment: 0,
    leftIndent: 0,
    gapHalf: 0,
    cantSplit: !1,
    header: !1,
    rowHeight: 0,
    rtl: !1,
    positionCode: void 0,
    absLeft: void 0,
    absTop: void 0,
    distanceLeft: void 0,
    distanceTop: void 0,
    tableWidth: void 0,
    autoFit: void 0,
    widthBefore: void 0,
    widthAfter: void 0,
    defTable: void 0,
    operations: []
  };
  for (const n of e || [])
    switch (n.name) {
      case "styleId":
        t.styleId = n.value;
        break;
      case "alignment":
        t.alignment = (r = n.value) != null ? r : 0;
        break;
      case "leftIndent":
        t.leftIndent = n.value || 0;
        break;
      case "gapHalf":
        t.gapHalf = n.value || 0;
        break;
      case "cantSplit":
      case "header":
      case "rtl":
        t[n.name] = !!n.value;
        break;
      case "rowHeight":
        t.rowHeight = n.value || 0;
        break;
      case "positionCode":
        t.positionCode = n.value;
        break;
      case "absLeft":
        t.absLeft = n.value;
        break;
      case "absTop":
        t.absTop = n.value;
        break;
      case "distanceLeft":
        t.distanceLeft = n.value;
        break;
      case "distanceTop":
        t.distanceTop = n.value;
        break;
      case "tableWidth":
        t.tableWidth = n.value;
        break;
      case "autoFit":
        t.autoFit = n.value;
        break;
      case "widthBefore":
        t.widthBefore = n.value;
        break;
      case "widthAfter":
        t.widthAfter = n.value;
        break;
      case "defTable":
        t.defTable = n.value;
        break;
      default:
        t.operations.push(n);
        break;
    }
  return t;
}
function xt(e) {
  return e != null && e.inTable ? Math.max(1, e.itap || 0 || (e.dtap ? e.dtap : 1)) : 0;
}
function ls(e) {
  return Rr[e] || "left";
}
function us(e) {
  return Fr[e] || (e ? "single" : "none");
}
function fs(e) {
  return kr[e] || "top";
}
function Z(e, t, r) {
  if (!t)
    return;
  const n = Math.max(0, t.first || 0), s = Math.max(n, t.lim || n);
  for (let o = n; o < s && o < e.length; o += 1)
    r(e[o], o);
}
function ds(e) {
  const t = e == null ? void 0 : e.defTable;
  if (!t || !Array.isArray(t.cells))
    return [];
  const r = t.cells.map((n, s) => {
    var o, a, i, l, c, u, f, h, b, m;
    return {
      index: s,
      width: n == null ? void 0 : n.wWidth,
      ftsWidth: (o = n == null ? void 0 : n.tcgrf) == null ? void 0 : o.ftsWidth,
      borders: (n == null ? void 0 : n.borders) || {},
      merge: ((a = n == null ? void 0 : n.tcgrf) == null ? void 0 : a.horzMerge) || 0,
      vertMerge: ((i = n == null ? void 0 : n.tcgrf) == null ? void 0 : i.vertMerge) || 0,
      vertAlign: ((l = n == null ? void 0 : n.tcgrf) == null ? void 0 : l.vertAlign) || 0,
      fitText: !!((c = n == null ? void 0 : n.tcgrf) != null && c.fitText),
      noWrap: !!((u = n == null ? void 0 : n.tcgrf) != null && u.noWrap),
      hideMark: !!((f = n == null ? void 0 : n.tcgrf) != null && f.hideMark),
      textFlow: ((h = n == null ? void 0 : n.tcgrf) == null ? void 0 : h.textFlow) || 0,
      rightBoundary: (b = t.rgdxaCenter) == null ? void 0 : b[s + 1],
      leftBoundary: (m = t.rgdxaCenter) == null ? void 0 : m[s]
    };
  });
  for (const n of e.operations || [])
    switch (n.name) {
      case "merge":
        Z(r, n.value, (s, o) => {
          const a = n.value;
          o === a.first ? s.merge = 2 : s.merge = 1;
        });
        break;
      case "split":
        Z(r, n.value, (s) => {
          s.merge = 0;
        });
        break;
      case "cellWidth":
      case "columnWidth":
        Z(r, n.value.range, (s) => {
          const o = n.value;
          s.width = o.width, s.ftsWidth = o.ftsWidth;
        });
        break;
      case "vertMerge":
        Z(r, n.value.range, (s) => {
          s.vertMerge = n.value.value;
        });
        break;
      case "vertAlign":
        Z(r, n.value.range, (s) => {
          s.vertAlign = n.value.value;
        });
        break;
      case "setBorder":
        Z(r, n.value.range, (s) => {
          s.borders = oe(J({}, s.borders || {}), { all: n.value.border });
        });
        break;
      case "setShading":
        Z(r, n.value.range, (s) => {
          s.shading = n.value.value;
        });
        break;
      case "fitText":
        Z(r, n.value.range, (s) => {
          s.fitText = !!n.value.value;
        });
        break;
      case "cellNoWrap":
        Z(r, n.value.range, (s) => {
          s.noWrap = !!n.value.value;
        });
        break;
      case "textFlow":
        Z(r, n.value.range, (s) => {
          s.textFlow = n.value.value;
        });
        break;
    }
  return r;
}
function de(...e) {
  const t = /* @__PURE__ */ new Map();
  for (const r of e)
    for (const n of r || [])
      t.set(`${n.kind}:${n.name}`, n);
  return Array.from(t.values());
}
function ps(e) {
  const t = { para: [], char: [], table: [] };
  for (const r of e || [])
    r.kind === "para" ? t.para.push(r) : r.kind === "char" ? t.char.push(r) : r.kind === "table" && t.table.push(r);
  return t;
}
function hs(e, t) {
  const r = new k(e), n = r.u16(t), s = t + 2, o = n * 2, a = r.utf16le(s, o), i = s + o + 2;
  return { value: a.replace(/\u0000+$/, ""), nextOffset: i };
}
function ms(e, t) {
  const r = new k(e), n = r.u16(t), s = r.u16(t + 2), o = r.u16(t + 4);
  return {
    sti: n & 4095,
    flags1: n >> 12,
    stk: s & 15,
    istdBase: s >> 4 & 4095,
    cupx: o & 15,
    istdNext: o >> 4 & 4095,
    bchUpe: r.u16(t + 6),
    grfstd: r.u16(t + 8)
  };
}
function zt(e, t) {
  var c;
  const n = new k(e).u16(t), s = t + 2, o = s + n, a = e.subarray(s, Math.min(o, e.length));
  let i = 0, l = [];
  return a.length >= 2 && (i = a[0] | ((c = a[1]) != null ? c : 0) << 8, l = Fe(a, 2, a.length)), { cbUpx: n, styleId: i, properties: l, nextOffset: Pe(o) };
}
function ut(e, t) {
  const n = new k(e).u16(t), s = t + 2, o = s + n, a = e.subarray(s, Math.min(o, e.length)), i = Fe(a, 0, a.length);
  return { cbUpx: n, properties: i, nextOffset: Pe(o) };
}
function gs(e, t) {
  const n = new k(e).u16(t), s = t + 2, o = s + n, a = e.subarray(s, Math.min(o, e.length)), i = Fe(a, 0, a.length);
  return { cbUpx: n, properties: i, nextOffset: Pe(o) };
}
function bs(e, t, r) {
  if (!e.length)
    return {
      istd: r,
      empty: !0,
      name: "",
      stdfBase: { istdBase: 4095, istdNext: 0, stk: 1, cupx: 0 },
      paraProps: [],
      charProps: [],
      tableProps: []
    };
  const n = Math.max(10, Math.min(t || 10, e.length)), s = ms(e, 0);
  let o = n;
  const a = hs(e, o);
  o = a.nextOffset;
  let i = [], l = [], c = [];
  try {
    if (s.stk === 1) {
      if (s.cupx >= 1 && o + 2 <= e.length) {
        const u = zt(e, o);
        i = u.properties, o = u.nextOffset;
      }
      s.cupx >= 2 && o + 2 <= e.length && (l = ut(e, o).properties);
    } else if (s.stk === 2)
      o + 2 <= e.length && (l = ut(e, o).properties);
    else if (s.stk === 3) {
      if (s.cupx >= 1 && o + 2 <= e.length) {
        const u = gs(e, o);
        c = u.properties, o = u.nextOffset;
      }
      if (s.cupx >= 2 && o + 2 <= e.length) {
        const u = zt(e, o);
        i = u.properties, o = u.nextOffset;
      }
      s.cupx >= 3 && o + 2 <= e.length && (l = ut(e, o).properties);
    }
  } catch (u) {
  }
  return {
    istd: r,
    name: a.value,
    stdfBase: s,
    paraProps: i,
    charProps: l,
    tableProps: c,
    empty: !1
  };
}
function xs(e, t) {
  const r = t.fcStshf, n = t.lcbStshf;
  if (r == null || n == null || n <= 0)
    return { styles: /* @__PURE__ */ new Map(), header: null, resolveStyle(x) {
      return yt(/* @__PURE__ */ new Map(), x);
    } };
  const s = e.subarray(r, r + n), o = new k(s), a = o.u16(0), i = 2, l = o.u16(i + 0), c = o.u16(i + 2), u = o.u16(i + 10), f = o.u16(i + 12), h = o.u16(i + 14), b = { cbStshi: a, cstd: l, cbSTDBaseInFile: c, ftcAsci: u, ftcFE: f, ftcOther: h };
  let m = 2 + a;
  const y = /* @__PURE__ */ new Map();
  for (let x = 0; x < l && m + 2 <= s.length; x += 1) {
    const T = o.u16(m), S = m + 2, E = S + T;
    if (T === 0) {
      y.set(x, {
        istd: x,
        empty: !0,
        name: "",
        stdfBase: { istdBase: 4095, istdNext: 0, stk: 1, cupx: 0 },
        paraProps: [],
        charProps: [],
        tableProps: []
      }), m = Pe(E);
      continue;
    }
    const O = s.subarray(S, Math.min(E, s.length)), A = bs(O, c, x);
    y.set(x, A), m = Pe(E);
  }
  return {
    header: b,
    styles: y,
    resolveStyle(x) {
      return yt(y, x);
    }
  };
}
function yt(e, t, r = /* @__PURE__ */ new Set()) {
  var o;
  if (t == null || t === 4095 || r.has(t))
    return { paraProps: [], charProps: [], tableProps: [], styleIds: [] };
  const n = e.get(t);
  if (!n || n.empty)
    return { paraProps: [], charProps: [], tableProps: [], styleIds: [] };
  r.add(t);
  const s = yt(e, (o = n.stdfBase) == null ? void 0 : o.istdBase, r);
  return {
    styleIds: [...s.styleIds, t],
    paraProps: de(s.paraProps, n.paraProps),
    charProps: de(s.charProps, n.charProps),
    tableProps: de(s.tableProps, n.tableProps)
  };
}
function ys(e, t, r, n) {
  let s = (n == null ? void 0 : n.index) || 0;
  for (; s < e.length && e[s].cpEnd <= t; )
    s += 1;
  n && (n.index = s);
  const o = [];
  let a = s;
  for (; a < e.length && e[a].cpStart < r; )
    e[a].cpEnd > t && o.push(e[a]), a += 1;
  return o;
}
function ws(e) {
  return String(e || "").trim();
}
function qt(e) {
  const t = String(e || "").replace(/[\r\n\t]+/g, " ").replace(/\s+/g, " ").trim();
  if (!t)
    return null;
  const r = t.toUpperCase();
  if (r.startsWith("HYPERLINK")) {
    const n = t.slice(9).trim(), s = n.match(/"([^"]+)"/), o = (s == null ? void 0 : s[1]) || n.split(/\s+/)[0] || "";
    return o ? { type: "hyperlink", href: o } : null;
  }
  if (r.startsWith("INCLUDEPICTURE")) {
    const n = t.slice(14).trim(), s = n.match(/"([^"]+)"/);
    return { type: "includePicture", target: (s == null ? void 0 : s[1]) || n.split(/\s+/)[0] || "" };
  }
  return r.startsWith("EMBED") ? { type: "embed", raw: t } : r.startsWith("LINK") ? { type: "link", raw: t } : { type: "unknown", raw: t };
}
function Ts(e, t) {
  if (!t.text)
    return;
  const r = e[e.length - 1];
  if (r && r.type === "text" && r.href === t.href && fn(r.style, t.style)) {
    r.text += t.text;
    return;
  }
  e.push(t);
}
function ie(e, t, r) {
  if (!r)
    return;
  const n = e.length ? e[e.length - 1].nodes : t;
  r.type === "text" ? Ts(n, r) : n.push(r);
}
function Ps(e, t) {
  const r = [
    `_${t}`,
    `_${String(t)}`,
    `_${t.toString(16)}`,
    `_${t.toString(16).toUpperCase()}`
  ];
  for (const n of r)
    if (e.has(n))
      return e.get(n) || null;
  return null;
}
function Ss(e, t, r, n, s, o = {}) {
  return function(i) {
    const l = i == null ? void 0 : i.pictureOffset;
    if (l == null)
      return null;
    if (s.has(l))
      return s.get(l) || null;
    let c = null;
    const u = Ps(t, l);
    if ((i.ole2 || i.object || i.data) && (u != null && u.attachment) && (c = u.attachment, n.add(u.entry.name)), !c && (e != null && e.length)) {
      const f = ss(e, l, o);
      f && f.mime !== "application/octet-stream" ? c = f : !c && (u != null && u.attachment) ? (c = u.attachment, n.add(u.entry.name)) : f && (c = f);
    }
    return !c && (u != null && u.attachment) && (c = u.attachment, n.add(u.entry.name)), c && r.push(c), s.set(l, c), c;
  };
}
function Cs(e) {
  switch (e) {
    case q.nonBreakingHyphen:
      return "-";
    case q.nonRequiredHyphen:
      return "";
    case q.annotationRef:
      return "";
    default:
      return e;
  }
}
function Es(e, t) {
  var s;
  const r = [], n = [];
  for (const o of e)
    if (!(!o.text || o.state.hidden))
      for (const a of o.text) {
        const i = Cs(a);
        if (i === q.fieldStart) {
          n.push({ instruction: "", parsed: null, readingInstruction: !0, nodes: [] });
          continue;
        }
        if (i === q.fieldSeparator) {
          const c = n[n.length - 1];
          c && (c.parsed = qt(c.instruction), c.readingInstruction = !1);
          continue;
        }
        if (i === q.fieldEnd) {
          const c = n.pop();
          if (!c)
            continue;
          let u = c.nodes;
          if (c.readingInstruction && (c.parsed = qt(c.instruction)), ((s = c.parsed) == null ? void 0 : s.type) === "hyperlink") {
            const f = c.parsed.href;
            u = u.map((h) => h.type === "lineBreak" || h.type === "pageBreak" ? h : oe(J({}, h), { href: f }));
          }
          for (const f of u)
            ie(n, r, f);
          continue;
        }
        const l = n[n.length - 1];
        if (l != null && l.readingInstruction) {
          l.instruction += i;
          continue;
        }
        if (i === q.hardLineBreak) {
          ie(n, r, { type: "lineBreak" });
          continue;
        }
        if (i === q.pageBreak) {
          ie(n, r, { type: "pageBreak" });
          continue;
        }
        if (i === q.picture) {
          const c = t(o.state);
          (c == null ? void 0 : c.type) === "image" ? ie(n, r, { type: "image", asset: c, style: o.state }) : (c == null ? void 0 : c.type) === "attachment" && ie(n, r, { type: "attachment", asset: c, style: o.state });
          continue;
        }
        ie(n, r, {
          type: "text",
          text: ar(i),
          style: o.state
        });
      }
  for (; n.length; ) {
    const o = n.pop();
    for (const a of o.nodes)
      ie(n, r, a);
  }
  return r;
}
function Bs(e, t, r, n, s, o, a) {
  const i = ys(r, e.cpStart, e.cpEnd, a), l = /* @__PURE__ */ new Set([e.cpStart, e.cpEnd]);
  for (const f of i)
    l.add(Math.max(e.cpStart, f.cpStart)), l.add(Math.min(e.cpEnd, f.cpEnd));
  const c = Array.from(l).sort((f, h) => f - h), u = [];
  for (let f = 0; f < c.length - 1; f += 1) {
    const h = c[f], b = c[f + 1];
    if (b <= h)
      continue;
    const m = i.find((v) => v.cpStart <= h && v.cpEnd >= b), y = (m == null ? void 0 : m.properties) || [], x = lt(y), T = x.charStyleId != null ? n.resolveStyle(x.charStyleId).charProps : [], S = de(s, T, y), E = lt(S), O = o(E.fontFamilyId);
    O && (E.fontFamily = O.name || O.altName || void 0);
    const A = h - e.cpStart, D = b - e.cpStart, I = t.slice(A, D);
    if (!I)
      continue;
    const U = u[u.length - 1];
    if (U && fn(U.state, E)) {
      U.text += I, U.cpEnd = b;
      continue;
    }
    u.push({ cpStart: h, cpEnd: b, text: I, state: E });
  }
  if (!u.length && t) {
    const f = lt(s), h = o(f.fontFamilyId);
    h && (f.fontFamily = h.name || h.altName || void 0), u.push({ cpStart: e.cpStart, cpEnd: e.cpEnd, text: t, state: f });
  }
  return u;
}
function Fs(e, t, r, n, s, o, a) {
  var O, A;
  const i = ps(e.properties || []), l = e.styleId || ((O = i.para.find((D) => D.name === "styleId")) == null ? void 0 : O.value) || 0, c = r.resolveStyle(l), u = de(c.paraProps, i.para), f = cs(u), h = We(i.table), b = h.styleId != null ? r.resolveStyle(h.styleId).tableProps : [], m = de(b, i.table), y = We(m), x = c.charProps, S = Bs(e, t, s, r, x, (D) => n.byIndex(D), a), E = Es(S, o);
  return {
    id: G("para"),
    cpStart: e.cpStart,
    cpEnd: e.cpEnd,
    terminator: e.terminator || "",
    text: t,
    rawProperties: e.properties || [],
    styleId: l,
    styleName: ws((A = r.styles.get(l)) == null ? void 0 : A.name),
    paraProps: u,
    paraState: f,
    tableProps: m,
    tableState: y,
    segments: S,
    inlines: E
  };
}
function Rs(e) {
  var t, r, n, s;
  for (const o of e)
    for (let a = 0; a < o.cells.length; a += 1) {
      const i = o.cells[a], l = ((t = i.meta) == null ? void 0 : t.merge) || 0;
      if (i.colIndex = a, i.colspan = 1, i.rowspan = 1, i.hidden = !1, l === 1) {
        i.hidden = !0;
        continue;
      }
      if (l > 1) {
        let c = a + 1;
        for (; c < o.cells.length && (((r = o.cells[c].meta) == null ? void 0 : r.merge) || 0) === 1; )
          o.cells[c].hidden = !0, i.colspan += 1, c += 1;
      }
    }
  for (let o = 0; o < e.length; o += 1) {
    const a = e[o];
    for (const i of a.cells) {
      if (i.hidden)
        continue;
      const l = ((n = i.meta) == null ? void 0 : n.vertMerge) || 0;
      if (l === 1) {
        i.hidden = !0;
        continue;
      }
      if (l > 1) {
        let c = o + 1;
        for (; c < e.length; ) {
          let u = !0;
          for (let f = i.colIndex || 0; f < (i.colIndex || 0) + (i.colspan || 1); f += 1) {
            const h = e[c].cells[f];
            if (!h || (((s = h.meta) == null ? void 0 : s.vertMerge) || 0) !== 1) {
              u = !1;
              break;
            }
          }
          if (!u)
            break;
          for (let f = i.colIndex || 0; f < (i.colIndex || 0) + (i.colspan || 1); f += 1)
            e[c].cells[f].hidden = !0;
          i.rowspan = (i.rowspan || 1) + 1, c += 1;
        }
      }
    }
  }
}
function wt(e) {
  return {
    type: "paragraph",
    id: e.id,
    styleId: e.styleId,
    styleName: e.styleName,
    paraState: e.paraState,
    inlines: e.inlines,
    text: e.text
  };
}
function ks(e) {
  var a, i, l;
  const t = [];
  let r = { cells: [] }, n = [];
  for (const c of e)
    if (n.push(c), c.terminator === q.cellMark && (r.cells.push({
      id: G("cell"),
      paragraphs: n.map(wt),
      meta: null
    }), n = [], c.paraState.tableRowEnd || c.paraState.innerTableRowEnd)) {
      const u = ds(c.tableState);
      for (; u.length && r.cells.length > u.length && r.cells[r.cells.length - 1].paragraphs.every((h) => !h.text && !(h.inlines || []).length); )
        r.cells.pop();
      r.cells.forEach((h, b) => {
        h.meta = u[b] || { index: b };
      });
      const f = u.length ? (u[u.length - 1].rightBoundary || 0) - (u[0].leftBoundary || 0) : 0;
      t.push({
        id: G("row"),
        cells: r.cells,
        state: c.tableState,
        gridWidthTwips: f
      }), r = { cells: [] };
    }
  n.length && r.cells.push({ id: G("cell"), paragraphs: n.map(wt), meta: null }), r.cells.length && t.push({ id: G("row"), cells: r.cells, state: ((a = e[0]) == null ? void 0 : a.tableState) || We([]), gridWidthTwips: 0 }), Rs(t);
  const s = ((i = t.find((c) => c.gridWidthTwips)) == null ? void 0 : i.gridWidthTwips) || 0, o = Math.max(...e.map((c) => xt(c.paraState)), 1);
  return {
    type: "table",
    id: G("table"),
    depth: o,
    rows: t,
    state: ((l = t[0]) == null ? void 0 : l.state) || We([]),
    gridWidthTwips: s
  };
}
function As(e) {
  const t = [];
  let r = 0;
  for (; r < e.length; ) {
    const n = e[r];
    if (xt(n.paraState) <= 0) {
      t.push(wt(n)), r += 1;
      continue;
    }
    const o = [];
    for (; r < e.length && xt(e[r].paraState) > 0; )
      o.push(e[r]), r += 1;
    t.push(ks(o));
  }
  return t;
}
function vs(e, t = {}) {
  const r = [], n = gr(e, t);
  r.push(...n.warnings || []);
  const s = n.getStream("/WordDocument");
  if (!s)
    throw new Error("Missing WordDocument stream");
  const o = vr(s);
  if (o.base.wIdent !== 42476 && Le(r, `Unexpected FIB identifier: 0x${o.base.wIdent.toString(16)}`), o.base.fEncrypted)
    throw new Error("Encrypted .doc files are not supported yet");
  const a = n.getStream(o.base.fWhichTblStm ? "/1Table" : "/0Table");
  if (!a)
    throw new Error("Missing table stream");
  const i = n.getStream("/Data") || new Uint8Array(0), l = Tr(a, o.fibRgFcLcb), c = Cr(s, l), u = c.join(""), f = o.fibRgLw.ccpText > 0 ? o.fibRgLw.ccpText : u.length, h = xs(a, o.fibRgFcLcb), b = Qr(a, o.fibRgFcLcb), m = qr(s, a, o, l).filter((P) => P.cpStart < f), y = Vr(s, a, o, l).filter((P) => P.cpStart < f).map((P) => oe(J({}, P), { cpEnd: Math.min(P.cpEnd, f) })), x = y.length ? y.map((P) => ({
    cpStart: P.cpStart,
    cpEnd: P.cpEnd,
    terminator: u[P.cpEnd - 1] || "",
    styleId: P.styleId,
    properties: P.properties
  })) : Br(u.slice(0, f)).map((P) => oe(J({}, P), { styleId: 0, properties: [] })), T = is(n), S = [], E = /* @__PURE__ */ new Set(), A = Ss(i, T, S, E, /* @__PURE__ */ new Map(), t), D = { index: 0 }, I = x.map((P) => {
    const j = Er(s, l, c, P.cpStart, P.cpEnd), z = P.terminator === q.paragraph || P.terminator === q.cellMark ? P.terminator : "", se = z && j.endsWith(z) ? j.slice(0, -1) : j;
    return Fs(oe(J({}, P), { terminator: z }), se, h, b, m, A, D);
  }), U = As(I), v = Array.from(T.values()).filter((P) => (P == null ? void 0 : P.attachment) && !E.has(P.entry.name)).map((P) => P.attachment);
  for (const P of v)
    S.push(P);
  return v.length && U.push({ type: "attachments", id: G("attachments"), items: v }), {
    kind: "msdoc",
    version: 1,
    warnings: r,
    meta: {
      fib: {
        wIdent: o.base.wIdent,
        nFib: o.base.nFib,
        fWhichTblStm: o.base.fWhichTblStm,
        fComplex: o.base.fComplex,
        fEncrypted: o.base.fEncrypted,
        ccpText: o.fibRgLw.ccpText
      },
      counts: {
        paragraphs: I.length,
        blocks: U.length,
        assets: S.length,
        styles: h.styles.size,
        fonts: b.fonts.length
      }
    },
    fonts: b.fonts,
    styles: Array.from(h.styles.values()).map((P) => {
      var j, z, se;
      return {
        istd: P.istd,
        name: P.name,
        type: (j = P.stdfBase) == null ? void 0 : j.stk,
        basedOn: (z = P.stdfBase) == null ? void 0 : z.istdBase,
        next: (se = P.stdfBase) == null ? void 0 : se.istdNext
      };
    }),
    blocks: U,
    assets: S
  };
}
const Se = {
  1: "#000000",
  2: "#0000ff",
  3: "#00ffff",
  4: "#00ff00",
  5: "#ff00ff",
  6: "#ff0000",
  7: "#ffff00",
  8: "#ffffff",
  9: "#000080",
  10: "#008080",
  11: "#008000",
  12: "#800080",
  13: "#800000",
  14: "#808000",
  15: "#808080",
  16: "#c0c0c0"
};
function Ce(e) {
  return Object.entries(e).filter(([, t]) => t != null && t !== "").map(([t, r]) => `${t}:${r}`).join(";");
}
function re(e) {
  var a;
  if (!e)
    return null;
  const t = Math.max(1, Math.round(((a = e.lineWidth) != null ? a : 8) / 8 * 1.3333)), r = e.borderType, n = r === 6 ? "double" : r === 3 ? "dotted" : r === 2 ? "dashed" : "solid", s = e.color, o = s && Se[s] || "#666";
  return `${t}px ${n} ${o}`;
}
function Os(e) {
  var f, h, b, m;
  const t = {
    "text-align": ls(e.alignment)
  }, r = ne(e.spacingBefore), n = ne(e.spacingAfter), s = ne(e.leftIndent), o = ne(e.rightIndent), a = ne(e.firstLineIndent);
  if (r && (t["margin-top"] = `${r}px`), n && (t["margin-bottom"] = `${n}px`), s && (t["margin-left"] = `${s}px`), o && (t["margin-right"] = `${o}px`), a && (t["text-indent"] = `${a}px`), e.lineSpacing) {
    const y = Math.abs(e.lineSpacing) / 240;
    y && (t["line-height"] = String(Math.max(1, y)));
  }
  e.keepLines && (t["break-inside"] = "avoid"), e.keepNext && (t["break-after"] = "avoid"), e.pageBreakBefore && (t["break-before"] = "page"), e.rtlPara && (t.direction = "rtl");
  const i = re((f = e.borders) == null ? void 0 : f.top), l = re((h = e.borders) == null ? void 0 : h.right), c = re((b = e.borders) == null ? void 0 : b.bottom), u = re((m = e.borders) == null ? void 0 : m.left);
  return i && (t["border-top"] = i), l && (t["border-right"] = l), c && (t["border-bottom"] = c), u && (t["border-left"] = u), t;
}
function Is(e) {
  const t = us(e);
  if (!e || t === "none")
    return {};
  const r = { "text-decoration-line": "underline" };
  return t === "double" || t === "wavy-double" ? r["text-decoration-style"] = "double" : t.includes("dot") || t === "dotted-heavy" ? r["text-decoration-style"] = "dotted" : t.includes("dash") ? r["text-decoration-style"] = "dashed" : t.includes("wave") || t.includes("wavy") ? r["text-decoration-style"] = "wavy" : r["text-decoration-style"] = "solid", r;
}
function gn(e) {
  var n;
  const t = {};
  (e.bold || e.boldBi) && (t["font-weight"] = "700"), (e.italic || e.italicBi) && (t["font-style"] = "italic"), (e.strike || e.doubleStrike) && (t["text-decoration-line"] = `${t["text-decoration-line"] ? `${t["text-decoration-line"]} ` : ""}line-through`), Object.assign(t, Is(e.underline)), e.fontSizeHalfPoints && (t["font-size"] = `${e.fontSizeHalfPoints / 2}pt`), e.fontFamily && (t["font-family"] = `'${String(e.fontFamily).replace(/'/g, "\\'")}', sans-serif`), e.colorIndex && Se[e.colorIndex] && (t.color = Se[e.colorIndex]);
  const r = typeof e.highlight == "number" ? e.highlight : (n = e.highlight) == null ? void 0 : n.index;
  return r && bt[r] && (t["background-color"] = bt[r]), e.smallCaps && (t["font-variant-caps"] = "small-caps"), e.caps && (t["text-transform"] = "uppercase"), e.scale && e.scale !== 100 && (t.display = "inline-block", t.transform = `scaleX(${e.scale / 100})`, t["transform-origin"] = "left center"), e.positionHalfPoints > 0 && (t["vertical-align"] = "super"), e.positionHalfPoints < 0 && (t["vertical-align"] = "sub"), e.outline && (t["text-shadow"] = "0 0 0.02em currentColor"), (e.shadow || e.emboss || e.imprint) && (t["text-shadow"] = t["text-shadow"] ? `${t["text-shadow"]}, 0.06em 0.06em 0.08em rgba(0,0,0,.25)` : "0.06em 0.06em 0.08em rgba(0,0,0,.25)"), e.rtl && (t.direction = "rtl"), t;
}
function Ds(e) {
  const t = te(e.text), r = gn(e.style);
  r["white-space"] = "break-spaces";
  const n = Ce(r), s = `<span${n ? ` style="${n}"` : ""}>${t}</span>`;
  return e.href ? `<a class="msdoc-link" href="${te(e.href)}" target="_blank" rel="noreferrer noopener">${s}</a>` : s;
}
function Ls(e) {
  const t = gn(e.style);
  t["max-width"] = "100%", t.height = "auto";
  const r = `<img class="msdoc-image" src="${te(e.asset.dataUrl)}" alt="" style="${Ce(t)}">`;
  return e.href ? `<a class="msdoc-link" href="${te(e.href)}" target="_blank" rel="noreferrer noopener">${r}</a>` : r;
}
function _s(e) {
  const t = te(e.asset.name || "embedded-file"), r = `<a class="msdoc-attachment" href="${te(e.asset.dataUrl)}" download="${t}">📎 ${t}</a>`;
  return e.href ? `<a class="msdoc-link" href="${te(e.href)}" target="_blank" rel="noreferrer noopener">${r}</a>` : r;
}
function Ms(e) {
  return e.map((t) => t.type === "text" ? Ds(t) : t.type === "image" ? Ls(t) : t.type === "attachment" ? _s(t) : t.type === "lineBreak" ? "<br>" : t.type === "pageBreak" ? '<span class="msdoc-page-break"></span>' : "").join("");
}
function bn(e, t = {}) {
  const r = t.inline ? "div" : "p", n = Ce(Os(e.paraState)), o = Ms(e.inlines || []) || "<br>", a = ["msdoc-paragraph"];
  return e.styleName && a.push(`msdoc-style-${un(e.styleName)}`), `<${r} class="${a.join(" ")}"${n ? ` style="${n}"` : ""}>${o}</${r}>`;
}
function Ns(e) {
  var u, f, h, b, m, y, x, T, S, E, O, A, D, I, U, v, P, j;
  const t = {}, r = ((u = e.meta) == null ? void 0 : u.rightBoundary) != null && ((f = e.meta) == null ? void 0 : f.leftBoundary) != null ? e.meta.rightBoundary - e.meta.leftBoundary : (h = e.meta) == null ? void 0 : h.width, n = ne(r);
  n && (t.width = `${n}px`), (b = e.meta) != null && b.noWrap && (t["white-space"] = "nowrap"), (m = e.meta) != null && m.fitText && (t["text-align"] = "justify"), ((y = e.meta) == null ? void 0 : y.vertAlign) != null && (t["vertical-align"] = fs(e.meta.vertAlign));
  const s = re((T = (x = e.meta) == null ? void 0 : x.borders) == null ? void 0 : T.all), o = re((E = (S = e.meta) == null ? void 0 : S.borders) == null ? void 0 : E.top) || s, a = re((A = (O = e.meta) == null ? void 0 : O.borders) == null ? void 0 : A.right) || s, i = re((I = (D = e.meta) == null ? void 0 : D.borders) == null ? void 0 : I.bottom) || s, l = re((v = (U = e.meta) == null ? void 0 : U.borders) == null ? void 0 : v.left) || s;
  o && (t["border-top"] = o), a && (t["border-right"] = a), i && (t["border-bottom"] = i), l && (t["border-left"] = l);
  const c = typeof ((P = e.meta) == null ? void 0 : P.shading) == "object" && ((j = e.meta) != null && j.shading) && "color" in e.meta.shading ? e.meta.shading.color : void 0;
  return c && Se[c] && (t["background-color"] = Se[c]), t;
}
function Us(e) {
  var s, o, a;
  const t = {}, r = ne(e.gridWidthTwips || ((o = (s = e.state) == null ? void 0 : s.tableWidth) == null ? void 0 : o.wWidth));
  r ? t.width = `${r}px` : t.width = "100%";
  const n = ne((a = e.state) == null ? void 0 : a.leftIndent);
  return n && (t["margin-left"] = `${n}px`), t["border-collapse"] = "collapse", t["table-layout"] = "fixed", t;
}
function $s(e) {
  const t = e.rows.map((r) => {
    var a;
    const n = (a = r.state) != null && a.rowHeight ? ne(Math.abs(r.state.rowHeight)) : null, s = n ? ` style="height:${n}px"` : "", o = r.cells.filter((i) => !i.hidden).map((i) => {
      var f, h;
      const l = [];
      ((f = i.colspan) != null ? f : 1) > 1 && l.push(` colspan="${i.colspan}"`), ((h = i.rowspan) != null ? h : 1) > 1 && l.push(` rowspan="${i.rowspan}"`);
      const c = Ce(Ns(i)), u = i.paragraphs.map((b) => bn(b, { inline: !0 })).join("");
      return `<td class="msdoc-cell"${l.join("")}${c ? ` style="${c}"` : ""}>${u || '<div class="msdoc-paragraph"><br></div>'}</td>`;
    }).join("");
    return `<tr class="msdoc-row"${s}>${o}</tr>`;
  }).join("");
  return `<table class="msdoc-table msdoc-table-depth-${e.depth}" style="${Ce(Us(e))}"><tbody>${t}</tbody></table>`;
}
function Hs(e) {
  return `<section class="msdoc-attachments"><div class="msdoc-attachments-title">Embedded attachments</div><ul>${e.items.map((r) => `<li><a class="msdoc-attachment" href="${te(r.dataUrl)}" download="${te(r.name || "embedded-file")}">📎 ${te(r.name || "embedded-file")}</a></li>`).join("")}</ul></section>`;
}
function xn() {
  return `
.msdoc-root{box-sizing:border-box;max-width:100%;padding:24px;background:#fff;color:#111;font:14px/1.6 -apple-system,BlinkMacSystemFont,"Segoe UI",sans-serif}
.msdoc-root *{box-sizing:border-box}
.msdoc-paragraph{margin:0 0 8px;white-space:normal;word-break:break-word;overflow-wrap:anywhere}
.msdoc-paragraph:last-child{margin-bottom:0}
.msdoc-table{margin:12px 0;border-collapse:collapse;border-spacing:0;max-width:100%}
.msdoc-cell{padding:6px 8px;vertical-align:top;word-break:break-word;overflow-wrap:anywhere}
.msdoc-link{color:#1a73e8;text-decoration:none}
.msdoc-link:hover{text-decoration:underline}
.msdoc-image{display:inline-block;vertical-align:middle}
.msdoc-attachment{display:inline-flex;align-items:center;gap:6px;padding:4px 8px;border:1px solid #d0d7de;border-radius:6px;background:#f6f8fa;color:#0969da;text-decoration:none}
.msdoc-attachments{margin-top:20px;padding-top:12px;border-top:1px solid #e5e7eb}
.msdoc-attachments-title{font-weight:600;margin-bottom:8px}
.msdoc-page-break{display:block;height:0;border-top:1px dashed #cbd5e1;margin:16px 0}
`;
}
function js(e, t = {}) {
  var s;
  const r = (s = t.css) != null ? s : xn();
  return {
    html: e.blocks.map((o) => o.type === "paragraph" ? bn(o) : o.type === "table" ? $s(o) : o.type === "attachments" ? Hs(o) : "").join(""),
    css: r,
    warnings: e.warnings || [],
    meta: e.meta,
    assets: e.assets || [],
    parsed: e
  };
}
function Ws(e) {
  return B(this, null, function* () {
    if (e instanceof ArrayBuffer)
      return e;
    if (ArrayBuffer.isView(e)) {
      const t = new Uint8Array(e.byteLength);
      return t.set(new Uint8Array(e.buffer, e.byteOffset, e.byteLength)), t.buffer;
    }
    if (typeof Blob != "undefined" && e instanceof Blob)
      return yield e.arrayBuffer();
    if (typeof e == "string") {
      const t = yield fetch(e);
      if (!t.ok)
        throw new Error(`Failed to fetch document: ${t.status}`);
      return yield t.arrayBuffer();
    }
    throw new TypeError("Unsupported input type");
  });
}
function zs(r) {
  return B(this, arguments, function* (e, t = {}) {
    const n = yield Ws(e);
    if (t.workerClient)
      return t.workerClient.parseToHtml(n, {
        parseOptions: t.parseOptions || {},
        renderOptions: t.renderOptions || {}
      });
    const s = vs(n, t.parseOptions || {});
    return js(s, t.renderOptions || {});
  });
}
const yn = '<span class="msdoc-page-break"></span>', qs = '<p class="msdoc-paragraph"><br></p>', Vs = `
.msdoc-stage{
  box-sizing:border-box;
  min-height:100%;
  padding:32px 24px 48px;
  background:#ececec;
  display:flex;
  flex-direction:column;
  align-items:center;
  gap:24px;
}
.msdoc-page{
  width:min(100%,794px);
}
.msdoc-page > .msdoc-root{
  width:100%;
  max-width:none;
  min-height:1123px;
  padding:96px 88px;
  background:#fff;
  border:1px solid #d9d9d9;
  box-shadow:0 1px 3px rgba(0,0,0,0.08), 0 12px 32px rgba(0,0,0,0.12);
}
.msdoc-page .msdoc-page-break{
  display:none;
}
@media (max-width: 860px){
  .msdoc-stage{
    padding:16px 12px 24px;
    gap:16px;
  }
  .msdoc-page{
    width:100%;
  }
  .msdoc-page > .msdoc-root{
    min-height:auto;
    padding:24px 20px;
    box-shadow:none;
  }
}
`;
function Js(e) {
  return e.replace(/<(p|table|section)([^>]*?)style="([^"]*?\bbreak-before\s*:\s*page;?[^"]*?)"([^>]*)>/gi, (t) => `${yn}${t}`);
}
function Ks(e) {
  return `<div class="msdoc-stage">${Js(e).split(yn).map((n) => `<section class="msdoc-page"><div class="msdoc-root">${n || qs}</div></section>`).join("")}</div>`;
}
function Gs(e, t) {
  return B(this, null, function* () {
    const r = yield zs(e, {
      renderOptions: {
        css: `${xn()}
${Vs}`
      }
    });
    return t.innerHTML = `<style data-msdoc>${r.css}</style>${Ks(r.html)}`, {
      $el: t,
      unmount() {
        t.innerHTML = "";
      }
    };
  });
}
const Xs = Be(() => import("./PptxRender.js"));
function Qs(e, t) {
  return B(this, null, function* () {
    return he({
      render: () => me(Xs, {
        data: e
      }, null)
    }).mount(t);
  });
}
const Ys = Be(() => import("./XlsxTable.js"));
function Vt(e, t, r) {
  return B(this, null, function* () {
    const n = he({
      render: () => me(Ys, {
        data: e,
        type: r
      }, null)
    });
    return n.mount(t), n;
  });
}
const Zs = Be(() => import("./PdfView.js"));
function eo(e, t) {
  return B(this, null, function* () {
    const r = he({
      render: () => me(Zs, {
        data: e
      }, null)
    });
    return r.mount(t), r;
  });
}
function to(e) {
  return B(this, null, function* () {
    return new Promise((t, r) => {
      const n = new FileReader();
      n.onload = (s) => {
        var o;
        return t((o = s.target) == null ? void 0 : o.result);
      }, n.onerror = (s) => r(s), n.readAsArrayBuffer(e);
    });
  });
}
function no(e) {
  return B(this, null, function* () {
    return new Promise((t, r) => {
      const n = new FileReader();
      n.onload = (s) => {
        var a;
        const o = (a = s.target) == null ? void 0 : a.result;
        typeof o == "string" && t(o);
      }, n.onerror = (s) => r(s), n.readAsDataURL(new Blob([e]));
    });
  });
}
function wn(e) {
  return B(this, null, function* () {
    return new Promise((t, r) => {
      const n = new FileReader();
      n.onload = (s) => {
        var a;
        const o = (a = s.target) == null ? void 0 : a.result;
        typeof o == "string" && t(o);
      }, n.onerror = (s) => r(s), n.readAsText(new Blob([e]), "utf-8");
    });
  });
}
const ro = Be(() => import("./ImageViewer.js"));
function so(e, t) {
  return B(this, null, function* () {
    const r = yield no(e), n = he({
      render: () => me(ro, {
        image: r
      }, null)
    });
    return n.mount(t), n;
  });
}
const oo = Be(() => import("./MarkdownViewer.js"));
function ao(e, t) {
  return B(this, null, function* () {
    const r = yield wn(e), n = he({
      render: () => me(oo, {
        data: r
      }, null)
    });
    return n.mount(t), n;
  });
}
const io = { class: "code-area" }, co = /* @__PURE__ */ ln({
  __name: "CodeViewer",
  props: {
    value: {}
  },
  setup(e) {
    return (t, r) => (Oe(), Ie("pre", io, "    " + De(t.value) + `
  `, 1));
  }
});
const Tn = (e, t) => {
  const r = e.__vccOpts || e;
  for (const [n, s] of t)
    r[n] = s;
  return r;
}, lo = /* @__PURE__ */ Tn(co, [["__scopeId", "data-v-3be501eb"]]);
function uo(e, t) {
  return B(this, null, function* () {
    const r = yield wn(e), n = he({
      render: () => me(lo, {
        value: r
      }, null)
    });
    return n.mount(t), n;
  });
}
function fo(e, t) {
  const r = document.createElement("video");
  r.width = 840, r.height = 480, r.controls = !0, Object.assign(r.style, {
    margin: "40px auto",
    display: "block"
  });
  const n = document.createElement("source");
  n.src = URL.createObjectURL(new Blob([e])), r.appendChild(n), t.appendChild(r);
}
const ve = (e) => ({
  $el: e,
  unmount() {
  }
}), po = [
  // 使用docxjs支持，目前效果最好的渲染器
  {
    accepts: ["docx"],
    handler: (e, t) => B(void 0, null, function* () {
      return yield sr(e, t), window.dispatchEvent(new Event("resize")), ve(t);
    })
  },
  {
    accepts: ["doc"],
    handler: (e, t) => B(void 0, null, function* () {
      return Gs(e, t);
    })
  },
  // 使用pptx2html，已通过默认值更替
  {
    accepts: ["pptx"],
    handler: (e, t) => B(void 0, null, function* () {
      return yield Qs(e, t), window.dispatchEvent(new Event("resize")), ve(t);
    })
  },
  // 使用exceljs + handsontable，全部样式
  {
    accepts: ["xlsx"],
    handler: (e, t) => B(void 0, null, function* () {
      return Vt(e, t, "xml");
    })
  },
  // 使用js-xlsx + handsontable，无样式
  {
    accepts: ["xlsm", "xlsb", "xls", "csv", "ods", "fods", "numbers"],
    handler: (e, t) => B(void 0, null, function* () {
      return Vt(e, t, "binary");
    })
  },
  // 使用pdfjs，渲染pdf，效果最好
  {
    accepts: ["pdf"],
    handler: (e, t) => B(void 0, null, function* () {
      return eo(e, t);
    })
  },
  // 图片过滤器
  {
    accepts: ["gif", "jpg", "jpeg", "bmp", "tiff", "tif", "png", "svg", "webp"],
    handler: (e, t) => B(void 0, null, function* () {
      return so(e, t);
    })
  },
  {
    accepts: ["md", "markdown"],
    handler: (e, t) => B(void 0, null, function* () {
      return ao(e, t);
    })
  },
  // 纯文本预览
  {
    accepts: ["txt", "json", "js", "css", "java", "py", "html", "jsx", "ts", "tsx", "xml", "log"],
    handler: (e, t) => B(void 0, null, function* () {
      return uo(e, t);
    })
  },
  // 视频预览，仅支持MP4
  {
    accepts: ["mp4"],
    handler: (e, t) => B(void 0, null, function* () {
      return fo(e, t), ve(t);
    })
  },
  // 错误处理
  {
    accepts: ["error"],
    handler: (e, t, r) => B(void 0, null, function* () {
      return t.innerHTML = `<div style='text-align: center; margin-top: 80px'>不支持.${r}格式的在线预览，请下载后预览或转换为支持的格式</div>
<div style='text-align: center'>支持doc、docx、xlsx、pptx、pdf，以及纯文本格式和各种图片格式的在线预览</div>`, ve(t);
    })
  }
], Pn = po.reduce((e, { accepts: t, handler: r }) => (t.forEach((n) => e.set(n, r)), e), /* @__PURE__ */ new Map()), Jt = Pn.get("error");
function ho(e) {
  const t = e.lastIndexOf(".");
  return e.substring(t + 1);
}
function mo(e, t, r) {
  return B(this, null, function* () {
    const n = Pn.get(t.toLowerCase());
    if (n)
      return n(e, r);
    if (Jt)
      return Jt(e, r, t);
  });
}
function Sn(e, t) {
  return function() {
    return e.apply(t, arguments);
  };
}
const { toString: go } = Object.prototype, { getPrototypeOf: Ft } = Object, Ke = ((e) => (t) => {
  const r = go.call(t);
  return e[r] || (e[r] = r.slice(8, -1).toLowerCase());
})(/* @__PURE__ */ Object.create(null)), Q = (e) => (e = e.toLowerCase(), (t) => Ke(t) === e), Ge = (e) => (t) => typeof t === e, { isArray: ge } = Array, Ee = Ge("undefined");
function bo(e) {
  return e !== null && !Ee(e) && e.constructor !== null && !Ee(e.constructor) && W(e.constructor.isBuffer) && e.constructor.isBuffer(e);
}
const Cn = Q("ArrayBuffer");
function xo(e) {
  let t;
  return typeof ArrayBuffer != "undefined" && ArrayBuffer.isView ? t = ArrayBuffer.isView(e) : t = e && e.buffer && Cn(e.buffer), t;
}
const yo = Ge("string"), W = Ge("function"), En = Ge("number"), Xe = (e) => e !== null && typeof e == "object", wo = (e) => e === !0 || e === !1, Me = (e) => {
  if (Ke(e) !== "object")
    return !1;
  const t = Ft(e);
  return (t === null || t === Object.prototype || Object.getPrototypeOf(t) === null) && !(Symbol.toStringTag in e) && !(Symbol.iterator in e);
}, To = Q("Date"), Po = Q("File"), So = Q("Blob"), Co = Q("FileList"), Eo = (e) => Xe(e) && W(e.pipe), Bo = (e) => {
  let t;
  return e && (typeof FormData == "function" && e instanceof FormData || W(e.append) && ((t = Ke(e)) === "formdata" || // detect form-data instance
  t === "object" && W(e.toString) && e.toString() === "[object FormData]"));
}, Fo = Q("URLSearchParams"), [Ro, ko, Ao, vo] = ["ReadableStream", "Request", "Response", "Headers"].map(Q), Oo = (e) => e.trim ? e.trim() : e.replace(/^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g, "");
function Re(e, t, { allOwnKeys: r = !1 } = {}) {
  if (e === null || typeof e == "undefined")
    return;
  let n, s;
  if (typeof e != "object" && (e = [e]), ge(e))
    for (n = 0, s = e.length; n < s; n++)
      t.call(null, e[n], n, e);
  else {
    const o = r ? Object.getOwnPropertyNames(e) : Object.keys(e), a = o.length;
    let i;
    for (n = 0; n < a; n++)
      i = o[n], t.call(null, e[i], i, e);
  }
}
function Bn(e, t) {
  t = t.toLowerCase();
  const r = Object.keys(e);
  let n = r.length, s;
  for (; n-- > 0; )
    if (s = r[n], t === s.toLowerCase())
      return s;
  return null;
}
const ce = (() => typeof globalThis != "undefined" ? globalThis : typeof self != "undefined" ? self : typeof window != "undefined" ? window : global)(), Fn = (e) => !Ee(e) && e !== ce;
function Tt() {
  const { caseless: e } = Fn(this) && this || {}, t = {}, r = (n, s) => {
    const o = e && Bn(t, s) || s;
    Me(t[o]) && Me(n) ? t[o] = Tt(t[o], n) : Me(n) ? t[o] = Tt({}, n) : ge(n) ? t[o] = n.slice() : t[o] = n;
  };
  for (let n = 0, s = arguments.length; n < s; n++)
    arguments[n] && Re(arguments[n], r);
  return t;
}
const Io = (e, t, r, { allOwnKeys: n } = {}) => (Re(t, (s, o) => {
  r && W(s) ? e[o] = Sn(s, r) : e[o] = s;
}, { allOwnKeys: n }), e), Do = (e) => (e.charCodeAt(0) === 65279 && (e = e.slice(1)), e), Lo = (e, t, r, n) => {
  e.prototype = Object.create(t.prototype, n), e.prototype.constructor = e, Object.defineProperty(e, "super", {
    value: t.prototype
  }), r && Object.assign(e.prototype, r);
}, _o = (e, t, r, n) => {
  let s, o, a;
  const i = {};
  if (t = t || {}, e == null)
    return t;
  do {
    for (s = Object.getOwnPropertyNames(e), o = s.length; o-- > 0; )
      a = s[o], (!n || n(a, e, t)) && !i[a] && (t[a] = e[a], i[a] = !0);
    e = r !== !1 && Ft(e);
  } while (e && (!r || r(e, t)) && e !== Object.prototype);
  return t;
}, Mo = (e, t, r) => {
  e = String(e), (r === void 0 || r > e.length) && (r = e.length), r -= t.length;
  const n = e.indexOf(t, r);
  return n !== -1 && n === r;
}, No = (e) => {
  if (!e)
    return null;
  if (ge(e))
    return e;
  let t = e.length;
  if (!En(t))
    return null;
  const r = new Array(t);
  for (; t-- > 0; )
    r[t] = e[t];
  return r;
}, Uo = ((e) => (t) => e && t instanceof e)(typeof Uint8Array != "undefined" && Ft(Uint8Array)), $o = (e, t) => {
  const n = (e && e[Symbol.iterator]).call(e);
  let s;
  for (; (s = n.next()) && !s.done; ) {
    const o = s.value;
    t.call(e, o[0], o[1]);
  }
}, Ho = (e, t) => {
  let r;
  const n = [];
  for (; (r = e.exec(t)) !== null; )
    n.push(r);
  return n;
}, jo = Q("HTMLFormElement"), Wo = (e) => e.toLowerCase().replace(
  /[-_\s]([a-z\d])(\w*)/g,
  function(r, n, s) {
    return n.toUpperCase() + s;
  }
), Kt = (({ hasOwnProperty: e }) => (t, r) => e.call(t, r))(Object.prototype), zo = Q("RegExp"), Rn = (e, t) => {
  const r = Object.getOwnPropertyDescriptors(e), n = {};
  Re(r, (s, o) => {
    let a;
    (a = t(s, o, e)) !== !1 && (n[o] = a || s);
  }), Object.defineProperties(e, n);
}, qo = (e) => {
  Rn(e, (t, r) => {
    if (W(e) && ["arguments", "caller", "callee"].indexOf(r) !== -1)
      return !1;
    const n = e[r];
    if (W(n)) {
      if (t.enumerable = !1, "writable" in t) {
        t.writable = !1;
        return;
      }
      t.set || (t.set = () => {
        throw Error("Can not rewrite read-only method '" + r + "'");
      });
    }
  });
}, Vo = (e, t) => {
  const r = {}, n = (s) => {
    s.forEach((o) => {
      r[o] = !0;
    });
  };
  return ge(e) ? n(e) : n(String(e).split(t)), r;
}, Jo = () => {
}, Ko = (e, t) => e != null && Number.isFinite(e = +e) ? e : t, ft = "abcdefghijklmnopqrstuvwxyz", Gt = "0123456789", kn = {
  DIGIT: Gt,
  ALPHA: ft,
  ALPHA_DIGIT: ft + ft.toUpperCase() + Gt
}, Go = (e = 16, t = kn.ALPHA_DIGIT) => {
  let r = "";
  const { length: n } = t;
  for (; e--; )
    r += t[Math.random() * n | 0];
  return r;
};
function Xo(e) {
  return !!(e && W(e.append) && e[Symbol.toStringTag] === "FormData" && e[Symbol.iterator]);
}
const Qo = (e) => {
  const t = new Array(10), r = (n, s) => {
    if (Xe(n)) {
      if (t.indexOf(n) >= 0)
        return;
      if (!("toJSON" in n)) {
        t[s] = n;
        const o = ge(n) ? [] : {};
        return Re(n, (a, i) => {
          const l = r(a, s + 1);
          !Ee(l) && (o[i] = l);
        }), t[s] = void 0, o;
      }
    }
    return n;
  };
  return r(e, 0);
}, Yo = Q("AsyncFunction"), Zo = (e) => e && (Xe(e) || W(e)) && W(e.then) && W(e.catch), An = ((e, t) => e ? setImmediate : t ? ((r, n) => (ce.addEventListener("message", ({ source: s, data: o }) => {
  s === ce && o === r && n.length && n.shift()();
}, !1), (s) => {
  n.push(s), ce.postMessage(r, "*");
}))(`axios@${Math.random()}`, []) : (r) => setTimeout(r))(
  typeof setImmediate == "function",
  W(ce.postMessage)
), ea = typeof queueMicrotask != "undefined" ? queueMicrotask.bind(ce) : typeof process != "undefined" && process.nextTick || An, p = {
  isArray: ge,
  isArrayBuffer: Cn,
  isBuffer: bo,
  isFormData: Bo,
  isArrayBufferView: xo,
  isString: yo,
  isNumber: En,
  isBoolean: wo,
  isObject: Xe,
  isPlainObject: Me,
  isReadableStream: Ro,
  isRequest: ko,
  isResponse: Ao,
  isHeaders: vo,
  isUndefined: Ee,
  isDate: To,
  isFile: Po,
  isBlob: So,
  isRegExp: zo,
  isFunction: W,
  isStream: Eo,
  isURLSearchParams: Fo,
  isTypedArray: Uo,
  isFileList: Co,
  forEach: Re,
  merge: Tt,
  extend: Io,
  trim: Oo,
  stripBOM: Do,
  inherits: Lo,
  toFlatObject: _o,
  kindOf: Ke,
  kindOfTest: Q,
  endsWith: Mo,
  toArray: No,
  forEachEntry: $o,
  matchAll: Ho,
  isHTMLForm: jo,
  hasOwnProperty: Kt,
  hasOwnProp: Kt,
  // an alias to avoid ESLint no-prototype-builtins detection
  reduceDescriptors: Rn,
  freezeMethods: qo,
  toObjectSet: Vo,
  toCamelCase: Wo,
  noop: Jo,
  toFiniteNumber: Ko,
  findKey: Bn,
  global: ce,
  isContextDefined: Fn,
  ALPHABET: kn,
  generateString: Go,
  isSpecCompliantForm: Xo,
  toJSONObject: Qo,
  isAsyncFn: Yo,
  isThenable: Zo,
  setImmediate: An,
  asap: ea
};
function C(e, t, r, n, s) {
  Error.call(this), Error.captureStackTrace ? Error.captureStackTrace(this, this.constructor) : this.stack = new Error().stack, this.message = e, this.name = "AxiosError", t && (this.code = t), r && (this.config = r), n && (this.request = n), s && (this.response = s, this.status = s.status ? s.status : null);
}
p.inherits(C, Error, {
  toJSON: function() {
    return {
      // Standard
      message: this.message,
      name: this.name,
      // Microsoft
      description: this.description,
      number: this.number,
      // Mozilla
      fileName: this.fileName,
      lineNumber: this.lineNumber,
      columnNumber: this.columnNumber,
      stack: this.stack,
      // Axios
      config: p.toJSONObject(this.config),
      code: this.code,
      status: this.status
    };
  }
});
const vn = C.prototype, On = {};
[
  "ERR_BAD_OPTION_VALUE",
  "ERR_BAD_OPTION",
  "ECONNABORTED",
  "ETIMEDOUT",
  "ERR_NETWORK",
  "ERR_FR_TOO_MANY_REDIRECTS",
  "ERR_DEPRECATED",
  "ERR_BAD_RESPONSE",
  "ERR_BAD_REQUEST",
  "ERR_CANCELED",
  "ERR_NOT_SUPPORT",
  "ERR_INVALID_URL"
  // eslint-disable-next-line func-names
].forEach((e) => {
  On[e] = { value: e };
});
Object.defineProperties(C, On);
Object.defineProperty(vn, "isAxiosError", { value: !0 });
C.from = (e, t, r, n, s, o) => {
  const a = Object.create(vn);
  return p.toFlatObject(e, a, function(l) {
    return l !== Error.prototype;
  }, (i) => i !== "isAxiosError"), C.call(a, e.message, t, r, n, s), a.cause = e, a.name = e.name, o && Object.assign(a, o), a;
};
const ta = null;
function Pt(e) {
  return p.isPlainObject(e) || p.isArray(e);
}
function In(e) {
  return p.endsWith(e, "[]") ? e.slice(0, -2) : e;
}
function Xt(e, t, r) {
  return e ? e.concat(t).map(function(s, o) {
    return s = In(s), !r && o ? "[" + s + "]" : s;
  }).join(r ? "." : "") : t;
}
function na(e) {
  return p.isArray(e) && !e.some(Pt);
}
const ra = p.toFlatObject(p, {}, null, function(t) {
  return /^is[A-Z]/.test(t);
});
function Qe(e, t, r) {
  if (!p.isObject(e))
    throw new TypeError("target must be an object");
  t = t || new FormData(), r = p.toFlatObject(r, {
    metaTokens: !0,
    dots: !1,
    indexes: !1
  }, !1, function(y, x) {
    return !p.isUndefined(x[y]);
  });
  const n = r.metaTokens, s = r.visitor || u, o = r.dots, a = r.indexes, l = (r.Blob || typeof Blob != "undefined" && Blob) && p.isSpecCompliantForm(t);
  if (!p.isFunction(s))
    throw new TypeError("visitor must be a function");
  function c(m) {
    if (m === null)
      return "";
    if (p.isDate(m))
      return m.toISOString();
    if (!l && p.isBlob(m))
      throw new C("Blob is not supported. Use a Buffer instead.");
    return p.isArrayBuffer(m) || p.isTypedArray(m) ? l && typeof Blob == "function" ? new Blob([m]) : Buffer.from(m) : m;
  }
  function u(m, y, x) {
    let T = m;
    if (m && !x && typeof m == "object") {
      if (p.endsWith(y, "{}"))
        y = n ? y : y.slice(0, -2), m = JSON.stringify(m);
      else if (p.isArray(m) && na(m) || (p.isFileList(m) || p.endsWith(y, "[]")) && (T = p.toArray(m)))
        return y = In(y), T.forEach(function(E, O) {
          !(p.isUndefined(E) || E === null) && t.append(
            // eslint-disable-next-line no-nested-ternary
            a === !0 ? Xt([y], O, o) : a === null ? y : y + "[]",
            c(E)
          );
        }), !1;
    }
    return Pt(m) ? !0 : (t.append(Xt(x, y, o), c(m)), !1);
  }
  const f = [], h = Object.assign(ra, {
    defaultVisitor: u,
    convertValue: c,
    isVisitable: Pt
  });
  function b(m, y) {
    if (!p.isUndefined(m)) {
      if (f.indexOf(m) !== -1)
        throw Error("Circular reference detected in " + y.join("."));
      f.push(m), p.forEach(m, function(T, S) {
        (!(p.isUndefined(T) || T === null) && s.call(
          t,
          T,
          p.isString(S) ? S.trim() : S,
          y,
          h
        )) === !0 && b(T, y ? y.concat(S) : [S]);
      }), f.pop();
    }
  }
  if (!p.isObject(e))
    throw new TypeError("data must be an object");
  return b(e), t;
}
function Qt(e) {
  const t = {
    "!": "%21",
    "'": "%27",
    "(": "%28",
    ")": "%29",
    "~": "%7E",
    "%20": "+",
    "%00": "\0"
  };
  return encodeURIComponent(e).replace(/[!'()~]|%20|%00/g, function(n) {
    return t[n];
  });
}
function Rt(e, t) {
  this._pairs = [], e && Qe(e, this, t);
}
const Dn = Rt.prototype;
Dn.append = function(t, r) {
  this._pairs.push([t, r]);
};
Dn.toString = function(t) {
  const r = t ? function(n) {
    return t.call(this, n, Qt);
  } : Qt;
  return this._pairs.map(function(s) {
    return r(s[0]) + "=" + r(s[1]);
  }, "").join("&");
};
function sa(e) {
  return encodeURIComponent(e).replace(/%3A/gi, ":").replace(/%24/g, "$").replace(/%2C/gi, ",").replace(/%20/g, "+").replace(/%5B/gi, "[").replace(/%5D/gi, "]");
}
function Ln(e, t, r) {
  if (!t)
    return e;
  const n = r && r.encode || sa;
  p.isFunction(r) && (r = {
    serialize: r
  });
  const s = r && r.serialize;
  let o;
  if (s ? o = s(t, r) : o = p.isURLSearchParams(t) ? t.toString() : new Rt(t, r).toString(n), o) {
    const a = e.indexOf("#");
    a !== -1 && (e = e.slice(0, a)), e += (e.indexOf("?") === -1 ? "?" : "&") + o;
  }
  return e;
}
class oa {
  constructor() {
    this.handlers = [];
  }
  /**
   * Add a new interceptor to the stack
   *
   * @param {Function} fulfilled The function to handle `then` for a `Promise`
   * @param {Function} rejected The function to handle `reject` for a `Promise`
   *
   * @return {Number} An ID used to remove interceptor later
   */
  use(t, r, n) {
    return this.handlers.push({
      fulfilled: t,
      rejected: r,
      synchronous: n ? n.synchronous : !1,
      runWhen: n ? n.runWhen : null
    }), this.handlers.length - 1;
  }
  /**
   * Remove an interceptor from the stack
   *
   * @param {Number} id The ID that was returned by `use`
   *
   * @returns {Boolean} `true` if the interceptor was removed, `false` otherwise
   */
  eject(t) {
    this.handlers[t] && (this.handlers[t] = null);
  }
  /**
   * Clear all interceptors from the stack
   *
   * @returns {void}
   */
  clear() {
    this.handlers && (this.handlers = []);
  }
  /**
   * Iterate over all the registered interceptors
   *
   * This method is particularly useful for skipping over any
   * interceptors that may have become `null` calling `eject`.
   *
   * @param {Function} fn The function to call for each interceptor
   *
   * @returns {void}
   */
  forEach(t) {
    p.forEach(this.handlers, function(n) {
      n !== null && t(n);
    });
  }
}
const Yt = oa, _n = {
  silentJSONParsing: !0,
  forcedJSONParsing: !0,
  clarifyTimeoutError: !1
}, aa = typeof URLSearchParams != "undefined" ? URLSearchParams : Rt, ia = typeof FormData != "undefined" ? FormData : null, ca = typeof Blob != "undefined" ? Blob : null, la = {
  isBrowser: !0,
  classes: {
    URLSearchParams: aa,
    FormData: ia,
    Blob: ca
  },
  protocols: ["http", "https", "file", "blob", "url", "data"]
}, kt = typeof window != "undefined" && typeof document != "undefined", St = typeof navigator == "object" && navigator || void 0, ua = kt && (!St || ["ReactNative", "NativeScript", "NS"].indexOf(St.product) < 0), fa = (() => typeof WorkerGlobalScope != "undefined" && // eslint-disable-next-line no-undef
self instanceof WorkerGlobalScope && typeof self.importScripts == "function")(), da = kt && window.location.href || "http://localhost", pa = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  hasBrowserEnv: kt,
  hasStandardBrowserEnv: ua,
  hasStandardBrowserWebWorkerEnv: fa,
  navigator: St,
  origin: da
}, Symbol.toStringTag, { value: "Module" })), H = J(J({}, pa), la);
function ha(e, t) {
  return Qe(e, new H.classes.URLSearchParams(), Object.assign({
    visitor: function(r, n, s, o) {
      return H.isNode && p.isBuffer(r) ? (this.append(n, r.toString("base64")), !1) : o.defaultVisitor.apply(this, arguments);
    }
  }, t));
}
function ma(e) {
  return p.matchAll(/\w+|\[(\w*)]/g, e).map((t) => t[0] === "[]" ? "" : t[1] || t[0]);
}
function ga(e) {
  const t = {}, r = Object.keys(e);
  let n;
  const s = r.length;
  let o;
  for (n = 0; n < s; n++)
    o = r[n], t[o] = e[o];
  return t;
}
function Mn(e) {
  function t(r, n, s, o) {
    let a = r[o++];
    if (a === "__proto__")
      return !0;
    const i = Number.isFinite(+a), l = o >= r.length;
    return a = !a && p.isArray(s) ? s.length : a, l ? (p.hasOwnProp(s, a) ? s[a] = [s[a], n] : s[a] = n, !i) : ((!s[a] || !p.isObject(s[a])) && (s[a] = []), t(r, n, s[a], o) && p.isArray(s[a]) && (s[a] = ga(s[a])), !i);
  }
  if (p.isFormData(e) && p.isFunction(e.entries)) {
    const r = {};
    return p.forEachEntry(e, (n, s) => {
      t(ma(n), s, r, 0);
    }), r;
  }
  return null;
}
function ba(e, t, r) {
  if (p.isString(e))
    try {
      return (t || JSON.parse)(e), p.trim(e);
    } catch (n) {
      if (n.name !== "SyntaxError")
        throw n;
    }
  return (r || JSON.stringify)(e);
}
const At = {
  transitional: _n,
  adapter: ["xhr", "http", "fetch"],
  transformRequest: [function(t, r) {
    const n = r.getContentType() || "", s = n.indexOf("application/json") > -1, o = p.isObject(t);
    if (o && p.isHTMLForm(t) && (t = new FormData(t)), p.isFormData(t))
      return s ? JSON.stringify(Mn(t)) : t;
    if (p.isArrayBuffer(t) || p.isBuffer(t) || p.isStream(t) || p.isFile(t) || p.isBlob(t) || p.isReadableStream(t))
      return t;
    if (p.isArrayBufferView(t))
      return t.buffer;
    if (p.isURLSearchParams(t))
      return r.setContentType("application/x-www-form-urlencoded;charset=utf-8", !1), t.toString();
    let i;
    if (o) {
      if (n.indexOf("application/x-www-form-urlencoded") > -1)
        return ha(t, this.formSerializer).toString();
      if ((i = p.isFileList(t)) || n.indexOf("multipart/form-data") > -1) {
        const l = this.env && this.env.FormData;
        return Qe(
          i ? { "files[]": t } : t,
          l && new l(),
          this.formSerializer
        );
      }
    }
    return o || s ? (r.setContentType("application/json", !1), ba(t)) : t;
  }],
  transformResponse: [function(t) {
    const r = this.transitional || At.transitional, n = r && r.forcedJSONParsing, s = this.responseType === "json";
    if (p.isResponse(t) || p.isReadableStream(t))
      return t;
    if (t && p.isString(t) && (n && !this.responseType || s)) {
      const a = !(r && r.silentJSONParsing) && s;
      try {
        return JSON.parse(t);
      } catch (i) {
        if (a)
          throw i.name === "SyntaxError" ? C.from(i, C.ERR_BAD_RESPONSE, this, null, this.response) : i;
      }
    }
    return t;
  }],
  /**
   * A timeout in milliseconds to abort a request. If set to 0 (default) a
   * timeout is not created.
   */
  timeout: 0,
  xsrfCookieName: "XSRF-TOKEN",
  xsrfHeaderName: "X-XSRF-TOKEN",
  maxContentLength: -1,
  maxBodyLength: -1,
  env: {
    FormData: H.classes.FormData,
    Blob: H.classes.Blob
  },
  validateStatus: function(t) {
    return t >= 200 && t < 300;
  },
  headers: {
    common: {
      Accept: "application/json, text/plain, */*",
      "Content-Type": void 0
    }
  }
};
p.forEach(["delete", "get", "head", "post", "put", "patch"], (e) => {
  At.headers[e] = {};
});
const vt = At, xa = p.toObjectSet([
  "age",
  "authorization",
  "content-length",
  "content-type",
  "etag",
  "expires",
  "from",
  "host",
  "if-modified-since",
  "if-unmodified-since",
  "last-modified",
  "location",
  "max-forwards",
  "proxy-authorization",
  "referer",
  "retry-after",
  "user-agent"
]), ya = (e) => {
  const t = {};
  let r, n, s;
  return e && e.split(`
`).forEach(function(a) {
    s = a.indexOf(":"), r = a.substring(0, s).trim().toLowerCase(), n = a.substring(s + 1).trim(), !(!r || t[r] && xa[r]) && (r === "set-cookie" ? t[r] ? t[r].push(n) : t[r] = [n] : t[r] = t[r] ? t[r] + ", " + n : n);
  }), t;
}, Zt = Symbol("internals");
function Te(e) {
  return e && String(e).trim().toLowerCase();
}
function Ne(e) {
  return e === !1 || e == null ? e : p.isArray(e) ? e.map(Ne) : String(e);
}
function wa(e) {
  const t = /* @__PURE__ */ Object.create(null), r = /([^\s,;=]+)\s*(?:=\s*([^,;]+))?/g;
  let n;
  for (; n = r.exec(e); )
    t[n[1]] = n[2];
  return t;
}
const Ta = (e) => /^[-_a-zA-Z0-9^`|~,!#$%&'*+.]+$/.test(e.trim());
function dt(e, t, r, n, s) {
  if (p.isFunction(n))
    return n.call(this, t, r);
  if (s && (t = r), !!p.isString(t)) {
    if (p.isString(n))
      return t.indexOf(n) !== -1;
    if (p.isRegExp(n))
      return n.test(t);
  }
}
function Pa(e) {
  return e.trim().toLowerCase().replace(/([a-z\d])(\w*)/g, (t, r, n) => r.toUpperCase() + n);
}
function Sa(e, t) {
  const r = p.toCamelCase(" " + t);
  ["get", "set", "has"].forEach((n) => {
    Object.defineProperty(e, n + r, {
      value: function(s, o, a) {
        return this[n].call(this, t, s, o, a);
      },
      configurable: !0
    });
  });
}
class Ye {
  constructor(t) {
    t && this.set(t);
  }
  set(t, r, n) {
    const s = this;
    function o(i, l, c) {
      const u = Te(l);
      if (!u)
        throw new Error("header name must be a non-empty string");
      const f = p.findKey(s, u);
      (!f || s[f] === void 0 || c === !0 || c === void 0 && s[f] !== !1) && (s[f || l] = Ne(i));
    }
    const a = (i, l) => p.forEach(i, (c, u) => o(c, u, l));
    if (p.isPlainObject(t) || t instanceof this.constructor)
      a(t, r);
    else if (p.isString(t) && (t = t.trim()) && !Ta(t))
      a(ya(t), r);
    else if (p.isHeaders(t))
      for (const [i, l] of t.entries())
        o(l, i, n);
    else
      t != null && o(r, t, n);
    return this;
  }
  get(t, r) {
    if (t = Te(t), t) {
      const n = p.findKey(this, t);
      if (n) {
        const s = this[n];
        if (!r)
          return s;
        if (r === !0)
          return wa(s);
        if (p.isFunction(r))
          return r.call(this, s, n);
        if (p.isRegExp(r))
          return r.exec(s);
        throw new TypeError("parser must be boolean|regexp|function");
      }
    }
  }
  has(t, r) {
    if (t = Te(t), t) {
      const n = p.findKey(this, t);
      return !!(n && this[n] !== void 0 && (!r || dt(this, this[n], n, r)));
    }
    return !1;
  }
  delete(t, r) {
    const n = this;
    let s = !1;
    function o(a) {
      if (a = Te(a), a) {
        const i = p.findKey(n, a);
        i && (!r || dt(n, n[i], i, r)) && (delete n[i], s = !0);
      }
    }
    return p.isArray(t) ? t.forEach(o) : o(t), s;
  }
  clear(t) {
    const r = Object.keys(this);
    let n = r.length, s = !1;
    for (; n--; ) {
      const o = r[n];
      (!t || dt(this, this[o], o, t, !0)) && (delete this[o], s = !0);
    }
    return s;
  }
  normalize(t) {
    const r = this, n = {};
    return p.forEach(this, (s, o) => {
      const a = p.findKey(n, o);
      if (a) {
        r[a] = Ne(s), delete r[o];
        return;
      }
      const i = t ? Pa(o) : String(o).trim();
      i !== o && delete r[o], r[i] = Ne(s), n[i] = !0;
    }), this;
  }
  concat(...t) {
    return this.constructor.concat(this, ...t);
  }
  toJSON(t) {
    const r = /* @__PURE__ */ Object.create(null);
    return p.forEach(this, (n, s) => {
      n != null && n !== !1 && (r[s] = t && p.isArray(n) ? n.join(", ") : n);
    }), r;
  }
  [Symbol.iterator]() {
    return Object.entries(this.toJSON())[Symbol.iterator]();
  }
  toString() {
    return Object.entries(this.toJSON()).map(([t, r]) => t + ": " + r).join(`
`);
  }
  get [Symbol.toStringTag]() {
    return "AxiosHeaders";
  }
  static from(t) {
    return t instanceof this ? t : new this(t);
  }
  static concat(t, ...r) {
    const n = new this(t);
    return r.forEach((s) => n.set(s)), n;
  }
  static accessor(t) {
    const n = (this[Zt] = this[Zt] = {
      accessors: {}
    }).accessors, s = this.prototype;
    function o(a) {
      const i = Te(a);
      n[i] || (Sa(s, a), n[i] = !0);
    }
    return p.isArray(t) ? t.forEach(o) : o(t), this;
  }
}
Ye.accessor(["Content-Type", "Content-Length", "Accept", "Accept-Encoding", "User-Agent", "Authorization"]);
p.reduceDescriptors(Ye.prototype, ({ value: e }, t) => {
  let r = t[0].toUpperCase() + t.slice(1);
  return {
    get: () => e,
    set(n) {
      this[r] = n;
    }
  };
});
p.freezeMethods(Ye);
const X = Ye;
function pt(e, t) {
  const r = this || vt, n = t || r, s = X.from(n.headers);
  let o = n.data;
  return p.forEach(e, function(i) {
    o = i.call(r, o, s.normalize(), t ? t.status : void 0);
  }), s.normalize(), o;
}
function Nn(e) {
  return !!(e && e.__CANCEL__);
}
function be(e, t, r) {
  C.call(this, e == null ? "canceled" : e, C.ERR_CANCELED, t, r), this.name = "CanceledError";
}
p.inherits(be, C, {
  __CANCEL__: !0
});
function Un(e, t, r) {
  const n = r.config.validateStatus;
  !r.status || !n || n(r.status) ? e(r) : t(new C(
    "Request failed with status code " + r.status,
    [C.ERR_BAD_REQUEST, C.ERR_BAD_RESPONSE][Math.floor(r.status / 100) - 4],
    r.config,
    r.request,
    r
  ));
}
function Ca(e) {
  const t = /^([-+\w]{1,25})(:?\/\/|:)/.exec(e);
  return t && t[1] || "";
}
function Ea(e, t) {
  e = e || 10;
  const r = new Array(e), n = new Array(e);
  let s = 0, o = 0, a;
  return t = t !== void 0 ? t : 1e3, function(l) {
    const c = Date.now(), u = n[o];
    a || (a = c), r[s] = l, n[s] = c;
    let f = o, h = 0;
    for (; f !== s; )
      h += r[f++], f = f % e;
    if (s = (s + 1) % e, s === o && (o = (o + 1) % e), c - a < t)
      return;
    const b = u && c - u;
    return b ? Math.round(h * 1e3 / b) : void 0;
  };
}
function Ba(e, t) {
  let r = 0, n = 1e3 / t, s, o;
  const a = (c, u = Date.now()) => {
    r = u, s = null, o && (clearTimeout(o), o = null), e.apply(null, c);
  };
  return [(...c) => {
    const u = Date.now(), f = u - r;
    f >= n ? a(c, u) : (s = c, o || (o = setTimeout(() => {
      o = null, a(s);
    }, n - f)));
  }, () => s && a(s)];
}
const ze = (e, t, r = 3) => {
  let n = 0;
  const s = Ea(50, 250);
  return Ba((o) => {
    const a = o.loaded, i = o.lengthComputable ? o.total : void 0, l = a - n, c = s(l), u = a <= i;
    n = a;
    const f = {
      loaded: a,
      total: i,
      progress: i ? a / i : void 0,
      bytes: l,
      rate: c || void 0,
      estimated: c && i && u ? (i - a) / c : void 0,
      event: o,
      lengthComputable: i != null,
      [t ? "download" : "upload"]: !0
    };
    e(f);
  }, r);
}, en = (e, t) => {
  const r = e != null;
  return [(n) => t[0]({
    lengthComputable: r,
    total: e,
    loaded: n
  }), t[1]];
}, tn = (e) => (...t) => p.asap(() => e(...t)), Fa = H.hasStandardBrowserEnv ? ((e, t) => (r) => (r = new URL(r, H.origin), e.protocol === r.protocol && e.host === r.host && (t || e.port === r.port)))(
  new URL(H.origin),
  H.navigator && /(msie|trident)/i.test(H.navigator.userAgent)
) : () => !0, Ra = H.hasStandardBrowserEnv ? (
  // Standard browser envs support document.cookie
  {
    write(e, t, r, n, s, o) {
      const a = [e + "=" + encodeURIComponent(t)];
      p.isNumber(r) && a.push("expires=" + new Date(r).toGMTString()), p.isString(n) && a.push("path=" + n), p.isString(s) && a.push("domain=" + s), o === !0 && a.push("secure"), document.cookie = a.join("; ");
    },
    read(e) {
      const t = document.cookie.match(new RegExp("(^|;\\s*)(" + e + ")=([^;]*)"));
      return t ? decodeURIComponent(t[3]) : null;
    },
    remove(e) {
      this.write(e, "", Date.now() - 864e5);
    }
  }
) : (
  // Non-standard browser env (web workers, react-native) lack needed support.
  {
    write() {
    },
    read() {
      return null;
    },
    remove() {
    }
  }
);
function ka(e) {
  return /^([a-z][a-z\d+\-.]*:)?\/\//i.test(e);
}
function Aa(e, t) {
  return t ? e.replace(/\/?\/$/, "") + "/" + t.replace(/^\/+/, "") : e;
}
function $n(e, t) {
  return e && !ka(t) ? Aa(e, t) : t;
}
const nn = (e) => e instanceof X ? J({}, e) : e;
function ue(e, t) {
  t = t || {};
  const r = {};
  function n(c, u, f, h) {
    return p.isPlainObject(c) && p.isPlainObject(u) ? p.merge.call({ caseless: h }, c, u) : p.isPlainObject(u) ? p.merge({}, u) : p.isArray(u) ? u.slice() : u;
  }
  function s(c, u, f, h) {
    if (p.isUndefined(u)) {
      if (!p.isUndefined(c))
        return n(void 0, c, f, h);
    } else
      return n(c, u, f, h);
  }
  function o(c, u) {
    if (!p.isUndefined(u))
      return n(void 0, u);
  }
  function a(c, u) {
    if (p.isUndefined(u)) {
      if (!p.isUndefined(c))
        return n(void 0, c);
    } else
      return n(void 0, u);
  }
  function i(c, u, f) {
    if (f in t)
      return n(c, u);
    if (f in e)
      return n(void 0, c);
  }
  const l = {
    url: o,
    method: o,
    data: o,
    baseURL: a,
    transformRequest: a,
    transformResponse: a,
    paramsSerializer: a,
    timeout: a,
    timeoutMessage: a,
    withCredentials: a,
    withXSRFToken: a,
    adapter: a,
    responseType: a,
    xsrfCookieName: a,
    xsrfHeaderName: a,
    onUploadProgress: a,
    onDownloadProgress: a,
    decompress: a,
    maxContentLength: a,
    maxBodyLength: a,
    beforeRedirect: a,
    transport: a,
    httpAgent: a,
    httpsAgent: a,
    cancelToken: a,
    socketPath: a,
    responseEncoding: a,
    validateStatus: i,
    headers: (c, u, f) => s(nn(c), nn(u), f, !0)
  };
  return p.forEach(Object.keys(Object.assign({}, e, t)), function(u) {
    const f = l[u] || s, h = f(e[u], t[u], u);
    p.isUndefined(h) && f !== i || (r[u] = h);
  }), r;
}
const Hn = (e) => {
  const t = ue({}, e);
  let { data: r, withXSRFToken: n, xsrfHeaderName: s, xsrfCookieName: o, headers: a, auth: i } = t;
  t.headers = a = X.from(a), t.url = Ln($n(t.baseURL, t.url), e.params, e.paramsSerializer), i && a.set(
    "Authorization",
    "Basic " + btoa((i.username || "") + ":" + (i.password ? unescape(encodeURIComponent(i.password)) : ""))
  );
  let l;
  if (p.isFormData(r)) {
    if (H.hasStandardBrowserEnv || H.hasStandardBrowserWebWorkerEnv)
      a.setContentType(void 0);
    else if ((l = a.getContentType()) !== !1) {
      const [c, ...u] = l ? l.split(";").map((f) => f.trim()).filter(Boolean) : [];
      a.setContentType([c || "multipart/form-data", ...u].join("; "));
    }
  }
  if (H.hasStandardBrowserEnv && (n && p.isFunction(n) && (n = n(t)), n || n !== !1 && Fa(t.url))) {
    const c = s && o && Ra.read(o);
    c && a.set(s, c);
  }
  return t;
}, va = typeof XMLHttpRequest != "undefined", Oa = va && function(e) {
  return new Promise(function(r, n) {
    const s = Hn(e);
    let o = s.data;
    const a = X.from(s.headers).normalize();
    let { responseType: i, onUploadProgress: l, onDownloadProgress: c } = s, u, f, h, b, m;
    function y() {
      b && b(), m && m(), s.cancelToken && s.cancelToken.unsubscribe(u), s.signal && s.signal.removeEventListener("abort", u);
    }
    let x = new XMLHttpRequest();
    x.open(s.method.toUpperCase(), s.url, !0), x.timeout = s.timeout;
    function T() {
      if (!x)
        return;
      const E = X.from(
        "getAllResponseHeaders" in x && x.getAllResponseHeaders()
      ), A = {
        data: !i || i === "text" || i === "json" ? x.responseText : x.response,
        status: x.status,
        statusText: x.statusText,
        headers: E,
        config: e,
        request: x
      };
      Un(function(I) {
        r(I), y();
      }, function(I) {
        n(I), y();
      }, A), x = null;
    }
    "onloadend" in x ? x.onloadend = T : x.onreadystatechange = function() {
      !x || x.readyState !== 4 || x.status === 0 && !(x.responseURL && x.responseURL.indexOf("file:") === 0) || setTimeout(T);
    }, x.onabort = function() {
      x && (n(new C("Request aborted", C.ECONNABORTED, e, x)), x = null);
    }, x.onerror = function() {
      n(new C("Network Error", C.ERR_NETWORK, e, x)), x = null;
    }, x.ontimeout = function() {
      let O = s.timeout ? "timeout of " + s.timeout + "ms exceeded" : "timeout exceeded";
      const A = s.transitional || _n;
      s.timeoutErrorMessage && (O = s.timeoutErrorMessage), n(new C(
        O,
        A.clarifyTimeoutError ? C.ETIMEDOUT : C.ECONNABORTED,
        e,
        x
      )), x = null;
    }, o === void 0 && a.setContentType(null), "setRequestHeader" in x && p.forEach(a.toJSON(), function(O, A) {
      x.setRequestHeader(A, O);
    }), p.isUndefined(s.withCredentials) || (x.withCredentials = !!s.withCredentials), i && i !== "json" && (x.responseType = s.responseType), c && ([h, m] = ze(c, !0), x.addEventListener("progress", h)), l && x.upload && ([f, b] = ze(l), x.upload.addEventListener("progress", f), x.upload.addEventListener("loadend", b)), (s.cancelToken || s.signal) && (u = (E) => {
      x && (n(!E || E.type ? new be(null, e, x) : E), x.abort(), x = null);
    }, s.cancelToken && s.cancelToken.subscribe(u), s.signal && (s.signal.aborted ? u() : s.signal.addEventListener("abort", u)));
    const S = Ca(s.url);
    if (S && H.protocols.indexOf(S) === -1) {
      n(new C("Unsupported protocol " + S + ":", C.ERR_BAD_REQUEST, e));
      return;
    }
    x.send(o || null);
  });
}, Ia = (e, t) => {
  const { length: r } = e = e ? e.filter(Boolean) : [];
  if (t || r) {
    let n = new AbortController(), s;
    const o = function(c) {
      if (!s) {
        s = !0, i();
        const u = c instanceof Error ? c : this.reason;
        n.abort(u instanceof C ? u : new be(u instanceof Error ? u.message : u));
      }
    };
    let a = t && setTimeout(() => {
      a = null, o(new C(`timeout ${t} of ms exceeded`, C.ETIMEDOUT));
    }, t);
    const i = () => {
      e && (a && clearTimeout(a), a = null, e.forEach((c) => {
        c.unsubscribe ? c.unsubscribe(o) : c.removeEventListener("abort", o);
      }), e = null);
    };
    e.forEach((c) => c.addEventListener("abort", o));
    const { signal: l } = n;
    return l.unsubscribe = () => p.asap(i), l;
  }
}, Da = Ia, La = function* (e, t) {
  let r = e.byteLength;
  if (!t || r < t) {
    yield e;
    return;
  }
  let n = 0, s;
  for (; n < r; )
    s = n + t, yield e.slice(n, s), n = s;
}, _a = function(e, t) {
  return rt(this, null, function* () {
    try {
      for (var r = Lt(Ma(e)), n, s, o; n = !(s = yield new ae(r.next())).done; n = !1) {
        const a = s.value;
        yield* st(La(a, t));
      }
    } catch (s) {
      o = [s];
    } finally {
      try {
        n && (s = r.return) && (yield new ae(s.call(r)));
      } finally {
        if (o)
          throw o[0];
      }
    }
  });
}, Ma = function(e) {
  return rt(this, null, function* () {
    if (e[Symbol.asyncIterator]) {
      yield* st(e);
      return;
    }
    const t = e.getReader();
    try {
      for (; ; ) {
        const { done: r, value: n } = yield new ae(t.read());
        if (r)
          break;
        yield n;
      }
    } finally {
      yield new ae(t.cancel());
    }
  });
}, rn = (e, t, r, n) => {
  const s = _a(e, t);
  let o = 0, a, i = (c) => {
    a || (a = !0, n && n(c));
  };
  return new ReadableStream({
    pull(c) {
      return B(this, null, function* () {
        try {
          const { done: u, value: f } = yield s.next();
          if (u) {
            i(), c.close();
            return;
          }
          let h = f.byteLength;
          if (r) {
            let b = o += h;
            r(b);
          }
          c.enqueue(new Uint8Array(f));
        } catch (u) {
          throw i(u), u;
        }
      });
    },
    cancel(c) {
      return i(c), s.return();
    }
  }, {
    highWaterMark: 2
  });
}, Ze = typeof fetch == "function" && typeof Request == "function" && typeof Response == "function", jn = Ze && typeof ReadableStream == "function", Na = Ze && (typeof TextEncoder == "function" ? ((e) => (t) => e.encode(t))(new TextEncoder()) : (e) => B(void 0, null, function* () {
  return new Uint8Array(yield new Response(e).arrayBuffer());
})), Wn = (e, ...t) => {
  try {
    return !!e(...t);
  } catch (r) {
    return !1;
  }
}, Ua = jn && Wn(() => {
  let e = !1;
  const t = new Request(H.origin, {
    body: new ReadableStream(),
    method: "POST",
    get duplex() {
      return e = !0, "half";
    }
  }).headers.has("Content-Type");
  return e && !t;
}), sn = 64 * 1024, Ct = jn && Wn(() => p.isReadableStream(new Response("").body)), qe = {
  stream: Ct && ((e) => e.body)
};
Ze && ((e) => {
  ["text", "arrayBuffer", "blob", "formData", "stream"].forEach((t) => {
    !qe[t] && (qe[t] = p.isFunction(e[t]) ? (r) => r[t]() : (r, n) => {
      throw new C(`Response type '${t}' is not supported`, C.ERR_NOT_SUPPORT, n);
    });
  });
})(new Response());
const $a = (e) => B(void 0, null, function* () {
  if (e == null)
    return 0;
  if (p.isBlob(e))
    return e.size;
  if (p.isSpecCompliantForm(e))
    return (yield new Request(H.origin, {
      method: "POST",
      body: e
    }).arrayBuffer()).byteLength;
  if (p.isArrayBufferView(e) || p.isArrayBuffer(e))
    return e.byteLength;
  if (p.isURLSearchParams(e) && (e = e + ""), p.isString(e))
    return (yield Na(e)).byteLength;
}), Ha = (e, t) => B(void 0, null, function* () {
  const r = p.toFiniteNumber(e.getContentLength());
  return r == null ? $a(t) : r;
}), ja = Ze && ((e) => B(void 0, null, function* () {
  let {
    url: t,
    method: r,
    data: n,
    signal: s,
    cancelToken: o,
    timeout: a,
    onDownloadProgress: i,
    onUploadProgress: l,
    responseType: c,
    headers: u,
    withCredentials: f = "same-origin",
    fetchOptions: h
  } = Hn(e);
  c = c ? (c + "").toLowerCase() : "text";
  let b = Da([s, o && o.toAbortSignal()], a), m;
  const y = b && b.unsubscribe && (() => {
    b.unsubscribe();
  });
  let x;
  try {
    if (l && Ua && r !== "get" && r !== "head" && (x = yield Ha(u, n)) !== 0) {
      let A = new Request(t, {
        method: "POST",
        body: n,
        duplex: "half"
      }), D;
      if (p.isFormData(n) && (D = A.headers.get("content-type")) && u.setContentType(D), A.body) {
        const [I, U] = en(
          x,
          ze(tn(l))
        );
        n = rn(A.body, sn, I, U);
      }
    }
    p.isString(f) || (f = f ? "include" : "omit");
    const T = "credentials" in Request.prototype;
    m = new Request(t, oe(J({}, h), {
      signal: b,
      method: r.toUpperCase(),
      headers: u.normalize().toJSON(),
      body: n,
      duplex: "half",
      credentials: T ? f : void 0
    }));
    let S = yield fetch(m);
    const E = Ct && (c === "stream" || c === "response");
    if (Ct && (i || E && y)) {
      const A = {};
      ["status", "statusText", "headers"].forEach((v) => {
        A[v] = S[v];
      });
      const D = p.toFiniteNumber(S.headers.get("content-length")), [I, U] = i && en(
        D,
        ze(tn(i), !0)
      ) || [];
      S = new Response(
        rn(S.body, sn, I, () => {
          U && U(), y && y();
        }),
        A
      );
    }
    c = c || "text";
    let O = yield qe[p.findKey(qe, c) || "text"](S, e);
    return !E && y && y(), yield new Promise((A, D) => {
      Un(A, D, {
        data: O,
        headers: X.from(S.headers),
        status: S.status,
        statusText: S.statusText,
        config: e,
        request: m
      });
    });
  } catch (T) {
    throw y && y(), T && T.name === "TypeError" && /fetch/i.test(T.message) ? Object.assign(
      new C("Network Error", C.ERR_NETWORK, e, m),
      {
        cause: T.cause || T
      }
    ) : C.from(T, T && T.code, e, m);
  }
})), Et = {
  http: ta,
  xhr: Oa,
  fetch: ja
};
p.forEach(Et, (e, t) => {
  if (e) {
    try {
      Object.defineProperty(e, "name", { value: t });
    } catch (r) {
    }
    Object.defineProperty(e, "adapterName", { value: t });
  }
});
const on = (e) => `- ${e}`, Wa = (e) => p.isFunction(e) || e === null || e === !1, zn = {
  getAdapter: (e) => {
    e = p.isArray(e) ? e : [e];
    const { length: t } = e;
    let r, n;
    const s = {};
    for (let o = 0; o < t; o++) {
      r = e[o];
      let a;
      if (n = r, !Wa(r) && (n = Et[(a = String(r)).toLowerCase()], n === void 0))
        throw new C(`Unknown adapter '${a}'`);
      if (n)
        break;
      s[a || "#" + o] = n;
    }
    if (!n) {
      const o = Object.entries(s).map(
        ([i, l]) => `adapter ${i} ` + (l === !1 ? "is not supported by the environment" : "is not available in the build")
      );
      let a = t ? o.length > 1 ? `since :
` + o.map(on).join(`
`) : " " + on(o[0]) : "as no adapter specified";
      throw new C(
        "There is no suitable adapter to dispatch the request " + a,
        "ERR_NOT_SUPPORT"
      );
    }
    return n;
  },
  adapters: Et
};
function ht(e) {
  if (e.cancelToken && e.cancelToken.throwIfRequested(), e.signal && e.signal.aborted)
    throw new be(null, e);
}
function an(e) {
  return ht(e), e.headers = X.from(e.headers), e.data = pt.call(
    e,
    e.transformRequest
  ), ["post", "put", "patch"].indexOf(e.method) !== -1 && e.headers.setContentType("application/x-www-form-urlencoded", !1), zn.getAdapter(e.adapter || vt.adapter)(e).then(function(n) {
    return ht(e), n.data = pt.call(
      e,
      e.transformResponse,
      n
    ), n.headers = X.from(n.headers), n;
  }, function(n) {
    return Nn(n) || (ht(e), n && n.response && (n.response.data = pt.call(
      e,
      e.transformResponse,
      n.response
    ), n.response.headers = X.from(n.response.headers))), Promise.reject(n);
  });
}
const qn = "1.7.9", et = {};
["object", "boolean", "number", "function", "string", "symbol"].forEach((e, t) => {
  et[e] = function(n) {
    return typeof n === e || "a" + (t < 1 ? "n " : " ") + e;
  };
});
const cn = {};
et.transitional = function(t, r, n) {
  function s(o, a) {
    return "[Axios v" + qn + "] Transitional option '" + o + "'" + a + (n ? ". " + n : "");
  }
  return (o, a, i) => {
    if (t === !1)
      throw new C(
        s(a, " has been removed" + (r ? " in " + r : "")),
        C.ERR_DEPRECATED
      );
    return r && !cn[a] && (cn[a] = !0, console.warn(
      s(
        a,
        " has been deprecated since v" + r + " and will be removed in the near future"
      )
    )), t ? t(o, a, i) : !0;
  };
};
et.spelling = function(t) {
  return (r, n) => (console.warn(`${n} is likely a misspelling of ${t}`), !0);
};
function za(e, t, r) {
  if (typeof e != "object")
    throw new C("options must be an object", C.ERR_BAD_OPTION_VALUE);
  const n = Object.keys(e);
  let s = n.length;
  for (; s-- > 0; ) {
    const o = n[s], a = t[o];
    if (a) {
      const i = e[o], l = i === void 0 || a(i, o, e);
      if (l !== !0)
        throw new C("option " + o + " must be " + l, C.ERR_BAD_OPTION_VALUE);
      continue;
    }
    if (r !== !0)
      throw new C("Unknown option " + o, C.ERR_BAD_OPTION);
  }
}
const Ue = {
  assertOptions: za,
  validators: et
}, ee = Ue.validators;
class Ve {
  constructor(t) {
    this.defaults = t, this.interceptors = {
      request: new Yt(),
      response: new Yt()
    };
  }
  /**
   * Dispatch a request
   *
   * @param {String|Object} configOrUrl The config specific for this request (merged with this.defaults)
   * @param {?Object} config
   *
   * @returns {Promise} The Promise to be fulfilled
   */
  request(t, r) {
    return B(this, null, function* () {
      try {
        return yield this._request(t, r);
      } catch (n) {
        if (n instanceof Error) {
          let s = {};
          Error.captureStackTrace ? Error.captureStackTrace(s) : s = new Error();
          const o = s.stack ? s.stack.replace(/^.+\n/, "") : "";
          try {
            n.stack ? o && !String(n.stack).endsWith(o.replace(/^.+\n.+\n/, "")) && (n.stack += `
` + o) : n.stack = o;
          } catch (a) {
          }
        }
        throw n;
      }
    });
  }
  _request(t, r) {
    typeof t == "string" ? (r = r || {}, r.url = t) : r = t || {}, r = ue(this.defaults, r);
    const { transitional: n, paramsSerializer: s, headers: o } = r;
    n !== void 0 && Ue.assertOptions(n, {
      silentJSONParsing: ee.transitional(ee.boolean),
      forcedJSONParsing: ee.transitional(ee.boolean),
      clarifyTimeoutError: ee.transitional(ee.boolean)
    }, !1), s != null && (p.isFunction(s) ? r.paramsSerializer = {
      serialize: s
    } : Ue.assertOptions(s, {
      encode: ee.function,
      serialize: ee.function
    }, !0)), Ue.assertOptions(r, {
      baseUrl: ee.spelling("baseURL"),
      withXsrfToken: ee.spelling("withXSRFToken")
    }, !0), r.method = (r.method || this.defaults.method || "get").toLowerCase();
    let a = o && p.merge(
      o.common,
      o[r.method]
    );
    o && p.forEach(
      ["delete", "get", "head", "post", "put", "patch", "common"],
      (m) => {
        delete o[m];
      }
    ), r.headers = X.concat(a, o);
    const i = [];
    let l = !0;
    this.interceptors.request.forEach(function(y) {
      typeof y.runWhen == "function" && y.runWhen(r) === !1 || (l = l && y.synchronous, i.unshift(y.fulfilled, y.rejected));
    });
    const c = [];
    this.interceptors.response.forEach(function(y) {
      c.push(y.fulfilled, y.rejected);
    });
    let u, f = 0, h;
    if (!l) {
      const m = [an.bind(this), void 0];
      for (m.unshift.apply(m, i), m.push.apply(m, c), h = m.length, u = Promise.resolve(r); f < h; )
        u = u.then(m[f++], m[f++]);
      return u;
    }
    h = i.length;
    let b = r;
    for (f = 0; f < h; ) {
      const m = i[f++], y = i[f++];
      try {
        b = m(b);
      } catch (x) {
        y.call(this, x);
        break;
      }
    }
    try {
      u = an.call(this, b);
    } catch (m) {
      return Promise.reject(m);
    }
    for (f = 0, h = c.length; f < h; )
      u = u.then(c[f++], c[f++]);
    return u;
  }
  getUri(t) {
    t = ue(this.defaults, t);
    const r = $n(t.baseURL, t.url);
    return Ln(r, t.params, t.paramsSerializer);
  }
}
p.forEach(["delete", "get", "head", "options"], function(t) {
  Ve.prototype[t] = function(r, n) {
    return this.request(ue(n || {}, {
      method: t,
      url: r,
      data: (n || {}).data
    }));
  };
});
p.forEach(["post", "put", "patch"], function(t) {
  function r(n) {
    return function(o, a, i) {
      return this.request(ue(i || {}, {
        method: t,
        headers: n ? {
          "Content-Type": "multipart/form-data"
        } : {},
        url: o,
        data: a
      }));
    };
  }
  Ve.prototype[t] = r(), Ve.prototype[t + "Form"] = r(!0);
});
const $e = Ve;
class Ot {
  constructor(t) {
    if (typeof t != "function")
      throw new TypeError("executor must be a function.");
    let r;
    this.promise = new Promise(function(o) {
      r = o;
    });
    const n = this;
    this.promise.then((s) => {
      if (!n._listeners)
        return;
      let o = n._listeners.length;
      for (; o-- > 0; )
        n._listeners[o](s);
      n._listeners = null;
    }), this.promise.then = (s) => {
      let o;
      const a = new Promise((i) => {
        n.subscribe(i), o = i;
      }).then(s);
      return a.cancel = function() {
        n.unsubscribe(o);
      }, a;
    }, t(function(o, a, i) {
      n.reason || (n.reason = new be(o, a, i), r(n.reason));
    });
  }
  /**
   * Throws a `CanceledError` if cancellation has been requested.
   */
  throwIfRequested() {
    if (this.reason)
      throw this.reason;
  }
  /**
   * Subscribe to the cancel signal
   */
  subscribe(t) {
    if (this.reason) {
      t(this.reason);
      return;
    }
    this._listeners ? this._listeners.push(t) : this._listeners = [t];
  }
  /**
   * Unsubscribe from the cancel signal
   */
  unsubscribe(t) {
    if (!this._listeners)
      return;
    const r = this._listeners.indexOf(t);
    r !== -1 && this._listeners.splice(r, 1);
  }
  toAbortSignal() {
    const t = new AbortController(), r = (n) => {
      t.abort(n);
    };
    return this.subscribe(r), t.signal.unsubscribe = () => this.unsubscribe(r), t.signal;
  }
  /**
   * Returns an object that contains a new `CancelToken` and a function that, when called,
   * cancels the `CancelToken`.
   */
  static source() {
    let t;
    return {
      token: new Ot(function(s) {
        t = s;
      }),
      cancel: t
    };
  }
}
const qa = Ot;
function Va(e) {
  return function(r) {
    return e.apply(null, r);
  };
}
function Ja(e) {
  return p.isObject(e) && e.isAxiosError === !0;
}
const Bt = {
  Continue: 100,
  SwitchingProtocols: 101,
  Processing: 102,
  EarlyHints: 103,
  Ok: 200,
  Created: 201,
  Accepted: 202,
  NonAuthoritativeInformation: 203,
  NoContent: 204,
  ResetContent: 205,
  PartialContent: 206,
  MultiStatus: 207,
  AlreadyReported: 208,
  ImUsed: 226,
  MultipleChoices: 300,
  MovedPermanently: 301,
  Found: 302,
  SeeOther: 303,
  NotModified: 304,
  UseProxy: 305,
  Unused: 306,
  TemporaryRedirect: 307,
  PermanentRedirect: 308,
  BadRequest: 400,
  Unauthorized: 401,
  PaymentRequired: 402,
  Forbidden: 403,
  NotFound: 404,
  MethodNotAllowed: 405,
  NotAcceptable: 406,
  ProxyAuthenticationRequired: 407,
  RequestTimeout: 408,
  Conflict: 409,
  Gone: 410,
  LengthRequired: 411,
  PreconditionFailed: 412,
  PayloadTooLarge: 413,
  UriTooLong: 414,
  UnsupportedMediaType: 415,
  RangeNotSatisfiable: 416,
  ExpectationFailed: 417,
  ImATeapot: 418,
  MisdirectedRequest: 421,
  UnprocessableEntity: 422,
  Locked: 423,
  FailedDependency: 424,
  TooEarly: 425,
  UpgradeRequired: 426,
  PreconditionRequired: 428,
  TooManyRequests: 429,
  RequestHeaderFieldsTooLarge: 431,
  UnavailableForLegalReasons: 451,
  InternalServerError: 500,
  NotImplemented: 501,
  BadGateway: 502,
  ServiceUnavailable: 503,
  GatewayTimeout: 504,
  HttpVersionNotSupported: 505,
  VariantAlsoNegotiates: 506,
  InsufficientStorage: 507,
  LoopDetected: 508,
  NotExtended: 510,
  NetworkAuthenticationRequired: 511
};
Object.entries(Bt).forEach(([e, t]) => {
  Bt[t] = e;
});
const Ka = Bt;
function Vn(e) {
  const t = new $e(e), r = Sn($e.prototype.request, t);
  return p.extend(r, $e.prototype, t, { allOwnKeys: !0 }), p.extend(r, t, null, { allOwnKeys: !0 }), r.create = function(s) {
    return Vn(ue(e, s));
  }, r;
}
const N = Vn(vt);
N.Axios = $e;
N.CanceledError = be;
N.CancelToken = qa;
N.isCancel = Nn;
N.VERSION = qn;
N.toFormData = Qe;
N.AxiosError = C;
N.Cancel = N.CanceledError;
N.all = function(t) {
  return Promise.all(t);
};
N.spread = Va;
N.isAxiosError = Ja;
N.mergeConfig = ue;
N.AxiosHeaders = X;
N.formToJSON = (e) => Mn(p.isHTMLForm(e) ? new FormData(e) : e);
N.getAdapter = zn.getAdapter;
N.HttpStatusCode = Ka;
N.default = N;
const Ga = N, Xa = { class: "file-viewer" }, Qa = { class: "name" }, Ya = {
  key: 0,
  class: "content loading"
}, Za = /* @__PURE__ */ ln({
  __name: "FileViewer",
  props: {
    file: {},
    url: {}
  },
  setup(e) {
    const t = e, r = we(!1), n = we(""), s = we(""), o = we(""), a = we();
    return (() => {
      const i = {
        loading: "正在加载中，请耐心等待...",
        reading: "正在努力解析文件...",
        errorLoading: (f) => `加载文件异常：${f}`,
        errorReading: (f) => `读取文件异常：${f}`
      };
      let l;
      const c = {
        // 从url加载
        loadFromUrl() {
          return B(this, null, function* () {
            const { url: f } = t;
            if (!f)
              return;
            this.startLoading(i.loading);
            const h = f.substring(f.lastIndexOf("/") + 1);
            try {
              const { data: b } = yield Ga({ url: f, method: "get", responseType: "blob" });
              if (!b)
                return this.showError("文件下载失败");
              const m = this.wrap(b, h);
              return this.resolveFile(m);
            } catch (b) {
              this.showError(i.errorLoading(b));
            } finally {
              this.endLoading();
            }
          });
        },
        // 包装file
        wrap(f, h) {
          if (f instanceof File)
            return f;
          if (f instanceof Blob && h)
            return new File([f], h, {});
          if (f instanceof ArrayBuffer)
            return this.wrap(new Blob([f]));
          throw new Error("不支持的文件类型格式！");
        },
        // 处理并解析文件
        resolveFile(f) {
          return B(this, null, function* () {
            r.value && this.endLoading();
            const h = this.wrap(f);
            this.startLoading(i.reading);
            try {
              o.value = h.name && decodeURIComponent(h.name) || "";
              const b = yield to(h);
              b instanceof ArrayBuffer && (l = yield this.displayResult(b, h));
            } catch (b) {
              console.error(b), this.showError(i.errorReading(b));
            } finally {
              this.endLoading();
            }
          });
        },
        // 展示渲染最终效果
        displayResult(f, h) {
          const { name: b } = h, m = ho(b), y = a.value;
          if (!y)
            return Promise.resolve();
          l && (y.lastChild && y.removeChild(y.lastChild), l.unmount());
          const x = document.createElement("div");
          x.className = "file-render";
          const T = y.appendChild(x);
          return new Promise((S, E) => mo(f, m, T).then(S).catch(E));
        },
        showError(f) {
          n.value = f;
        },
        startLoading(f) {
          r.value = !0, s.value = f, n.value = "";
        },
        endLoading() {
          r.value = !1, s.value = "";
        }
      };
      tr(() => {
        if (t.file) {
          c.resolveFile(t.file);
          return;
        }
        c.loadFromUrl();
      }), _t(() => t.url, () => {
        t.file || c.loadFromUrl();
      }), _t(() => t.file, (f) => {
        if (f) {
          c.resolveFile(f);
          return;
        }
        c.loadFromUrl();
      });
    })(), (i, l) => (Oe(), Ie("div", Xa, [
      ot("div", Qa, De(o.value), 1),
      n.value ? (Oe(), Ie("div", Ya, De(n.value), 1)) : (Oe(), Ie(nr, { key: 1 }, [
        Mt(ot("div", { class: "content loading" }, De(s.value), 513), [
          [Nt, r.value]
        ]),
        Mt(ot("div", {
          class: "content",
          ref_key: "output",
          ref: a
        }, null, 512), [
          [Nt, !r.value]
        ])
      ], 64))
    ]));
  }
});
const ei = /* @__PURE__ */ Tn(Za, [["__scopeId", "data-v-2ee323c7"]]), ti = [
  ["file-viewer", ei]
];
class ni {
  constructor() {
    ye(this, "installed", !1);
  }
  install(t) {
    this.installed || (ti.forEach(([r, n]) => t.component(r, n)), this.installed = !0);
  }
}
const oi = new ni();
export {
  ei as F,
  Tn as _,
  oi as i
};
